diff --git a/backstage.json b/backstage.json
index 51e2c1de..94b96520 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "1.31.2"
+  "version": "1.33.1"
 }
diff --git a/package.json b/package.json
index 2f01c7c5..00d1ff57 100644
--- a/package.json
+++ b/package.json
@@ -5,3 +5,3 @@
   "engines": {
-    "node": "18 || 20"
+    "node": "20 || 22"
   },
diff --git a/packages/app/package.json b/packages/app/package.json
index 72c2e57c..d36e3be1 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -42,3 +42,2 @@
     "@material-ui/icons": "^4.9.1",
-    "history": "^5.0.0",
     "react": "^18.0.2",
@@ -46,4 +45,3 @@
     "react-router": "^6.3.0",
-    "react-router-dom": "^6.3.0",
-    "react-use": "^17.2.4"
+    "react-router-dom": "^6.3.0"
   },
diff --git a/packages/backend/Dockerfile b/packages/backend/Dockerfile
index 09024eb4..6a4c257a 100644
--- a/packages/backend/Dockerfile
+++ b/packages/backend/Dockerfile
@@ -47,3 +47,3 @@ ENV NODE_ENV=production
 # This disables node snapshot for Node 20 to work with the Scaffolder
-ENV NODE_OPTIONS "--no-node-snapshot"
+ENV NODE_OPTIONS="--no-node-snapshot"
 
diff --git a/packages/backend/package.json b/packages/backend/package.json
index c124c22c..016daaef 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -44,10 +44,6 @@
     "node-gyp": "^10.0.0",
-    "pg": "^8.11.3",
-    "winston": "^3.2.1"
+    "pg": "^8.11.3"
   },
   "devDependencies": {
-    "@backstage/cli": "backstage:^",
-    "@types/express": "^4.17.6",
-    "@types/express-serve-static-core": "^4.17.5",
-    "@types/luxon": "^2.0.4"
+    "@backstage/cli": "backstage:^"
   },
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index f57d4c5f..69a73511 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -12,6 +12,6 @@ const backend = createBackend();
 
-backend.add(import('@backstage/plugin-app-backend/alpha'));
-backend.add(import('@backstage/plugin-proxy-backend/alpha'));
-backend.add(import('@backstage/plugin-scaffolder-backend/alpha'));
-backend.add(import('@backstage/plugin-techdocs-backend/alpha'));
+backend.add(import('@backstage/plugin-app-backend'));
+backend.add(import('@backstage/plugin-proxy-backend'));
+backend.add(import('@backstage/plugin-scaffolder-backend'));
+backend.add(import('@backstage/plugin-techdocs-backend'));
 
@@ -24,3 +24,3 @@ backend.add(import('@backstage/plugin-auth-backend-module-guest-provider'));
 // catalog plugin
-backend.add(import('@backstage/plugin-catalog-backend/alpha'));
+backend.add(import('@backstage/plugin-catalog-backend'));
 backend.add(
@@ -33,3 +33,3 @@ backend.add(import('@backstage/plugin-catalog-backend-module-logs'));
 // permission plugin
-backend.add(import('@backstage/plugin-permission-backend/alpha'));
+backend.add(import('@backstage/plugin-permission-backend'));
 // See https://backstage.io/docs/permissions/getting-started for how to create your own permission policy
@@ -40,3 +40,3 @@ backend.add(
 // search plugin
-backend.add(import('@backstage/plugin-search-backend/alpha'));
+backend.add(import('@backstage/plugin-search-backend'));
 
@@ -44,10 +44,10 @@ backend.add(import('@backstage/plugin-search-backend/alpha'));
 // See https://backstage.io/docs/features/search/search-engines
-backend.add(import('@backstage/plugin-search-backend-module-pg/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-pg'));
 
 // search collators
-backend.add(import('@backstage/plugin-search-backend-module-catalog/alpha'));
-backend.add(import('@backstage/plugin-search-backend-module-techdocs/alpha'));
+backend.add(import('@backstage/plugin-search-backend-module-catalog'));
+backend.add(import('@backstage/plugin-search-backend-module-techdocs'));
 
 // kubernetes
-backend.add(import('@backstage/plugin-kubernetes-backend/alpha'));
+backend.add(import('@backstage/plugin-kubernetes-backend'));
 
