diff --git a/.gitignore b/.gitignore
index 5f5cc73..4adebc5 100644
--- a/.gitignore
+++ b/.gitignore
@@ -32,2 +32,5 @@ site
 # Local configuration files
-*.local.yaml
\ No newline at end of file
+*.local.yaml
+
+# Sensitive credentials
+*-credentials.yaml
diff --git a/app-config.yaml b/app-config.yaml
index 19f3819..d5ce05d 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -38,10 +38,12 @@ proxy:
 
+# Reference documentation http://backstage.io/docs/features/techdocs/configuration
+# Note: After experimenting with basic setup, use CI/CD to generate docs
+# and an external cloud storage when deploying TechDocs for production use-case.
+# https://backstage.io/docs/features/techdocs/how-to-guides#how-to-migrate-from-techdocs-basic-to-recommended-deployment-approach
 techdocs:
-  requestUrl: http://localhost:7000/api/techdocs
-  storageUrl: http://localhost:7000/api/techdocs/static/docs
-  builder: 'local'
+  builder: 'local' # Alternatives - 'external'
   generators:
-    techdocs: 'docker'
+    techdocs: 'docker' # Alternatives - 'local'
   publisher:
-    type: 'local'
+    type: 'local' # Alternatives - 'googleGcs' or 'awsS3'. Read documentation for using alternatives.
 
@@ -70,3 +72,3 @@ catalog:
     # Backstage example templates
-    - type: github
+    - type: url
       target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml
@@ -74,3 +76,3 @@ catalog:
         - allow: [Template]
-    - type: github
+    - type: url
       target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/springboot-grpc-template/template.yaml
@@ -78,3 +80,3 @@ catalog:
         - allow: [Template]
-    - type: github
+    - type: url
       target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/create-react-app/template.yaml
@@ -82,3 +84,3 @@ catalog:
         - allow: [Template]
-    - type: github
+    - type: url
       target: https://github.com/spotify/cookiecutter-golang/blob/master/template.yaml
@@ -86,3 +88,3 @@ catalog:
         - allow: [Template]
-    - type: github
+    - type: url
       target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/docs-template/template.yaml
diff --git a/catalog-info.yaml b/catalog-info.yaml
index 41794b4..841a480 100644
--- a/catalog-info.yaml
+++ b/catalog-info.yaml
@@ -8,3 +8,3 @@ metadata:
     # github.com/project-slug: backstage/backstage
-    # backstage.io/techdocs-ref: github:https://github.com/backstage/backstage.git
+    # backstage.io/techdocs-ref: url:https://github.com/backstage/backstage
 spec:
diff --git a/package.json b/package.json
index b6840f9..b36958a 100644
--- a/package.json
+++ b/package.json
@@ -30,3 +30,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.4.5",
+    "@backstage/cli": "^0.6.0",
     "@spotify/prettier-config": "^7.0.0",
@@ -47,3 +47,3 @@
     "transformModules": [
-      "@kyma-project/asyncapi-react"
+      "@asyncapi/react-component"
     ]
diff --git a/packages/app/package.json b/packages/app/package.json
index 638e420..662ae6a 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -8,17 +8,18 @@
     "@material-ui/icons": "^4.9.1",
-    "@backstage/cli": "^0.4.5",
-    "@backstage/core": "^0.4.3",
-    "@backstage/plugin-api-docs": "^0.4.2",
-    "@backstage/plugin-catalog": "^0.2.10",
-    "@backstage/plugin-catalog-import": "^0.3.3",
-    "@backstage/plugin-scaffolder": "^0.3.6",
-    "@backstage/plugin-techdocs": "^0.5.2",
-    "@backstage/catalog-model": "^0.6.0",
-    "@backstage/plugin-circleci": "^0.2.5",
-    "@backstage/plugin-tech-radar": "^0.3.2",
-    "@backstage/plugin-github-actions": "^0.2.6",
-    "@backstage/plugin-user-settings": "^0.2.3",
-    "@backstage/plugin-search": "^0.2.5",
-    "@backstage/test-utils": "^0.1.6",
-    "@backstage/theme": "^0.2.2",
+    "@backstage/cli": "^0.6.0",
+    "@backstage/core": "^0.6.1",
+    "@backstage/plugin-api-docs": "^0.4.5",
+    "@backstage/plugin-catalog": "^0.3.1",
+    "@backstage/plugin-catalog-react": "^0.0.3",
+    "@backstage/plugin-catalog-import": "^0.4.0",
+    "@backstage/plugin-scaffolder": "^0.5.0",
+    "@backstage/plugin-techdocs": "^0.5.6",
+    "@backstage/catalog-model": "^0.7.1",
+    "@backstage/plugin-circleci": "^0.2.8",
+    "@backstage/plugin-tech-radar": "^0.3.5",
+    "@backstage/plugin-github-actions": "^0.3.2",
+    "@backstage/plugin-user-settings": "^0.2.5",
+    "@backstage/plugin-search": "^0.3.0",
+    "@backstage/test-utils": "^0.1.7",
+    "@backstage/theme": "^0.2.3",
     "history": "^5.0.0",
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index 693e66e..711e501 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -6,3 +6,3 @@ import {
   SidebarPage,
-  createRouteRef,
+  FlatRoutes,
 } from '@backstage/core';
@@ -11,8 +11,9 @@ import * as plugins from './plugins';
 import { AppSidebar } from './sidebar';
-import { Route, Routes, Navigate } from 'react-router';
+import { Route, Navigate } from 'react-router';
 import { Router as CatalogRouter } from '@backstage/plugin-catalog';
 import { Router as DocsRouter } from '@backstage/plugin-techdocs';
-import { Router as ImportComponentRouter } from '@backstage/plugin-catalog-import';
+import { CatalogImportPage } from '@backstage/plugin-catalog-import';
 import { Router as TechRadarRouter } from '@backstage/plugin-tech-radar';
 import { SearchPage as SearchRouter } from '@backstage/plugin-search';
+import { Router as SettingsRouter } from '@backstage/plugin-user-settings';
 
@@ -29,8 +30,2 @@ const deprecatedAppRoutes = app.getRoutes();
 
-const catalogRouteRef = createRouteRef({
-  path: '/catalog',
-  title: 'Service Catalog',
-});
-
-
 const App = () => (
@@ -42,9 +37,9 @@ const App = () => (
         <AppSidebar />
-        <Routes>
+        <FlatRoutes>
           <Navigate key="/" to="/catalog" />
           <Route
-            path="/catalog/*"
+            path="/catalog"
             element={<CatalogRouter EntityPage={EntityPage} />}
           />
-          <Route path="/docs/*" element={<DocsRouter />} />
+          <Route path="/docs" element={<DocsRouter />} />
           <Route
@@ -53,6 +48,3 @@ const App = () => (
           />
-          <Route
-            path="/catalog-import"
-            element={<ImportComponentRouter catalogRouteRef={catalogRouteRef} />}
-          />
+          <Route path="/catalog-import" element={<CatalogImportPage />} />
           <Route
@@ -61,4 +53,5 @@ const App = () => (
           />
+          <Route path="/settings" element={<SettingsRouter />} />
           {deprecatedAppRoutes}
-        </Routes>
+        </FlatRoutes>
       </SidebarPage>
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index b5e384f..a3a3715 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -26,4 +26,6 @@ import {
   AboutCard, EntityPageLayout,
-  useEntity
 } from '@backstage/plugin-catalog';
+import {
+  useEntity
+} from '@backstage/plugin-catalog-react';
 import {
diff --git a/packages/app/src/plugins.ts b/packages/app/src/plugins.ts
index d3c9d6e..28b42d5 100644
--- a/packages/app/src/plugins.ts
+++ b/packages/app/src/plugins.ts
@@ -8 +8,3 @@ export { plugin as TechDocsPlugin } from '@backstage/plugin-techdocs';
 export { plugin as TechRadar } from '@backstage/plugin-tech-radar';
+export { plugin as UserSettings } from '@backstage/plugin-user-settings';
+
diff --git a/packages/app/src/sidebar.tsx b/packages/app/src/sidebar.tsx
index 55fa056..9f7973c 100644
--- a/packages/app/src/sidebar.tsx
+++ b/packages/app/src/sidebar.tsx
@@ -22,21 +22,2 @@ import { SidebarSearch } from '@backstage/plugin-search';
 
-export const AppSidebar = () => (
-  <Sidebar>
-    <SidebarLogo />
-    <SidebarSearch />
-    <SidebarDivider />
-    {/* Global nav, not org-specific */}
-    <SidebarItem icon={HomeIcon} to="./" text="Home" />
-    <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
-    <SidebarItem icon={LibraryBooks} to="/docs" text="Docs" />
-    <SidebarItem icon={CreateComponentIcon} to="create" text="Create..." />
-    <SidebarDivider />
-    <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
-    {/* End global nav */}
-    <SidebarDivider />
-    <SidebarSpace />
-    <SidebarDivider />
-    <SidebarSettings />
-  </Sidebar>
-);
 
@@ -74 +55,21 @@ const SidebarLogo = () => {
 };
+
+export const AppSidebar = () => (
+  <Sidebar>
+    <SidebarLogo />
+    <SidebarSearch />
+    <SidebarDivider />
+    {/* Global nav, not org-specific */}
+    <SidebarItem icon={HomeIcon} to="./" text="Home" />
+    <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
+    <SidebarItem icon={LibraryBooks} to="/docs" text="Docs" />
+    <SidebarItem icon={CreateComponentIcon} to="create" text="Create..." />
+    <SidebarDivider />
+    <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+    {/* End global nav */}
+    <SidebarDivider />
+    <SidebarSpace />
+    <SidebarDivider />
+    <SidebarSettings />
+  </Sidebar>
+);
diff --git a/packages/backend/package.json b/packages/backend/package.json
index c7d4168..a848ab2 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -20,13 +20,13 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.4.2",
-    "@backstage/catalog-model": "^0.6.0",
+    "@backstage/backend-common": "^0.5.3",
+    "@backstage/catalog-model": "^0.7.1",
     "@backstage/config": "^0.1.2",
-    "@backstage/plugin-app-backend": "^0.3.3",
-    "@backstage/plugin-auth-backend": "^0.2.9",
-    "@backstage/plugin-catalog-backend": "^0.5.2",
-    "@backstage/plugin-proxy-backend": "^0.2.3",
-    "@backstage/plugin-scaffolder-backend": "^0.4.0",
-    "@backstage/plugin-techdocs-backend": "^0.5.2",
-    "@octokit/rest": "^18.0.0",
-    "@gitbeaker/node": "^25.2.0",
+    "@backstage/plugin-app-backend": "^0.3.7",
+    "@backstage/plugin-auth-backend": "^0.3.0",
+    "@backstage/plugin-catalog-backend": "^0.6.1",
+    "@backstage/plugin-proxy-backend": "^0.2.4",
+    "@backstage/plugin-scaffolder-backend": "^0.7.0",
+    "@backstage/plugin-techdocs-backend": "^0.6.0",
+    "@gitbeaker/node": "^28.0.2",
+    "@octokit/rest": "^18.0.12",
     "dockerode": "^3.2.1",
@@ -39,7 +39,6 @@
   "devDependencies": {
-    "@backstage/cli": "^0.4.5",
+    "@backstage/cli": "^0.6.0",
     "@types/dockerode": "^3.2.1",
     "@types/express": "^4.17.6",
-    "@types/express-serve-static-core": "^4.17.5",
-    "@types/helmet": "^0.0.47"
+    "@types/express-serve-static-core": "^4.17.5"
   },
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 2dc69fe..d68f90c 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -3,12 +3,6 @@ import {
   createRouter,
-  FilePreparer,
-  GithubPreparer,
-  GitlabPreparer,
   Preparers,
   Publishers,
-  GithubPublisher,
-  GitlabPublisher,
   CreateReactAppTemplater,
   Templaters,
-  RepoVisibilityOptions,
   CatalogEntityClient,
@@ -16,4 +10,2 @@ import {
 import { SingleHostDiscovery } from '@backstage/backend-common';
-import { Octokit } from '@octokit/rest';
-import { Gitlab } from '@gitbeaker/node';
 import type { PluginEnvironment } from '../types';
@@ -24,2 +16,3 @@ export default async function createPlugin({
   config,
+  database,
 }: PluginEnvironment) {
@@ -28,2 +21,3 @@ export default async function createPlugin({
   const templaters = new Templaters();
+
   templaters.register('cookiecutter', cookiecutterTemplater);
@@ -31,70 +25,4 @@ export default async function createPlugin({
 
-  const filePreparer = new FilePreparer();
-  const githubPreparer = new GithubPreparer();
-  const gitlabPreparer = new GitlabPreparer(config);
-  const preparers = new Preparers();
-
-  preparers.register('file', filePreparer);
-  preparers.register('github', githubPreparer);
-  preparers.register('gitlab', gitlabPreparer);
-  preparers.register('gitlab/api', gitlabPreparer);
-
-  const publishers = new Publishers();
-
-  const githubConfig = config.getOptionalConfig('scaffolder.github');
-
-  if (githubConfig) {
-    try {
-      const repoVisibility = githubConfig.getString(
-        'visibility',
-      ) as RepoVisibilityOptions;
-
-      const githubToken = githubConfig.getString('token');
-      const githubHost = githubConfig.getOptionalString('host');
-      const githubClient = new Octokit({ auth: githubToken, baseUrl: githubHost });
-      const githubPublisher = new GithubPublisher({
-        client: githubClient,
-        token: githubToken,
-        repoVisibility,
-      });
-      publishers.register('file', githubPublisher);
-      publishers.register('github', githubPublisher);
-    } catch (e) {
-      const providerName = 'github';
-      if (process.env.NODE_ENV !== 'development') {
-        throw new Error(
-          `Failed to initialize ${providerName} scaffolding provider, ${e.message}`,
-        );
-      }
-
-      logger.warn(
-        `Skipping ${providerName} scaffolding provider, ${e.message}`,
-      );
-    }
-  }
-
-  const gitLabConfig = config.getOptionalConfig('scaffolder.gitlab.api');
-  if (gitLabConfig) {
-    try {
-      const gitLabToken = gitLabConfig.getString('token');
-      const gitLabClient = new Gitlab({
-        host: gitLabConfig.getOptionalString('baseUrl'),
-        token: gitLabToken,
-      });
-      const gitLabPublisher = new GitlabPublisher(gitLabClient, gitLabToken);
-      publishers.register('gitlab', gitLabPublisher);
-      publishers.register('gitlab/api', gitLabPublisher);
-    } catch (e) {
-      const providerName = 'gitlab';
-      if (process.env.NODE_ENV !== 'development') {
-        throw new Error(
-          `Failed to initialize ${providerName} scaffolding provider, ${e.message}`,
-        );
-      }
-
-      logger.warn(
-        `Skipping ${providerName} scaffolding provider, ${e.message}`,
-      );
-    }
-  }
+  const preparers = await Preparers.fromConfig(config, { logger });
+  const publishers = await Publishers.fromConfig(config, { logger });
 
@@ -113,2 +41,3 @@ export default async function createPlugin({
     entityClient,
+    database,
   });
