diff --git a/.prettierignore b/.prettierignore
new file mode 100644
index 0000000..dfb0f1c
--- /dev/null
+++ b/.prettierignore
@@ -0,0 +1,4 @@
+dist
+dist-types
+coverage
+.vscode
diff --git a/README.md b/README.md
index 07cf6fd..8c7c437 100644
--- a/README.md
+++ b/README.md
@@ -9,2 +9,2 @@ yarn install
 yarn dev
-```
\ No newline at end of file
+```
diff --git a/app-config.production.yaml b/app-config.production.yaml
index 92f4574..5e36c23 100644
--- a/app-config.production.yaml
+++ b/app-config.production.yaml
@@ -2,7 +2,7 @@ app:
   # Should be the same as backend.baseUrl when using the `app-backend` plugin
-  baseUrl: http://localhost:7000
+  baseUrl: http://localhost:7007
 
 backend:
-  baseUrl: http://localhost:7000
+  baseUrl: http://localhost:7007
   listen:
-    port: 7000
+    port: 7007
diff --git a/app-config.yaml b/app-config.yaml
index 03c75b8..1a622a2 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -8,5 +8,10 @@ organization:
 backend:
-  baseUrl: http://localhost:7000
+  # Used for enabling authentication, secret is shared by all backend plugins
+  # See backend-to-backend-auth.md in the docs for information on the format
+  # auth:
+  #   keys:
+  #     - secret: ${BACKEND_SECRET}
+  baseUrl: http://localhost:7007
   listen:
-    port: 7000
+    port: 7007
   csp:
@@ -46,4 +51,4 @@ techdocs:
   builder: 'local' # Alternatives - 'external'
-  generators:
-    techdocs: 'docker' # Alternatives - 'local'
+  generator:
+    runIn: 'docker' # Alternatives - 'local'
   publisher:
@@ -56,5 +61,3 @@ auth:
 scaffolder:
-  github:
-    token: ${GITHUB_TOKEN}
-    visibility: public # or 'internal' or 'private'
+  # see https://backstage.io/docs/features/software-templates/configuration for software template options
 
@@ -86,3 +89,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/react-ssr-template/template.yaml
       rules:
@@ -90,3 +93,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/springboot-grpc-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/springboot-grpc-template/template.yaml
       rules:
@@ -98,3 +101,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/docs-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/docs-template/template.yaml
       rules:
diff --git a/backstage.json b/backstage.json
new file mode 100644
index 0000000..b565e30
--- /dev/null
+++ b/backstage.json
@@ -0,0 +1,3 @@
+{
+  "version": "0.4.9"
+}
diff --git a/catalog-info.yaml b/catalog-info.yaml
index 841a480..adeb66d 100644
--- a/catalog-info.yaml
+++ b/catalog-info.yaml
@@ -7,4 +7,4 @@ metadata:
   # annotations:
-    # github.com/project-slug: backstage/backstage
-    # backstage.io/techdocs-ref: url:https://github.com/backstage/backstage
+  #   github.com/project-slug: backstage/backstage
+  #   backstage.io/techdocs-ref: dir:.
 spec:
diff --git a/package.json b/package.json
index 9df0dea..564d0d1 100644
--- a/package.json
+++ b/package.json
@@ -5,3 +5,3 @@
   "engines": {
-    "node": "12 || 14"
+    "node": "14 || 16"
   },
@@ -17,3 +17,3 @@
     "diff": "lerna run diff --",
-    "test": "lerna run test --since origin/master -- --coverage",
+    "test": "backstage-cli test",
     "test:all": "lerna run test -- --coverage",
@@ -21,9 +21,6 @@
     "lint:all": "lerna run lint --",
-    "create-plugin": "backstage-cli create-plugin --scope internal --no-private",
+    "prettier:check": "prettier --check .",
+    "create-plugin": "backstage-cli create-plugin --scope internal",
     "remove-plugin": "backstage-cli remove-plugin"
   },
-  "resolutions": {
-    "graphql-language-service-interface": "2.8.2",
-    "graphql-language-service-parser": "1.9.0"
-  },
   "workspaces": {
@@ -35,7 +32,7 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.3",
-    "@spotify/prettier-config": "^7.0.0",
+    "@backstage/cli": "^0.10.3",
+    "@spotify/prettier-config": "^12.0.0",
     "concurrently": "^6.0.0",
     "lerna": "^4.0.0",
-    "prettier": "^1.19.1"
+    "prettier": "^2.3.2"
   },
@@ -43,3 +40,3 @@
   "lint-staged": {
-    "*.{js,jsx,ts,tsx}": [
+    "*.{js,jsx,ts,tsx,mjs,cjs}": [
       "eslint --fix",
@@ -50,7 +47,2 @@
     ]
-  },
-  "jest": {
-    "transformModules": [
-      "@asyncapi/react-component"
-    ]
   }
diff --git a/packages/app/package.json b/packages/app/package.json
index 9cb0d98..ba91715 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,22 +6,22 @@
   "dependencies": {
-    "@backstage/catalog-model": "^0.9.0",
-    "@backstage/cli": "^0.7.3",
-    "@backstage/core-app-api": "^0.1.4",
-    "@backstage/core-components": "^0.1.5",
-    "@backstage/core-plugin-api": "^0.1.3",
-    "@backstage/integration-react": "^0.1.4",
-    "@backstage/plugin-api-docs": "^0.6.1",
-    "@backstage/plugin-catalog": "^0.6.6",
-    "@backstage/plugin-catalog-import": "^0.5.12",
-    "@backstage/plugin-catalog-react": "^0.2.6",
-    "@backstage/plugin-github-actions": "^0.4.12",
-    "@backstage/plugin-org": "^0.3.16",
-    "@backstage/plugin-scaffolder": "^0.10.0",
-    "@backstage/plugin-search": "^0.4.2",
-    "@backstage/plugin-tech-radar": "^0.4.2",
-    "@backstage/plugin-techdocs": "^0.9.9",
-    "@backstage/plugin-user-settings": "^0.2.12",
-    "@backstage/test-utils": "^0.1.14",
-    "@backstage/theme": "^0.2.8",
-    "@material-ui/core": "^4.11.0",
+    "@backstage/app-defaults": "^0.1.3",
+    "@backstage/catalog-model": "^0.9.8",
+    "@backstage/cli": "^0.10.3",
+    "@backstage/core-app-api": "^0.3.0",
+    "@backstage/core-components": "^0.8.2",
+    "@backstage/core-plugin-api": "^0.4.0",
+    "@backstage/integration-react": "^0.1.16",
+    "@backstage/plugin-api-docs": "^0.6.19",
+    "@backstage/plugin-catalog": "^0.7.5",
+    "@backstage/plugin-catalog-import": "^0.7.6",
+    "@backstage/plugin-catalog-react": "^0.6.8",
+    "@backstage/plugin-github-actions": "^0.4.28",
+    "@backstage/plugin-org": "^0.3.32",
+    "@backstage/plugin-scaffolder": "^0.11.15",
+    "@backstage/plugin-search": "^0.5.2",
+    "@backstage/plugin-tech-radar": "^0.5.0",
+    "@backstage/plugin-techdocs": "^0.12.11",
+    "@backstage/plugin-user-settings": "^0.3.14",
+    "@backstage/theme": "^0.2.14",
+    "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
@@ -35,2 +35,3 @@
   "devDependencies": {
+    "@backstage/test-utils": "^0.2.0",
     "@testing-library/jest-dom": "^5.10.1",
@@ -40,3 +41,3 @@
     "@types/node": "^14.14.32",
-    "@types/react-dom": "^16.9.8",
+    "@types/react-dom": "*",
     "cross-env": "^7.0.0",
@@ -49,7 +50,7 @@
     "build": "backstage-cli app:build",
-    "test": "backstage-cli test",
-    "lint": "backstage-cli lint",
     "clean": "backstage-cli clean",
+    "test": "backstage-cli test",
     "test:e2e": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:dev",
     "test:e2e:ci": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:run",
+    "lint": "backstage-cli lint",
     "cy:dev": "cypress open",
diff --git a/packages/app/public/index.html b/packages/app/public/index.html
index 5653173..1bd6001 100644
--- a/packages/app/public/index.html
+++ b/packages/app/public/index.html
@@ -49,8 +49,7 @@
     </style>
-    <title><%= app.title %></title>
-    <% if (app.googleAnalyticsTrackingId && typeof app.googleAnalyticsTrackingId
-    === 'string') { %>
+    <title><%= config.getString('app.title') %></title>
+    <% if (config.has('app.googleAnalyticsTrackingId')) { %>
     <script
       async
-      src="https://www.googletagmanager.com/gtag/js?id=<%= app.googleAnalyticsTrackingId %>"
+      src="https://www.googletagmanager.com/gtag/js?id=<%= config.getString('app.googleAnalyticsTrackingId') %>"
     ></script>
@@ -63,3 +62,6 @@
 
-      gtag('config', '<%= app.googleAnalyticsTrackingId %>');
+      gtag(
+        'config',
+        '<%= config.getString("app.googleAnalyticsTrackingId") %>',
+      );
     </script>
@@ -67,3 +69,3 @@
   </head>
-  <body style="margin: 0;">
+  <body style="margin: 0">
     <noscript>You need to enable JavaScript to run this app.</noscript>
diff --git a/packages/app/src/App.test.tsx b/packages/app/src/App.test.tsx
index 82bc479..b94cac7 100644
--- a/packages/app/src/App.test.tsx
+++ b/packages/app/src/App.test.tsx
@@ -12,5 +12,5 @@ describe('App', () => {
             app: { title: 'Test' },
-            backend: { baseUrl: 'http://localhost:7000' },
+            backend: { baseUrl: 'http://localhost:7007' },
             techdocs: {
-              storageUrl: 'http://localhost:7000/api/techdocs/static/docs',
+              storageUrl: 'http://localhost:7007/api/techdocs/static/docs',
             },
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index c513c7e..8a53583 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -8,10 +8,15 @@ import {
 } from '@backstage/plugin-catalog';
-import {CatalogImportPage, catalogImportPlugin} from '@backstage/plugin-catalog-import';
-import { 
-  ScaffolderPage, 
-  scaffolderPlugin 
-} from '@backstage/plugin-scaffolder';
+import {
+  CatalogImportPage,
+  catalogImportPlugin,
+} from '@backstage/plugin-catalog-import';
+import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
 import { SearchPage } from '@backstage/plugin-search';
 import { TechRadarPage } from '@backstage/plugin-tech-radar';
-import { TechdocsPage } from '@backstage/plugin-techdocs';
+import {
+  DefaultTechDocsHome,
+  TechDocsIndexPage,
+  techdocsPlugin,
+  TechDocsReaderPage,
+} from '@backstage/plugin-techdocs';
 import { UserSettingsPage } from '@backstage/plugin-user-settings';
@@ -19,2 +24,3 @@ import { apis } from './apis';
 import { entityPage } from './components/catalog/EntityPage';
+import { searchPage } from './components/search/SearchPage';
 import { Root } from './components/Root';
@@ -22,3 +28,4 @@ import { Root } from './components/Root';
 import { AlertDisplay, OAuthRequestDialog } from '@backstage/core-components';
-import { createApp, FlatRoutes } from '@backstage/core-app-api';
+import { createApp } from '@backstage/app-defaults';
+import { FlatRoutes } from '@backstage/core-app-api';
 
@@ -29,2 +36,3 @@ const app = createApp({
       createComponent: scaffolderPlugin.routes.root,
+      viewTechDoc: techdocsPlugin.routes.docRoot,
     });
@@ -44,3 +52,3 @@ const routes = (
   <FlatRoutes>
-    <Navigate key="/" to="/catalog" />
+    <Navigate key="/" to="catalog" />
     <Route path="/catalog" element={<CatalogIndexPage />} />
@@ -52,3 +60,9 @@ const routes = (
     </Route>
-    <Route path="/docs" element={<TechdocsPage />} />
+    <Route path="/docs" element={<TechDocsIndexPage />}>
+      <DefaultTechDocsHome />
+    </Route>
+    <Route
+      path="/docs/:namespace/:kind/:name/*"
+      element={<TechDocsReaderPage />}
+    />
     <Route path="/create" element={<ScaffolderPage />} />
@@ -60,3 +74,5 @@ const routes = (
     <Route path="/catalog-import" element={<CatalogImportPage />} />
-    <Route path="/search" element={<SearchPage />} />
+    <Route path="/search" element={<SearchPage />}>
+      {searchPage}
+    </Route>
     <Route path="/settings" element={<UserSettingsPage />} />
diff --git a/packages/app/src/apis.ts b/packages/app/src/apis.ts
index 080b20f..c89753a 100644
--- a/packages/app/src/apis.ts
+++ b/packages/app/src/apis.ts
@@ -1,5 +1,11 @@
 import {
-  ScmIntegrationsApi, scmIntegrationsApiRef
+  ScmIntegrationsApi,
+  scmIntegrationsApiRef,
+  ScmAuth,
 } from '@backstage/integration-react';
-import { AnyApiFactory, configApiRef, createApiFactory } from '@backstage/core-plugin-api';
+import {
+  AnyApiFactory,
+  configApiRef,
+  createApiFactory,
+} from '@backstage/core-plugin-api';
 
@@ -11,2 +17,3 @@ export const apis: AnyApiFactory[] = [
   }),
+  ScmAuth.createDefaultApiFactory(),
 ];
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index d8665a2..b4fa04f 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -27,3 +27,6 @@ import { NavLink } from 'react-router-dom';
 import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
-import { SidebarSearch } from '@backstage/plugin-search';
+import {
+  SidebarSearchModal,
+  SearchContextProvider,
+} from '@backstage/plugin-search';
 import {
@@ -76,6 +79,8 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       <SidebarLogo />
-      <SidebarSearch />
+      <SearchContextProvider>
+        <SidebarSearchModal />
+      </SearchContextProvider>
       <SidebarDivider />
       {/* Global nav, not org-specific */}
-      <SidebarItem icon={HomeIcon} to="/catalog" text="Home" />
+      <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
       <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index e3dd5be..d3b4b78 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -37,4 +37,8 @@ import {
   EntitySwitch,
+  EntityOrphanWarning,
+  EntityProcessingErrorsPanel,
   isComponentType,
   isKind,
+  hasCatalogProcessingErrors,
+  isOrphan,
 } from '@backstage/plugin-catalog';
@@ -80,4 +84,25 @@ const cicdContent = (
 
+const entityWarningContent = (
+  <>
+    <EntitySwitch>
+      <EntitySwitch.Case if={isOrphan}>
+        <Grid item xs={12}>
+          <EntityOrphanWarning />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+
+    <EntitySwitch>
+      <EntitySwitch.Case if={hasCatalogProcessingErrors}>
+        <Grid item xs={12}>
+          <EntityProcessingErrorsPanel />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+  </>
+);
+
 const overviewContent = (
   <Grid container spacing={3} alignItems="stretch">
+    {entityWarningContent}
     <Grid item md={6}>
@@ -159,2 +184,9 @@ const websiteEntityPage = (
 
+/**
+ * NOTE: This page is designed to work on small screens such as mobile devices.
+ * This is based on Material UI Grid. If breakpoints are used, each grid item must set the `xs` prop to a column size or to `true`,
+ * since this does not default. If no breakpoints are used, the items will equitably share the available space.
+ * https://material-ui.com/components/grid/#basic-grid.
+ */
+
 const defaultEntityPage = (
@@ -189,2 +221,3 @@ const apiPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item md={6}>
@@ -192,2 +225,5 @@ const apiPage = (
         </Grid>
+        <Grid item md={4} xs={12}>
+          <EntityLinksCard />
+        </Grid>
         <Grid container item md={12}>
@@ -217,2 +253,3 @@ const userPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item xs={12} md={6}>
@@ -232,2 +269,3 @@ const groupPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item xs={12} md={6}>
@@ -250,2 +288,3 @@ const systemPage = (
       <Grid container spacing={3} alignItems="stretch">
+        {entityWarningContent}
         <Grid item md={6}>
@@ -274,2 +313,3 @@ const domainPage = (
       <Grid container spacing={3} alignItems="stretch">
+        {entityWarningContent}
         <Grid item md={6}>
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index 7b3c2b2..95c8c64 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -4,2 +4,4 @@ import { makeStyles, Theme, Grid, List, Paper } from '@material-ui/core';
 import { CatalogResultListItem } from '@backstage/plugin-catalog';
+import { DocsResultListItem } from '@backstage/plugin-techdocs';
+
 import {
@@ -8,2 +10,3 @@ import {
   SearchResult,
+  SearchType,
   DefaultResultListItem,
@@ -36,3 +39,3 @@ const SearchPage = () => {
             <Paper className={classes.bar}>
-              <SearchBar debounceTime={100} />
+              <SearchBar />
             </Paper>
@@ -41,2 +44,7 @@ const SearchPage = () => {
             <Paper className={classes.filters}>
+              <SearchType
+                values={['techdocs', 'software-catalog']}
+                name="type"
+                defaultValue="software-catalog"
+              />
               <SearchFilter.Select
@@ -66,2 +74,9 @@ const SearchPage = () => {
                         );
+                      case 'techdocs':
+                        return (
+                          <DocsResultListItem
+                            key={document.location}
+                            result={document}
+                          />
+                        );
                       default:
diff --git a/packages/backend/README.md b/packages/backend/README.md
index 81e0f80..02426ef 100644
--- a/packages/backend/README.md
+++ b/packages/backend/README.md
@@ -38,3 +38,3 @@ the backend without using the auth or sentry features.
 
-The backend starts up on port 7000 per default.
+The backend starts up on port 7007 per default.
 
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 23213cb..330a913 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -6,5 +6,2 @@
   "private": true,
-  "engines": {
-    "node": "12 || 14"
-  },
   "scripts": {
@@ -20,17 +17,17 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.8.5",
-    "@backstage/catalog-model": "^0.9.0",
-    "@backstage/catalog-client": "^0.3.16",
-    "@backstage/config": "^0.1.5",
-    "@backstage/plugin-app-backend": "^0.3.14",
-    "@backstage/plugin-auth-backend": "^0.3.16",
-    "@backstage/plugin-catalog-backend": "^0.12.0",
-    "@backstage/plugin-proxy-backend": "^0.2.11",
-    "@backstage/plugin-scaffolder-backend": "^0.13.0",
-    "@backstage/plugin-search-backend": "^0.2.2",
-    "@backstage/plugin-search-backend-node": "^0.3.0",
-    "@backstage/plugin-techdocs-backend": "^0.8.6",
-    "@gitbeaker/node": "^30.2.0",
+    "@backstage/backend-common": "^0.10.0",
+    "@backstage/catalog-model": "^0.9.8",
+    "@backstage/catalog-client": "^0.5.3",
+    "@backstage/config": "^0.1.11",
+    "@backstage/plugin-app-backend": "^0.3.20",
+    "@backstage/plugin-auth-backend": "^0.5.2",
+    "@backstage/plugin-catalog-backend": "^0.19.3",
+    "@backstage/plugin-proxy-backend": "^0.2.15",
+    "@backstage/plugin-scaffolder-backend": "^0.15.18",
+    "@backstage/plugin-search-backend": "^0.3.0",
+    "@backstage/plugin-search-backend-node": "^0.4.3",
+    "@backstage/plugin-techdocs-backend": "^0.12.1",
+    "@gitbeaker/node": "^34.6.0",
     "@octokit/rest": "^18.5.3",
-    "dockerode": "^3.2.1",
+    "dockerode": "^3.3.1",
     "express": "^4.17.1",
@@ -42,4 +39,4 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.3",
-    "@types/dockerode": "^3.2.1",
+    "@backstage/cli": "^0.10.3",
+    "@types/dockerode": "^3.3.0",
     "@types/express": "^4.17.6",
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index f2b14b2..3f12122 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -19,2 +19,3 @@ import {
   UrlReaders,
+  ServerTokenManager,
 } from '@backstage/backend-common';
@@ -39,2 +40,3 @@ function makeCreateEnv(config: Config) {
   const databaseManager = DatabaseManager.fromConfig(config);
+  const tokenManager = ServerTokenManager.noop();
 
@@ -44,3 +46,3 @@ function makeCreateEnv(config: Config) {
     const cache = cacheManager.forPlugin(plugin);
-    return { logger, database, cache, config, reader, discovery };
+    return { logger, database, cache, config, reader, discovery, tokenManager };
   };
diff --git a/packages/backend/src/plugins/catalog.ts b/packages/backend/src/plugins/catalog.ts
index a132f0a..876cb6b 100644
--- a/packages/backend/src/plugins/catalog.ts
+++ b/packages/backend/src/plugins/catalog.ts
@@ -1,5 +1,3 @@
-import {
-  CatalogBuilder,
-  createRouter
-} from '@backstage/plugin-catalog-backend';
+import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
+import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
 import { Router } from 'express';
@@ -7,22 +5,10 @@ import { PluginEnvironment } from '../types';
 
-export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
   const builder = await CatalogBuilder.create(env);
-  const {
-    entitiesCatalog,
-    locationsCatalog,
-    locationService,
-    processingEngine,
-    locationAnalyzer,
-  } = await builder.build();
-
+  builder.addProcessor(new ScaffolderEntitiesProcessor());
+  const { processingEngine, router } = await builder.build();
   await processingEngine.start();
-
-  return await createRouter({
-    entitiesCatalog,
-    locationsCatalog,
-    locationService,
-    locationAnalyzer,
-    logger: env.logger,
-    config: env.config,
-  });
+  return router;
 }
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index a0201ce..6be2e97 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -1,9 +1,4 @@
-import {
-  DockerContainerRunner,
-  SingleHostDiscovery,
-} from '@backstage/backend-common';
+import { DockerContainerRunner } from '@backstage/backend-common';
 import { CatalogClient } from '@backstage/catalog-client';
-import {
-  createRouter,
-} from '@backstage/plugin-scaffolder-backend';
+import { createRouter } from '@backstage/plugin-scaffolder-backend';
 import Docker from 'dockerode';
@@ -17,2 +12,3 @@ export default async function createPlugin({
   reader,
+  discovery,
 }: PluginEnvironment): Promise<Router> {
@@ -20,4 +16,2 @@ export default async function createPlugin({
   const containerRunner = new DockerContainerRunner({ dockerClient });
-
-  const discovery = SingleHostDiscovery.fromConfig(config);
   const catalogClient = new CatalogClient({ discoveryApi: discovery });
diff --git a/packages/backend/src/plugins/search.ts b/packages/backend/src/plugins/search.ts
index 248ed37..f23b0c7 100644
--- a/packages/backend/src/plugins/search.ts
+++ b/packages/backend/src/plugins/search.ts
@@ -8,2 +8,3 @@ import { PluginEnvironment } from '../types';
 import { DefaultCatalogCollator } from '@backstage/plugin-catalog-backend';
+import { DefaultTechDocsCollator } from '@backstage/plugin-techdocs-backend';
 
@@ -12,2 +13,4 @@ export default async function createPlugin({
   discovery,
+  config,
+  tokenManager,
 }: PluginEnvironment) {
@@ -18,6 +21,19 @@ export default async function createPlugin({
   // Collators are responsible for gathering documents known to plugins. This
-  // particular collator gathers entities from the software catalog.
+  // collator gathers entities from the software catalog.
   indexBuilder.addCollator({
     defaultRefreshIntervalSeconds: 600,
-    collator: new DefaultCatalogCollator({ discovery }),
+    collator: DefaultCatalogCollator.fromConfig(config, {
+      discovery,
+      tokenManager,
+    }),
+  });
+
+  // collator gathers entities from techdocs.
+  indexBuilder.addCollator({
+    defaultRefreshIntervalSeconds: 600,
+    collator: DefaultTechDocsCollator.fromConfig(config, {
+      discovery,
+      logger,
+      tokenManager,
+    }),
   });
diff --git a/packages/backend/src/plugins/techdocs.ts b/packages/backend/src/plugins/techdocs.ts
index 906d86d..054c64d 100644
--- a/packages/backend/src/plugins/techdocs.ts
+++ b/packages/backend/src/plugins/techdocs.ts
@@ -16,2 +16,3 @@ export default async function createPlugin({
   reader,
+  cache,
 }: PluginEnvironment): Promise<Router> {
@@ -51,2 +52,3 @@ export default async function createPlugin({
     discovery,
+    cache,
   });
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index e5bfff2..b1e2e0a 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -6,2 +6,3 @@ import {
   PluginEndpointDiscovery,
+  TokenManager,
   UrlReader,
@@ -14,4 +15,5 @@ export type PluginEnvironment = {
   config: Config;
-  reader: UrlReader
+  reader: UrlReader;
   discovery: PluginEndpointDiscovery;
+  tokenManager: TokenManager;
 };
