diff --git a/app-config.yaml b/app-config.yaml
index 03c75b8..315e6ac 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -86,3 +86,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/react-ssr-template/template.yaml
       rules:
@@ -90,3 +90,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/springboot-grpc-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/springboot-grpc-template/template.yaml
       rules:
@@ -98,3 +98,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/docs-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/docs-template/template.yaml
       rules:
diff --git a/package.json b/package.json
index 016576d..9373787 100644
--- a/package.json
+++ b/package.json
@@ -35,3 +35,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.1",
+    "@backstage/cli": "^0.7.6",
     "@spotify/prettier-config": "^7.0.0",
diff --git a/packages/app/package.json b/packages/app/package.json
index fe7c01e..f1b4726 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,20 +6,22 @@
   "dependencies": {
-    "@backstage/catalog-model": "^0.8.3",
-    "@backstage/cli": "^0.7.1",
-    "@backstage/core": "^0.7.13",
-    "@backstage/integration-react": "^0.1.3",
-    "@backstage/plugin-api-docs": "^0.5.0",
-    "@backstage/plugin-catalog": "^0.6.3",
-    "@backstage/plugin-catalog-import": "^0.5.10",
-    "@backstage/plugin-catalog-react": "^0.2.3",
-    "@backstage/plugin-github-actions": "^0.4.9",
-    "@backstage/plugin-org": "^0.3.14",
-    "@backstage/plugin-scaffolder": "^0.9.8",
-    "@backstage/plugin-search": "^0.4.0",
-    "@backstage/plugin-tech-radar": "^0.4.0",
-    "@backstage/plugin-techdocs": "^0.9.6",
-    "@backstage/plugin-user-settings": "^0.2.11",
-    "@backstage/test-utils": "^0.1.13",
-    "@backstage/theme": "^0.2.8",
-    "@material-ui/core": "^4.11.0",
+    "@backstage/catalog-model": "^0.9.0",
+    "@backstage/cli": "^0.7.6",
+    "@backstage/core-app-api": "^0.1.6",
+    "@backstage/core-components": "^0.2.0",
+    "@backstage/core-plugin-api": "^0.1.4",
+    "@backstage/integration-react": "^0.1.5",
+    "@backstage/plugin-api-docs": "^0.6.4",
+    "@backstage/plugin-catalog": "^0.6.9",
+    "@backstage/plugin-catalog-import": "^0.5.15",
+    "@backstage/plugin-catalog-react": "^0.4.0",
+    "@backstage/plugin-github-actions": "^0.4.14",
+    "@backstage/plugin-org": "^0.3.18",
+    "@backstage/plugin-scaffolder": "^0.10.2",
+    "@backstage/plugin-search": "^0.4.5",
+    "@backstage/plugin-tech-radar": "^0.4.3",
+    "@backstage/plugin-techdocs": "^0.10.2",
+    "@backstage/plugin-user-settings": "^0.3.2",
+    "@backstage/test-utils": "^0.1.16",
+    "@backstage/theme": "^0.2.9",
+    "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index 07d030b..c513c7e 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -2,8 +2,2 @@ import React from 'react';
 import { Navigate, Route } from 'react-router';
-import {
-  AlertDisplay,
-  createApp,
-  FlatRoutes,
-  OAuthRequestDialog,
-} from '@backstage/core';
 import { apiDocsPlugin, ApiExplorerPage } from '@backstage/plugin-api-docs';
@@ -27,2 +21,5 @@ import { Root } from './components/Root';
 
+import { AlertDisplay, OAuthRequestDialog } from '@backstage/core-components';
+import { createApp, FlatRoutes } from '@backstage/core-app-api';
+
 const app = createApp({
diff --git a/packages/app/src/apis.ts b/packages/app/src/apis.ts
index d803416..080b20f 100644
--- a/packages/app/src/apis.ts
+++ b/packages/app/src/apis.ts
@@ -1,4 +1 @@
-import {
-  AnyApiFactory, configApiRef, createApiFactory
-} from '@backstage/core';
 import {
@@ -6,2 +3,3 @@ import {
 } from '@backstage/integration-react';
+import { AnyApiFactory, configApiRef, createApiFactory } from '@backstage/core-plugin-api';
 
diff --git a/packages/app/src/components/Root/LogoFull.tsx b/packages/app/src/components/Root/LogoFull.tsx
index 2fb7674..c7b1c84 100644
--- a/packages/app/src/components/Root/LogoFull.tsx
+++ b/packages/app/src/components/Root/LogoFull.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
diff --git a/packages/app/src/components/Root/LogoIcon.tsx b/packages/app/src/components/Root/LogoIcon.tsx
index 507e47d..073cf6e 100644
--- a/packages/app/src/components/Root/LogoIcon.tsx
+++ b/packages/app/src/components/Root/LogoIcon.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index 9c07e0c..d8665a2 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
@@ -25,2 +25,5 @@ import LogoFull from './LogoFull';
 import LogoIcon from './LogoIcon';
+import { NavLink } from 'react-router-dom';
+import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
+import { SidebarSearch } from '@backstage/plugin-search';
 import {
@@ -33,6 +36,4 @@ import {
   SidebarSpace,
-} from '@backstage/core';
-import { NavLink } from 'react-router-dom';
-import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
-import { SidebarSearch } from '@backstage/plugin-search';
+  SidebarScrollWrapper,
+} from '@backstage/core-components';
 
@@ -84,3 +85,5 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       <SidebarDivider />
-      <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+      <SidebarScrollWrapper>
+        <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+      </SidebarScrollWrapper>
       <SidebarSpace />
diff --git a/packages/app/src/components/Root/index.ts b/packages/app/src/components/Root/index.ts
index ab65cb2..dff706f 100644
--- a/packages/app/src/components/Root/index.ts
+++ b/packages/app/src/components/Root/index.ts
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 4f67dda..9a4659c 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
@@ -17,3 +17,2 @@ import React from 'react';
 import { Button, Grid } from '@material-ui/core';
-import { EmptyState } from '@backstage/core';
 import {
@@ -38,4 +37,8 @@ import {
   EntitySwitch,
+  EntityOrphanWarning,
+  EntityProcessingErrorsPanel,
   isComponentType,
   isKind,
+  hasCatalogProcessingErrors,
+  isOrphan,
 } from '@backstage/plugin-catalog';
@@ -52,2 +55,3 @@ import {
 import { EntityTechdocsContent } from '@backstage/plugin-techdocs';
+import { EmptyState } from '@backstage/core-components';
 
@@ -82,2 +86,18 @@ const overviewContent = (
   <Grid container spacing={3} alignItems="stretch">
+    <EntitySwitch>
+      <EntitySwitch.Case if={isOrphan}>
+        <Grid item xs={12}>
+          <EntityOrphanWarning />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+    
+    <EntitySwitch>
+      <EntitySwitch.Case if={hasCatalogProcessingErrors}>
+        <Grid item xs={12}>
+          <EntityProcessingErrorsPanel />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+
     <Grid item md={6}>
@@ -159,2 +179,9 @@ const websiteEntityPage = (
 
+/**
+ * NOTE: This page is designed to work on small screens such as mobile devices.
+ * This is based on Material UI Grid. If breakpoints are used, each grid item must set the `xs` prop to a column size or to `true`,
+ * since this does not default. If no breakpoints are used, the items will equitably share the asvailable space.
+ * https://material-ui.com/components/grid/#basic-grid.
+ */
+
 const defaultEntityPage = (
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index f525be8..7b3c2b2 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -3,3 +3,2 @@ import { makeStyles, Theme, Grid, List, Paper } from '@material-ui/core';
 
-import { Content, Header, Page } from '@backstage/core';
 import { CatalogResultListItem } from '@backstage/plugin-catalog';
@@ -11,2 +10,3 @@ import {
 } from '@backstage/plugin-search';
+import { Content, Header, Page } from '@backstage/core-components';
 
diff --git a/packages/backend/package.json b/packages/backend/package.json
index a1041d9..ddad52b 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -20,14 +20,14 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.8.3",
-    "@backstage/catalog-model": "^0.8.3",
-    "@backstage/catalog-client": "^0.3.13",
+    "@backstage/backend-common": "^0.8.7",
+    "@backstage/catalog-model": "^0.9.0",
+    "@backstage/catalog-client": "^0.3.18",
     "@backstage/config": "^0.1.5",
-    "@backstage/plugin-app-backend": "^0.3.14",
-    "@backstage/plugin-auth-backend": "^0.3.13",
-    "@backstage/plugin-catalog-backend": "^0.10.3",
-    "@backstage/plugin-proxy-backend": "^0.2.10",
-    "@backstage/plugin-scaffolder-backend": "^0.12.1",
-    "@backstage/plugin-search-backend": "^0.2.1",
-    "@backstage/plugin-search-backend-node": "^0.2.1",
-    "@backstage/plugin-techdocs-backend": "^0.8.3",
+    "@backstage/plugin-app-backend": "^0.3.15",
+    "@backstage/plugin-auth-backend": "^0.3.19",
+    "@backstage/plugin-catalog-backend": "^0.13.1",
+    "@backstage/plugin-proxy-backend": "^0.2.11",
+    "@backstage/plugin-scaffolder-backend": "^0.14.1",
+    "@backstage/plugin-search-backend": "^0.2.3",
+    "@backstage/plugin-search-backend-node": "^0.4.0",
+    "@backstage/plugin-techdocs-backend": "^0.9.0",
     "@gitbeaker/node": "^30.2.0",
@@ -38,3 +38,3 @@
     "knex": "^0.21.6",
-    "sqlite3": "^5.0.0",
+    "sqlite3": "^5.0.1",
     "winston": "^3.2.1"
@@ -42,3 +42,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.1",
+    "@backstage/cli": "^0.7.6",
     "@types/dockerode": "^3.2.1",
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 333ffa1..a0201ce 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -6,8 +6,3 @@ import { CatalogClient } from '@backstage/catalog-client';
 import {
-  CookieCutter,
-  CreateReactAppTemplater,
   createRouter,
-  Preparers,
-  Publishers,
-  Templaters,
 } from '@backstage/plugin-scaffolder-backend';
@@ -26,12 +21,2 @@ export default async function createPlugin({
 
-  const cookiecutterTemplater = new CookieCutter({ containerRunner });
-  const craTemplater = new CreateReactAppTemplater({ containerRunner });
-  const templaters = new Templaters();
-
-  templaters.register('cookiecutter', cookiecutterTemplater);
-  templaters.register('cra', craTemplater);
-
-  const preparers = await Preparers.fromConfig(config, { logger });
-  const publishers = await Publishers.fromConfig(config, { logger });
-
   const discovery = SingleHostDiscovery.fromConfig(config);
@@ -40,5 +25,3 @@ export default async function createPlugin({
   return await createRouter({
-    preparers,
-    templaters,
-    publishers,
+    containerRunner,
     logger,
