diff --git a/app-config.yaml b/app-config.yaml
index d5ce05d..8da6f21 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -61,3 +61,3 @@ catalog:
   rules:
-    - allow: [Component, API, Group, User, Template, Location]
+    - allow: [Component, System, API, Group, User, Template, Location]
   locations:
@@ -67,2 +67,6 @@ catalog:
 
+    # Backstage example systems
+    - type: url
+      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-systems.yaml
+
     # Backstage example APIs
@@ -71,2 +75,8 @@ catalog:
 
+    # Backstage example organization groups
+    - type: url
+      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme/org.yaml
+      rules:
+        - allow: [Group, User]
+
     # Backstage example templates
diff --git a/package.json b/package.json
index 1642e12..26c5585 100644
--- a/package.json
+++ b/package.json
@@ -8,2 +8,3 @@
   "scripts": {
+    "dev": "concurrently \"yarn start\" \"yarn start-backend\"",
     "start": "yarn workspace app start",
@@ -30,4 +31,5 @@
   "devDependencies": {
-    "@backstage/cli": "^0.6.2",
+    "@backstage/cli": "^0.6.6",
     "@spotify/prettier-config": "^7.0.0",
+    "concurrently": "^6.0.0",
     "lerna": "^4.0.0",
diff --git a/packages/app/package.json b/packages/app/package.json
index e3717af..6c21964 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,20 +6,21 @@
   "dependencies": {
+    "@backstage/catalog-model": "^0.7.4",
+    "@backstage/cli": "^0.6.6",
+    "@backstage/core": "^0.7.3",
+    "@backstage/integration-react": "^0.1.1",
+    "@backstage/plugin-api-docs": "^0.4.9",
+    "@backstage/plugin-catalog": "^0.5.1",
+    "@backstage/plugin-catalog-import": "^0.5.0",
+    "@backstage/plugin-catalog-react": "^0.1.3",
+    "@backstage/plugin-github-actions": "^0.4.2",
+    "@backstage/plugin-org": "^0.3.10",
+    "@backstage/plugin-scaffolder": "^0.8.0",
+    "@backstage/plugin-search": "^0.3.4",
+    "@backstage/plugin-tech-radar": "^0.3.8",
+    "@backstage/plugin-techdocs": "^0.7.0",
+    "@backstage/plugin-user-settings": "^0.2.8",
+    "@backstage/test-utils": "^0.1.9",
+    "@backstage/theme": "^0.2.5",
     "@material-ui/core": "^4.11.0",
     "@material-ui/icons": "^4.9.1",
-    "@backstage/cli": "^0.6.2",
-    "@backstage/core": "^0.6.3",
-    "@backstage/plugin-api-docs": "^0.4.7",
-    "@backstage/plugin-catalog": "^0.4.0",
-    "@backstage/plugin-catalog-react": "^0.1.0",
-    "@backstage/plugin-catalog-import": "^0.4.2",
-    "@backstage/plugin-scaffolder": "^0.6.0",
-    "@backstage/plugin-techdocs": "^0.5.8",
-    "@backstage/catalog-model": "^0.7.2",
-    "@backstage/plugin-circleci": "^0.2.10",
-    "@backstage/plugin-tech-radar": "^0.3.6",
-    "@backstage/plugin-github-actions": "^0.3.4",
-    "@backstage/plugin-user-settings": "^0.2.6",
-    "@backstage/plugin-search": "^0.3.2",
-    "@backstage/test-utils": "^0.1.8",
-    "@backstage/theme": "^0.2.3",
     "history": "^5.0.0",
@@ -36,3 +37,3 @@
     "@types/jest": "^26.0.7",
-    "@types/node": "^12.0.0",
+    "@types/node": "^14.14.32",
     "@types/react-dom": "^16.9.8",
diff --git a/packages/app/src/App.test.tsx b/packages/app/src/App.test.tsx
index 21a2eaa..82bc479 100644
--- a/packages/app/src/App.test.tsx
+++ b/packages/app/src/App.test.tsx
@@ -6,5 +6,5 @@ describe('App', () => {
   it('should render', async () => {
-    Object.defineProperty(process.env, 'APP_CONFIG', {
-      configurable: true,
-      value: [
+    process.env = {
+      NODE_ENV: 'test',
+      APP_CONFIG: [
         {
@@ -19,4 +19,4 @@ describe('App', () => {
         },
-      ],
-    });
+      ] as any,
+    };
 
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index cf31647..1cd7b77 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -1,26 +1,25 @@
 import React from 'react';
+import { Navigate, Route } from 'react-router';
 import {
-  createApp,
   AlertDisplay,
-  OAuthRequestDialog,
-  SidebarPage,
+  createApp,
   FlatRoutes,
+  OAuthRequestDialog,
 } from '@backstage/core';
-import { apis } from './apis';
-import * as plugins from './plugins';
-import { AppSidebar } from './sidebar';
-import { Route, Navigate } from 'react-router';
+import { apiDocsPlugin, ApiExplorerPage } from '@backstage/plugin-api-docs';
 import {
-  catalogPlugin,
-  CatalogIndexPage,
   CatalogEntityPage,
+  CatalogIndexPage,
+  catalogPlugin,
 } from '@backstage/plugin-catalog';
-import { Router as DocsRouter } from '@backstage/plugin-techdocs';
 import { CatalogImportPage } from '@backstage/plugin-catalog-import';
-import { Router as TechRadarRouter } from '@backstage/plugin-tech-radar';
-import { SearchPage as SearchRouter } from '@backstage/plugin-search';
-import { Router as SettingsRouter } from '@backstage/plugin-user-settings';
-
+import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
+import { SearchPage } from '@backstage/plugin-search';
+import { TechRadarPage } from '@backstage/plugin-tech-radar';
+import { TechdocsPage } from '@backstage/plugin-techdocs';
+import { UserSettingsPage } from '@backstage/plugin-user-settings';
+import { apis } from './apis';
 import { EntityPage } from './components/catalog/EntityPage';
-import { scaffolderPlugin, ScaffolderPage } from '@backstage/plugin-scaffolder';
+import { Root } from './components/Root';
+import * as plugins from './plugins';
 
@@ -33,3 +32,6 @@ const app = createApp({
     });
-  }
+    bind(apiDocsPlugin.externalRoutes, {
+      createComponent: scaffolderPlugin.routes.root,
+    });
+  },
 });
@@ -38,3 +40,25 @@ const AppProvider = app.getProvider();
 const AppRouter = app.getRouter();
-const deprecatedAppRoutes = app.getRoutes();
+
+const routes = (
+  <FlatRoutes>
+    <Navigate key="/" to="/catalog" />
+    <Route path="/catalog" element={<CatalogIndexPage />} />
+    <Route
+      path="/catalog/:namespace/:kind/:name"
+      element={<CatalogEntityPage />}
+    >
+      <EntityPage />
+    </Route>
+    <Route path="/docs" element={<TechdocsPage />} />
+    <Route path="/create" element={<ScaffolderPage />} />
+    <Route path="/api-docs" element={<ApiExplorerPage />} />
+    <Route
+      path="/tech-radar"
+      element={<TechRadarPage width={1500} height={800} />}
+    />
+    <Route path="/catalog-import" element={<CatalogImportPage />} />
+    <Route path="/search" element={<SearchPage />} />
+    <Route path="/settings" element={<UserSettingsPage />} />
+  </FlatRoutes>
+);
 
@@ -45,28 +69,3 @@ const App = () => (
     <AppRouter>
-      <SidebarPage>
-        <AppSidebar />
-        <FlatRoutes>
-          <Navigate key="/" to="/catalog" />
-          <Route path="/catalog" element={<CatalogIndexPage />} />
-          <Route
-            path="/catalog/:namespace/:kind/:name"
-            element={<CatalogEntityPage />}
-          >
-            <EntityPage />
-          </Route>
-          <Route path="/docs" element={<DocsRouter />} />
-          <Route path="/create" element={<ScaffolderPage />} />
-          <Route
-            path="/tech-radar"
-            element={<TechRadarRouter width={1500} height={800} />}
-          />
-          <Route path="/catalog-import" element={<CatalogImportPage />} />
-          <Route
-            path="/search"
-            element={<SearchRouter/>}
-          />
-          <Route path="/settings" element={<SettingsRouter />} />
-          {deprecatedAppRoutes}
-        </FlatRoutes>
-      </SidebarPage>
+      <Root>{routes}</Root>
     </AppRouter>
diff --git a/packages/app/src/apis.ts b/packages/app/src/apis.ts
index 24b22a8..d803416 100644
--- a/packages/app/src/apis.ts
+++ b/packages/app/src/apis.ts
@@ -1 +1,14 @@
-export const apis = [];
+import {
+  AnyApiFactory, configApiRef, createApiFactory
+} from '@backstage/core';
+import {
+  ScmIntegrationsApi, scmIntegrationsApiRef
+} from '@backstage/integration-react';
+
+export const apis: AnyApiFactory[] = [
+  createApiFactory({
+    api: scmIntegrationsApiRef,
+    deps: { configApi: configApiRef },
+    factory: ({ configApi }) => ScmIntegrationsApi.fromConfig(configApi),
+  }),
+];
diff --git a/packages/app/src/LogoFull.tsx b/packages/app/src/components/Root/LogoFull.tsx
similarity index 100%
rename from packages/app/src/LogoFull.tsx
rename to packages/app/src/components/Root/LogoFull.tsx
diff --git a/packages/app/src/LogoIcon.tsx b/packages/app/src/components/Root/LogoIcon.tsx
similarity index 100%
rename from packages/app/src/LogoIcon.tsx
rename to packages/app/src/components/Root/LogoIcon.tsx
diff --git a/packages/app/src/sidebar.tsx b/packages/app/src/components/Root/Root.tsx
similarity index 50%
rename from packages/app/src/sidebar.tsx
rename to packages/app/src/components/Root/Root.tsx
index 9f7973c..9c07e0c 100644
--- a/packages/app/src/sidebar.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -1,24 +1,39 @@
-import React, { useContext } from 'react';
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useContext, PropsWithChildren } from 'react';
+import { Link, makeStyles } from '@material-ui/core';
 import HomeIcon from '@material-ui/icons/Home';
-import LibraryBooks from '@material-ui/icons/LibraryBooks';
 import ExtensionIcon from '@material-ui/icons/Extension';
-import CreateComponentIcon from '@material-ui/icons/AddCircleOutline';
 import MapIcon from '@material-ui/icons/MyLocation';
-import { Link, makeStyles } from '@material-ui/core';
-import { NavLink } from 'react-router-dom';
+import LibraryBooks from '@material-ui/icons/LibraryBooks';
+import CreateComponentIcon from '@material-ui/icons/AddCircleOutline';
 import LogoFull from './LogoFull';
 import LogoIcon from './LogoIcon';
-import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
-
 import {
   Sidebar,
-  SidebarItem,
-  SidebarDivider,
+  SidebarPage,
   sidebarConfig,
   SidebarContext,
+  SidebarItem,
+  SidebarDivider,
   SidebarSpace,
 } from '@backstage/core';
+import { NavLink } from 'react-router-dom';
+import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
 import { SidebarSearch } from '@backstage/plugin-search';
 
-
 const useSidebarLogoStyles = makeStyles({
@@ -56,20 +71,22 @@ const SidebarLogo = () => {
 
-export const AppSidebar = () => (
-  <Sidebar>
-    <SidebarLogo />
-    <SidebarSearch />
-    <SidebarDivider />
-    {/* Global nav, not org-specific */}
-    <SidebarItem icon={HomeIcon} to="./" text="Home" />
-    <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
-    <SidebarItem icon={LibraryBooks} to="/docs" text="Docs" />
-    <SidebarItem icon={CreateComponentIcon} to="create" text="Create..." />
-    <SidebarDivider />
-    <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
-    {/* End global nav */}
-    <SidebarDivider />
-    <SidebarSpace />
-    <SidebarDivider />
-    <SidebarSettings />
-  </Sidebar>
+export const Root = ({ children }: PropsWithChildren<{}>) => (
+  <SidebarPage>
+    <Sidebar>
+      <SidebarLogo />
+      <SidebarSearch />
+      <SidebarDivider />
+      {/* Global nav, not org-specific */}
+      <SidebarItem icon={HomeIcon} to="/catalog" text="Home" />
+      <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
+      <SidebarItem icon={LibraryBooks} to="docs" text="Docs" />
+      <SidebarItem icon={CreateComponentIcon} to="create" text="Create..." />
+      {/* End global nav */}
+      <SidebarDivider />
+      <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+      <SidebarSpace />
+      <SidebarDivider />
+      <SidebarSettings />
+    </Sidebar>
+    {children}
+  </SidebarPage>
 );
diff --git a/packages/app/src/components/Root/index.ts b/packages/app/src/components/Root/index.ts
new file mode 100644
index 0000000..ab65cb2
--- /dev/null
+++ b/packages/app/src/components/Root/index.ts
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export { Root } from './Root';
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index a3a3715..d05f6bf 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -15,26 +15,39 @@
  */
-import { ApiEntity, Entity } from '@backstage/catalog-model';
-import { WarningPanel } from '@backstage/core';
+import React from 'react';
+import { Button, Grid } from '@material-ui/core';
+import {
+  ApiEntity,
+  DomainEntity,
+  Entity,
+  GroupEntity,
+  SystemEntity,
+  UserEntity,
+} from '@backstage/catalog-model';
+import { EmptyState } from '@backstage/core';
 import {
   ApiDefinitionCard,
-  ConsumedApisCard, 
-  ConsumingComponentsCard, 
-  ProvidedApisCard, 
-  ProvidingComponentsCard
+  ConsumedApisCard,
+  ConsumingComponentsCard,
+  EntityHasApisCard,
+  ProvidedApisCard,
+  ProvidingComponentsCard,
 } from '@backstage/plugin-api-docs';
 import {
-  AboutCard, EntityPageLayout,
+  AboutCard,
+  EntityHasComponentsCard,
+  EntityHasSystemsCard,
+  EntityPageLayout,
 } from '@backstage/plugin-catalog';
+import { useEntity } from '@backstage/plugin-catalog-react';
 import {
-  useEntity
-} from '@backstage/plugin-catalog-react';
-import {
-  isPluginApplicableToEntity as isCircleCIAvailable, Router as CircleCIRouter
-} from '@backstage/plugin-circleci';
-import {
-  isPluginApplicableToEntity as isGitHubActionsAvailable, Router as GitHubActionsRouter
+  isPluginApplicableToEntity as isGitHubActionsAvailable,
+  Router as GitHubActionsRouter,
 } from '@backstage/plugin-github-actions';
+import {
+  GroupProfileCard,
+  MembersListCard,
+  OwnershipCard,
+  UserProfileCard,
+} from '@backstage/plugin-org';
 import { EmbeddedDocsRouter as DocsRouter } from '@backstage/plugin-techdocs';
-import { Grid } from '@material-ui/core';
-import React from 'react';
 
@@ -47,10 +60,18 @@ const CICDSwitcher = ({ entity }: { entity: Entity }) => {
       return <GitHubActionsRouter entity={entity} />;
-    case isCircleCIAvailable(entity):
-      return <CircleCIRouter entity={entity} />;
     default:
       return (
-        <WarningPanel title="CI/CD switcher:">
-          No CI/CD is available for this entity. Check corresponding
-          annotations!
-        </WarningPanel>
+        <EmptyState
+          title="No CI/CD available for this entity"
+          missing="info"
+          description="You need to add an annotation to your component if you want to enable CI/CD for it. You can read more about annotations in Backstage by clicking the button below."
+          action={
+            <Button
+              variant="contained"
+              color="primary"
+              href="https://backstage.io/docs/features/software-catalog/well-known-annotations"
+            >
+              Read more
+            </Button>
+          }
+        />
       );
@@ -61,3 +82,3 @@ const OverviewContent = ({ entity }: { entity: Entity }) => (
   <Grid container spacing={3} alignItems="stretch">
-    <Grid item>
+    <Grid item md={6}>
       <AboutCard entity={entity} variant="gridItem" />
@@ -188,2 +209,92 @@ const ApiEntityPage = ({ entity }: { entity: Entity }) => (
 
+const UserOverviewContent = ({ entity }: { entity: UserEntity }) => (
+  <Grid container spacing={3}>
+    <Grid item xs={12} md={6}>
+      <UserProfileCard entity={entity} variant="gridItem" />
+    </Grid>
+    <Grid item xs={12} md={6}>
+      <OwnershipCard entity={entity} variant="gridItem" />
+    </Grid>
+  </Grid>
+);
+
+const UserEntityPage = ({ entity }: { entity: Entity }) => (
+  <EntityPageLayout>
+    <EntityPageLayout.Content
+      path="/*"
+      title="Overview"
+      element={<UserOverviewContent entity={entity as UserEntity} />}
+    />
+  </EntityPageLayout>
+);
+
+const GroupOverviewContent = ({ entity }: { entity: GroupEntity }) => (
+  <Grid container spacing={3}>
+    <Grid item xs={12} md={6}>
+      <GroupProfileCard entity={entity} variant="gridItem" />
+    </Grid>
+    <Grid item xs={12} md={6}>
+      <OwnershipCard entity={entity} variant="gridItem" />
+    </Grid>
+    <Grid item xs={12}>
+      <MembersListCard entity={entity} />
+    </Grid>
+  </Grid>
+);
+
+const GroupEntityPage = ({ entity }: { entity: Entity }) => (
+  <EntityPageLayout>
+    <EntityPageLayout.Content
+      path="/*"
+      title="Overview"
+      element={<GroupOverviewContent entity={entity as GroupEntity} />}
+    />
+  </EntityPageLayout>
+);
+
+const SystemOverviewContent = ({ entity }: { entity: SystemEntity }) => (
+  <Grid container spacing={3} alignItems="stretch">
+    <Grid item md={6}>
+      <AboutCard entity={entity} variant="gridItem" />
+    </Grid>
+    <Grid item md={6}>
+      <EntityHasComponentsCard variant="gridItem" />
+    </Grid>
+    <Grid item md={6}>
+      <EntityHasApisCard variant="gridItem" />
+    </Grid>
+  </Grid>
+);
+
+const SystemEntityPage = ({ entity }: { entity: Entity }) => (
+  <EntityPageLayout>
+    <EntityPageLayout.Content
+      path="/*"
+      title="Overview"
+      element={<SystemOverviewContent entity={entity as SystemEntity} />}
+    />
+  </EntityPageLayout>
+);
+
+const DomainOverviewContent = ({ entity }: { entity: DomainEntity }) => (
+  <Grid container spacing={3} alignItems="stretch">
+    <Grid item md={6}>
+      <AboutCard entity={entity} variant="gridItem" />
+    </Grid>
+    <Grid item md={6}>
+      <EntityHasSystemsCard variant="gridItem" />
+    </Grid>
+  </Grid>
+);
+
+const DomainEntityPage = ({ entity }: { entity: Entity }) => (
+  <EntityPageLayout>
+    <EntityPageLayout.Content
+      path="/*"
+      title="Overview"
+      element={<DomainOverviewContent entity={entity as DomainEntity} />}
+    />
+  </EntityPageLayout>
+);
+
 export const EntityPage = () => {
@@ -191,3 +302,3 @@ export const EntityPage = () => {
 
-  switch (entity?.kind?.toLowerCase()) {
+  switch (entity?.kind?.toLocaleLowerCase('en-US')) {
     case 'component':
@@ -196,2 +307,13 @@ export const EntityPage = () => {
       return <ApiEntityPage entity={entity} />;
+    case 'group':
+      return <GroupEntityPage entity={entity} />;
+    case 'user':
+      return <UserEntityPage entity={entity} />;
+    case 'system':
+      return <SystemEntityPage entity={entity} />;
+    case 'domain':
+      return <DomainEntityPage entity={entity} />;
+    case 'location':
+    case 'resource':
+    case 'template':
     default:
diff --git a/packages/app/src/plugins.ts b/packages/app/src/plugins.ts
index 28b42d5..df53885 100644
--- a/packages/app/src/plugins.ts
+++ b/packages/app/src/plugins.ts
@@ -3,3 +3,2 @@ export { plugin as CatalogPlugin } from '@backstage/plugin-catalog';
 export { plugin as CatalogImport } from '@backstage/plugin-catalog-import';
-export { plugin as Circleci } from '@backstage/plugin-circleci';
 export { plugin as GithubActions } from '@backstage/plugin-github-actions';
diff --git a/packages/backend/package.json b/packages/backend/package.json
index f347059..caee9e9 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -20,12 +20,12 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.5.4",
-    "@backstage/catalog-model": "^0.7.2",
-    "@backstage/catalog-client": "^0.3.6",
-    "@backstage/config": "^0.1.3",
-    "@backstage/plugin-app-backend": "^0.3.8",
-    "@backstage/plugin-auth-backend": "^0.3.2",
-    "@backstage/plugin-catalog-backend": "^0.6.3",
-    "@backstage/plugin-proxy-backend": "^0.2.4",
-    "@backstage/plugin-scaffolder-backend": "^0.8.0",
-    "@backstage/plugin-techdocs-backend": "^0.6.2",
+    "@backstage/backend-common": "^0.6.0",
+    "@backstage/catalog-model": "^0.7.4",
+    "@backstage/catalog-client": "^0.3.8",
+    "@backstage/config": "^0.1.4",
+    "@backstage/plugin-app-backend": "^0.3.10",
+    "@backstage/plugin-auth-backend": "^0.3.6",
+    "@backstage/plugin-catalog-backend": "^0.6.7",
+    "@backstage/plugin-proxy-backend": "^0.2.6",
+    "@backstage/plugin-scaffolder-backend": "^0.9.3",
+    "@backstage/plugin-techdocs-backend": "^0.7.0",
     "@gitbeaker/node": "^28.0.2",
@@ -34,3 +34,3 @@
     "express": "^4.17.1",
-    "express-promise-router": "^3.0.3",
+    "express-promise-router": "^4.1.0",
     "knex": "^0.21.6",
@@ -40,3 +40,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.6.2",
+    "@backstage/cli": "^0.6.6",
     "@types/dockerode": "^3.2.1",
diff --git a/packages/backend/src/plugins/app.ts b/packages/backend/src/plugins/app.ts
index 5131694..07fb04f 100644
--- a/packages/backend/src/plugins/app.ts
+++ b/packages/backend/src/plugins/app.ts
@@ -1,2 +1,3 @@
 import { createRouter } from '@backstage/plugin-app-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
@@ -6,3 +7,3 @@ export default async function createPlugin({
   config,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   return await createRouter({
diff --git a/packages/backend/src/plugins/auth.ts b/packages/backend/src/plugins/auth.ts
index fe19855..5216510 100644
--- a/packages/backend/src/plugins/auth.ts
+++ b/packages/backend/src/plugins/auth.ts
@@ -1,2 +1,3 @@
 import { createRouter } from '@backstage/plugin-auth-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
@@ -8,3 +9,3 @@ export default async function createPlugin({
   discovery,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   return await createRouter({ logger, config, database, discovery });
diff --git a/packages/backend/src/plugins/catalog.ts b/packages/backend/src/plugins/catalog.ts
index 10ac116..556b385 100644
--- a/packages/backend/src/plugins/catalog.ts
+++ b/packages/backend/src/plugins/catalog.ts
@@ -4,7 +4,8 @@ import {
   createRouter,
-  runPeriodically,
+  runPeriodically
 } from '@backstage/plugin-catalog-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
 
-export default async function createPlugin(env: PluginEnvironment) {
+export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
   const builder = new CatalogBuilder(env);
@@ -14,2 +15,3 @@ export default async function createPlugin(env: PluginEnvironment) {
     higherOrderOperation,
+    locationAnalyzer,
   } = await builder.build();
@@ -25,2 +27,3 @@ export default async function createPlugin(env: PluginEnvironment) {
     higherOrderOperation,
+    locationAnalyzer,
     logger: env.logger,
diff --git a/packages/backend/src/plugins/proxy.ts b/packages/backend/src/plugins/proxy.ts
index 388d3fc..506f6d9 100644
--- a/packages/backend/src/plugins/proxy.ts
+++ b/packages/backend/src/plugins/proxy.ts
@@ -1,3 +1,3 @@
-// @ts-ignore
 import { createRouter } from '@backstage/plugin-proxy-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
@@ -8,3 +8,3 @@ export default async function createPlugin({
   discovery,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   return await createRouter({ logger, config, discovery });
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 6f42aaa..a3416c5 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -1,3 +1,6 @@
+import { SingleHostDiscovery } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
 import {
   CookieCutter,
+  CreateReactAppTemplater,
   createRouter,
@@ -5,9 +8,7 @@ import {
   Publishers,
-  CreateReactAppTemplater,
-  Templaters,
+  Templaters
 } from '@backstage/plugin-scaffolder-backend';
-import { SingleHostDiscovery } from '@backstage/backend-common';
-import type { PluginEnvironment } from '../types';
 import Docker from 'dockerode';
-import { CatalogClient } from '@backstage/catalog-client';
+import { Router } from 'express';
+import type { PluginEnvironment } from '../types';
 
@@ -17,3 +18,4 @@ export default async function createPlugin({
   database,
-}: PluginEnvironment) {
+  reader,
+}: PluginEnvironment): Promise<Router> {
   const cookiecutterTemplater = new CookieCutter();
@@ -42,2 +44,3 @@ export default async function createPlugin({
     catalogClient,
+    reader
   });
diff --git a/packages/backend/src/plugins/techdocs.ts b/packages/backend/src/plugins/techdocs.ts
index 5c7ec50..0ba08ce 100644
--- a/packages/backend/src/plugins/techdocs.ts
+++ b/packages/backend/src/plugins/techdocs.ts
@@ -2,8 +2,10 @@ import {
   createRouter,
-  Preparers,
-  Generators,
-  Publisher,
+
+  Generators, Preparers,
+
+  Publisher
 } from '@backstage/plugin-techdocs-backend';
-import { PluginEnvironment } from '../types';
 import Docker from 'dockerode';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
 
@@ -14,3 +16,3 @@ export default async function createPlugin({
   reader,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   // Preparers are responsible for fetching source files for documentation.
