diff --git a/.prettierignore b/.prettierignore
new file mode 100644
index 0000000..dfb0f1c
--- /dev/null
+++ b/.prettierignore
@@ -0,0 +1,4 @@
+dist
+dist-types
+coverage
+.vscode
diff --git a/README.md b/README.md
index 07cf6fd..8c7c437 100644
--- a/README.md
+++ b/README.md
@@ -9,2 +9,2 @@ yarn install
 yarn dev
-```
\ No newline at end of file
+```
diff --git a/app-config.yaml b/app-config.yaml
index 03c75b8..93b0c3f 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -46,4 +46,4 @@ techdocs:
   builder: 'local' # Alternatives - 'external'
-  generators:
-    techdocs: 'docker' # Alternatives - 'local'
+  generator:
+    runIn: 'docker' # Alternatives - 'local'
   publisher:
@@ -86,3 +86,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/react-ssr-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/react-ssr-template/template.yaml
       rules:
@@ -90,3 +90,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/springboot-grpc-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/springboot-grpc-template/template.yaml
       rules:
@@ -98,3 +98,3 @@ catalog:
     - type: url
-      target: https://github.com/backstage/backstage/blob/master/plugins/scaffolder-backend/sample-templates/docs-template/template.yaml
+      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/docs-template/template.yaml
       rules:
diff --git a/catalog-info.yaml b/catalog-info.yaml
index 841a480..adeb66d 100644
--- a/catalog-info.yaml
+++ b/catalog-info.yaml
@@ -7,4 +7,4 @@ metadata:
   # annotations:
-    # github.com/project-slug: backstage/backstage
-    # backstage.io/techdocs-ref: url:https://github.com/backstage/backstage
+  #   github.com/project-slug: backstage/backstage
+  #   backstage.io/techdocs-ref: dir:.
 spec:
diff --git a/package.json b/package.json
index 016576d..c63b504 100644
--- a/package.json
+++ b/package.json
@@ -5,3 +5,3 @@
   "engines": {
-    "node": "12 || 14"
+    "node": "14 || 16"
   },
@@ -17,3 +17,3 @@
     "diff": "lerna run diff --",
-    "test": "lerna run test --since origin/master -- --coverage",
+    "test": "backstage-cli test",
     "test:all": "lerna run test -- --coverage",
@@ -21,9 +21,6 @@
     "lint:all": "lerna run lint --",
-    "create-plugin": "backstage-cli create-plugin --scope internal --no-private",
+    "prettier:check": "prettier --check .",
+    "create-plugin": "backstage-cli create-plugin --scope internal",
     "remove-plugin": "backstage-cli remove-plugin"
   },
-  "resolutions": {
-    "graphql-language-service-interface": "2.8.2",
-    "graphql-language-service-parser": "1.9.0"
-  },
   "workspaces": {
@@ -35,7 +32,7 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.1",
-    "@spotify/prettier-config": "^7.0.0",
+    "@backstage/cli": "^0.8.2",
+    "@spotify/prettier-config": "^11.0.0",
     "concurrently": "^6.0.0",
     "lerna": "^4.0.0",
-    "prettier": "^1.19.1"
+    "prettier": "^2.3.2"
   },
@@ -50,7 +47,2 @@
     ]
-  },
-  "jest": {
-    "transformModules": [
-      "@asyncapi/react-component"
-    ]
   }
diff --git a/packages/app/package.json b/packages/app/package.json
index fe7c01e..b585e49 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,20 +6,22 @@
   "dependencies": {
-    "@backstage/catalog-model": "^0.8.3",
-    "@backstage/cli": "^0.7.1",
-    "@backstage/core": "^0.7.13",
-    "@backstage/integration-react": "^0.1.3",
-    "@backstage/plugin-api-docs": "^0.5.0",
-    "@backstage/plugin-catalog": "^0.6.3",
-    "@backstage/plugin-catalog-import": "^0.5.10",
-    "@backstage/plugin-catalog-react": "^0.2.3",
-    "@backstage/plugin-github-actions": "^0.4.9",
-    "@backstage/plugin-org": "^0.3.14",
-    "@backstage/plugin-scaffolder": "^0.9.8",
-    "@backstage/plugin-search": "^0.4.0",
-    "@backstage/plugin-tech-radar": "^0.4.0",
-    "@backstage/plugin-techdocs": "^0.9.6",
-    "@backstage/plugin-user-settings": "^0.2.11",
-    "@backstage/test-utils": "^0.1.13",
-    "@backstage/theme": "^0.2.8",
-    "@material-ui/core": "^4.11.0",
+    "@backstage/catalog-model": "^0.9.6",
+    "@backstage/cli": "^0.8.2",
+    "@backstage/core-app-api": "^0.1.20",
+    "@backstage/core-components": "^0.7.3",
+    "@backstage/core-plugin-api": "^0.1.13",
+    "@backstage/integration-react": "^0.1.13",
+    "@backstage/plugin-api-docs": "^0.6.13",
+    "@backstage/plugin-catalog": "^0.7.2",
+    "@backstage/plugin-catalog-import": "^0.7.3",
+    "@backstage/plugin-catalog-react": "^0.6.3",
+    "@backstage/plugin-github-actions": "^0.4.23",
+    "@backstage/plugin-org": "^0.3.27",
+    "@backstage/plugin-scaffolder": "^0.11.10",
+    "@backstage/plugin-search": "^0.4.17",
+    "@backstage/plugin-tech-radar": "^0.4.11",
+    "@backstage/plugin-techdocs": "^0.12.5",
+    "@backstage/plugin-user-settings": "^0.3.10",
+    "@backstage/test-utils": "^0.1.21",
+    "@backstage/theme": "^0.2.13",
+    "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
@@ -38,3 +40,3 @@
     "@types/node": "^14.14.32",
-    "@types/react-dom": "^16.9.8",
+    "@types/react-dom": "*",
     "cross-env": "^7.0.0",
diff --git a/packages/app/public/index.html b/packages/app/public/index.html
index 5653173..b93896c 100644
--- a/packages/app/public/index.html
+++ b/packages/app/public/index.html
@@ -67,3 +67,3 @@
   </head>
-  <body style="margin: 0;">
+  <body style="margin: 0">
     <noscript>You need to enable JavaScript to run this app.</noscript>
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index 07d030b..4cd8368 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -2,8 +2,2 @@ import React from 'react';
 import { Navigate, Route } from 'react-router';
-import {
-  AlertDisplay,
-  createApp,
-  FlatRoutes,
-  OAuthRequestDialog,
-} from '@backstage/core';
 import { apiDocsPlugin, ApiExplorerPage } from '@backstage/plugin-api-docs';
@@ -14,10 +8,15 @@ import {
 } from '@backstage/plugin-catalog';
-import {CatalogImportPage, catalogImportPlugin} from '@backstage/plugin-catalog-import';
-import { 
-  ScaffolderPage, 
-  scaffolderPlugin 
-} from '@backstage/plugin-scaffolder';
+import {
+  CatalogImportPage,
+  catalogImportPlugin,
+} from '@backstage/plugin-catalog-import';
+import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
 import { SearchPage } from '@backstage/plugin-search';
 import { TechRadarPage } from '@backstage/plugin-tech-radar';
-import { TechdocsPage } from '@backstage/plugin-techdocs';
+import {
+  DefaultTechDocsHome,
+  TechDocsIndexPage,
+  techdocsPlugin,
+  TechDocsReaderPage,
+} from '@backstage/plugin-techdocs';
 import { UserSettingsPage } from '@backstage/plugin-user-settings';
@@ -25,4 +24,8 @@ import { apis } from './apis';
 import { entityPage } from './components/catalog/EntityPage';
+import { searchPage } from './components/search/SearchPage';
 import { Root } from './components/Root';
 
+import { AlertDisplay, OAuthRequestDialog } from '@backstage/core-components';
+import { createApp, FlatRoutes } from '@backstage/core-app-api';
+
 const app = createApp({
@@ -32,2 +35,3 @@ const app = createApp({
       createComponent: scaffolderPlugin.routes.root,
+      viewTechDoc: techdocsPlugin.routes.docRoot,
     });
@@ -47,3 +51,3 @@ const routes = (
   <FlatRoutes>
-    <Navigate key="/" to="/catalog" />
+    <Navigate key="/" to="catalog" />
     <Route path="/catalog" element={<CatalogIndexPage />} />
@@ -55,3 +59,9 @@ const routes = (
     </Route>
-    <Route path="/docs" element={<TechdocsPage />} />
+    <Route path="/docs" element={<TechDocsIndexPage />}>
+      <DefaultTechDocsHome />
+    </Route>
+    <Route
+      path="/docs/:namespace/:kind/:name/*"
+      element={<TechDocsReaderPage />}
+    />
     <Route path="/create" element={<ScaffolderPage />} />
@@ -63,3 +73,5 @@ const routes = (
     <Route path="/catalog-import" element={<CatalogImportPage />} />
-    <Route path="/search" element={<SearchPage />} />
+    <Route path="/search" element={<SearchPage />}>
+      {searchPage}
+    </Route>
     <Route path="/settings" element={<UserSettingsPage />} />
diff --git a/packages/app/src/apis.ts b/packages/app/src/apis.ts
index d803416..c89753a 100644
--- a/packages/app/src/apis.ts
+++ b/packages/app/src/apis.ts
@@ -1,7 +1,11 @@
 import {
-  AnyApiFactory, configApiRef, createApiFactory
-} from '@backstage/core';
-import {
-  ScmIntegrationsApi, scmIntegrationsApiRef
+  ScmIntegrationsApi,
+  scmIntegrationsApiRef,
+  ScmAuth,
 } from '@backstage/integration-react';
+import {
+  AnyApiFactory,
+  configApiRef,
+  createApiFactory,
+} from '@backstage/core-plugin-api';
 
@@ -13,2 +17,3 @@ export const apis: AnyApiFactory[] = [
   }),
+  ScmAuth.createDefaultApiFactory(),
 ];
diff --git a/packages/app/src/components/Root/LogoFull.tsx b/packages/app/src/components/Root/LogoFull.tsx
index 2fb7674..c7b1c84 100644
--- a/packages/app/src/components/Root/LogoFull.tsx
+++ b/packages/app/src/components/Root/LogoFull.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
diff --git a/packages/app/src/components/Root/LogoIcon.tsx b/packages/app/src/components/Root/LogoIcon.tsx
index 507e47d..073cf6e 100644
--- a/packages/app/src/components/Root/LogoIcon.tsx
+++ b/packages/app/src/components/Root/LogoIcon.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index 9c07e0c..ec59b0b 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
@@ -25,2 +25,5 @@ import LogoFull from './LogoFull';
 import LogoIcon from './LogoIcon';
+import { NavLink } from 'react-router-dom';
+import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
+import { SidebarSearchModal } from '@backstage/plugin-search';
 import {
@@ -33,6 +36,4 @@ import {
   SidebarSpace,
-} from '@backstage/core';
-import { NavLink } from 'react-router-dom';
-import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
-import { SidebarSearch } from '@backstage/plugin-search';
+  SidebarScrollWrapper,
+} from '@backstage/core-components';
 
@@ -75,6 +76,6 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       <SidebarLogo />
-      <SidebarSearch />
+      <SidebarSearchModal />
       <SidebarDivider />
       {/* Global nav, not org-specific */}
-      <SidebarItem icon={HomeIcon} to="/catalog" text="Home" />
+      <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
       <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
@@ -84,3 +85,5 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       <SidebarDivider />
-      <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+      <SidebarScrollWrapper>
+        <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+      </SidebarScrollWrapper>
       <SidebarSpace />
diff --git a/packages/app/src/components/Root/index.ts b/packages/app/src/components/Root/index.ts
index ab65cb2..dff706f 100644
--- a/packages/app/src/components/Root/index.ts
+++ b/packages/app/src/components/Root/index.ts
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 4f67dda..d3b4b78 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -1,3 +1,3 @@
 /*
- * Copyright 2020 Spotify AB
+ * Copyright 2020 The Backstage Authors
  *
@@ -17,3 +17,2 @@ import React from 'react';
 import { Button, Grid } from '@material-ui/core';
-import { EmptyState } from '@backstage/core';
 import {
@@ -38,4 +37,8 @@ import {
   EntitySwitch,
+  EntityOrphanWarning,
+  EntityProcessingErrorsPanel,
   isComponentType,
   isKind,
+  hasCatalogProcessingErrors,
+  isOrphan,
 } from '@backstage/plugin-catalog';
@@ -52,2 +55,3 @@ import {
 import { EntityTechdocsContent } from '@backstage/plugin-techdocs';
+import { EmptyState } from '@backstage/core-components';
 
@@ -80,4 +84,25 @@ const cicdContent = (
 
+const entityWarningContent = (
+  <>
+    <EntitySwitch>
+      <EntitySwitch.Case if={isOrphan}>
+        <Grid item xs={12}>
+          <EntityOrphanWarning />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+
+    <EntitySwitch>
+      <EntitySwitch.Case if={hasCatalogProcessingErrors}>
+        <Grid item xs={12}>
+          <EntityProcessingErrorsPanel />
+        </Grid>
+      </EntitySwitch.Case>
+    </EntitySwitch>
+  </>
+);
+
 const overviewContent = (
   <Grid container spacing={3} alignItems="stretch">
+    {entityWarningContent}
     <Grid item md={6}>
@@ -159,2 +184,9 @@ const websiteEntityPage = (
 
+/**
+ * NOTE: This page is designed to work on small screens such as mobile devices.
+ * This is based on Material UI Grid. If breakpoints are used, each grid item must set the `xs` prop to a column size or to `true`,
+ * since this does not default. If no breakpoints are used, the items will equitably share the available space.
+ * https://material-ui.com/components/grid/#basic-grid.
+ */
+
 const defaultEntityPage = (
@@ -189,2 +221,3 @@ const apiPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item md={6}>
@@ -192,2 +225,5 @@ const apiPage = (
         </Grid>
+        <Grid item md={4} xs={12}>
+          <EntityLinksCard />
+        </Grid>
         <Grid container item md={12}>
@@ -217,2 +253,3 @@ const userPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item xs={12} md={6}>
@@ -232,2 +269,3 @@ const groupPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item xs={12} md={6}>
@@ -250,2 +288,3 @@ const systemPage = (
       <Grid container spacing={3} alignItems="stretch">
+        {entityWarningContent}
         <Grid item md={6}>
@@ -274,2 +313,3 @@ const domainPage = (
       <Grid container spacing={3} alignItems="stretch">
+        {entityWarningContent}
         <Grid item md={6}>
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index f525be8..7b3c2b2 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -3,3 +3,2 @@ import { makeStyles, Theme, Grid, List, Paper } from '@material-ui/core';
 
-import { Content, Header, Page } from '@backstage/core';
 import { CatalogResultListItem } from '@backstage/plugin-catalog';
@@ -11,2 +10,3 @@ import {
 } from '@backstage/plugin-search';
+import { Content, Header, Page } from '@backstage/core-components';
 
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 7166279..0816912 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -6,5 +6,2 @@
   "private": true,
-  "engines": {
-    "node": "12 || 14"
-  },
   "scripts": {
@@ -20,14 +17,14 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.8.3",
-    "@backstage/catalog-model": "^0.8.3",
-    "@backstage/catalog-client": "^0.3.13",
-    "@backstage/config": "^0.1.5",
-    "@backstage/plugin-app-backend": "^0.3.14",
-    "@backstage/plugin-auth-backend": "^0.3.13",
-    "@backstage/plugin-catalog-backend": "^0.10.3",
-    "@backstage/plugin-proxy-backend": "^0.2.10",
-    "@backstage/plugin-scaffolder-backend": "^0.12.2",
-    "@backstage/plugin-search-backend": "^0.2.1",
-    "@backstage/plugin-search-backend-node": "^0.2.1",
-    "@backstage/plugin-techdocs-backend": "^0.8.3",
+    "@backstage/backend-common": "^0.9.9",
+    "@backstage/catalog-model": "^0.9.6",
+    "@backstage/catalog-client": "^0.5.1",
+    "@backstage/config": "^0.1.11",
+    "@backstage/plugin-app-backend": "^0.3.18",
+    "@backstage/plugin-auth-backend": "^0.4.7",
+    "@backstage/plugin-catalog-backend": "^0.17.3",
+    "@backstage/plugin-proxy-backend": "^0.2.13",
+    "@backstage/plugin-scaffolder-backend": "^0.15.12",
+    "@backstage/plugin-search-backend": "^0.2.6",
+    "@backstage/plugin-search-backend-node": "^0.4.3",
+    "@backstage/plugin-techdocs-backend": "^0.10.7",
     "@gitbeaker/node": "^30.2.0",
@@ -38,3 +35,3 @@
     "knex": "^0.21.6",
-    "sqlite3": "^5.0.0",
+    "sqlite3": "^5.0.1",
     "winston": "^3.2.1"
@@ -42,3 +39,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.1",
+    "@backstage/cli": "^0.8.2",
     "@types/dockerode": "^3.2.1",
diff --git a/packages/backend/src/plugins/catalog.ts b/packages/backend/src/plugins/catalog.ts
index a132f0a..876cb6b 100644
--- a/packages/backend/src/plugins/catalog.ts
+++ b/packages/backend/src/plugins/catalog.ts
@@ -1,5 +1,3 @@
-import {
-  CatalogBuilder,
-  createRouter
-} from '@backstage/plugin-catalog-backend';
+import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
+import { ScaffolderEntitiesProcessor } from '@backstage/plugin-scaffolder-backend';
 import { Router } from 'express';
@@ -7,22 +5,10 @@ import { PluginEnvironment } from '../types';
 
-export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
   const builder = await CatalogBuilder.create(env);
-  const {
-    entitiesCatalog,
-    locationsCatalog,
-    locationService,
-    processingEngine,
-    locationAnalyzer,
-  } = await builder.build();
-
+  builder.addProcessor(new ScaffolderEntitiesProcessor());
+  const { processingEngine, router } = await builder.build();
   await processingEngine.start();
-
-  return await createRouter({
-    entitiesCatalog,
-    locationsCatalog,
-    locationService,
-    locationAnalyzer,
-    logger: env.logger,
-    config: env.config,
-  });
+  return router;
 }
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 333ffa1..6be2e97 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -1,14 +1,4 @@
-import {
-  DockerContainerRunner,
-  SingleHostDiscovery,
-} from '@backstage/backend-common';
+import { DockerContainerRunner } from '@backstage/backend-common';
 import { CatalogClient } from '@backstage/catalog-client';
-import {
-  CookieCutter,
-  CreateReactAppTemplater,
-  createRouter,
-  Preparers,
-  Publishers,
-  Templaters,
-} from '@backstage/plugin-scaffolder-backend';
+import { createRouter } from '@backstage/plugin-scaffolder-backend';
 import Docker from 'dockerode';
@@ -22,2 +12,3 @@ export default async function createPlugin({
   reader,
+  discovery,
 }: PluginEnvironment): Promise<Router> {
@@ -25,14 +16,2 @@ export default async function createPlugin({
   const containerRunner = new DockerContainerRunner({ dockerClient });
-
-  const cookiecutterTemplater = new CookieCutter({ containerRunner });
-  const craTemplater = new CreateReactAppTemplater({ containerRunner });
-  const templaters = new Templaters();
-
-  templaters.register('cookiecutter', cookiecutterTemplater);
-  templaters.register('cra', craTemplater);
-
-  const preparers = await Preparers.fromConfig(config, { logger });
-  const publishers = await Publishers.fromConfig(config, { logger });
-
-  const discovery = SingleHostDiscovery.fromConfig(config);
   const catalogClient = new CatalogClient({ discoveryApi: discovery });
@@ -40,5 +19,3 @@ export default async function createPlugin({
   return await createRouter({
-    preparers,
-    templaters,
-    publishers,
+    containerRunner,
     logger,
diff --git a/packages/backend/src/plugins/search.ts b/packages/backend/src/plugins/search.ts
index 248ed37..7fc317d 100644
--- a/packages/backend/src/plugins/search.ts
+++ b/packages/backend/src/plugins/search.ts
@@ -12,2 +12,3 @@ export default async function createPlugin({
   discovery,
+  config,
 }: PluginEnvironment) {
@@ -21,3 +22,3 @@ export default async function createPlugin({
     defaultRefreshIntervalSeconds: 600,
-    collator: new DefaultCatalogCollator({ discovery }),
+    collator: DefaultCatalogCollator.fromConfig(config, { discovery }),
   });
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index e5bfff2..6c78a2a 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -14,3 +14,3 @@ export type PluginEnvironment = {
   config: Config;
-  reader: UrlReader
+  reader: UrlReader;
   discovery: PluginEndpointDiscovery;
