diff --git a/README.md b/README.md
index b2594d1..07cf6fd 100644
--- a/README.md
+++ b/README.md
@@ -3 +3,8 @@
 This is your newly scaffolded Backstage App, Good Luck!
+
+To start the app, run:
+
+```sh
+yarn install
+yarn dev
+```
\ No newline at end of file
diff --git a/app-config.yaml b/app-config.yaml
index 6b97bb0..6c3df07 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -20,2 +20,4 @@ backend:
     connection: ':memory:'
+  cache:
+    store: memory
   # workingDirectory: /tmp # Use this to configure a working directory for the scaffolder, defaults to the OS temp-dir
@@ -48,3 +50,3 @@ techdocs:
 auth:
-  # see https://backstage.io/docs/tutorials/quickstart-app-auth to know more about enabling auth providers
+  # see https://backstage.io/docs/auth/ to learn about auth providers
   providers: {}
@@ -58,3 +60,3 @@ catalog:
   rules:
-    - allow: [Component, System, API, Group, User, Template, Location]
+    - allow: [Component, System, API, Group, User, Resource, Location]
   locations:
@@ -72,2 +74,6 @@ catalog:
 
+    # Backstage example resources
+    - type: url
+      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-resources.yaml
+
     # Backstage example organization groups
@@ -75,4 +81,2 @@ catalog:
       target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme/org.yaml
-      rules:
-        - allow: [Group, User]
 
diff --git a/package.json b/package.json
index 839f8c2..51c6d41 100644
--- a/package.json
+++ b/package.json
@@ -35,3 +35,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.6.10",
+    "@backstage/cli": "^0.6.14",
     "@spotify/prettier-config": "^7.0.0",
diff --git a/packages/app/package.json b/packages/app/package.json
index 079f2bb..854dd4f 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,19 +6,19 @@
   "dependencies": {
-    "@backstage/catalog-model": "^0.7.8",
-    "@backstage/cli": "^0.6.10",
-    "@backstage/core": "^0.7.8",
-    "@backstage/integration-react": "^0.1.1",
-    "@backstage/plugin-api-docs": "^0.4.12",
-    "@backstage/plugin-catalog": "^0.5.6",
-    "@backstage/plugin-catalog-import": "^0.5.5",
-    "@backstage/plugin-catalog-react": "^0.1.5",
-    "@backstage/plugin-github-actions": "^0.4.5",
-    "@backstage/plugin-org": "^0.3.12",
-    "@backstage/plugin-scaffolder": "^0.9.3",
-    "@backstage/plugin-search": "^0.3.5",
-    "@backstage/plugin-tech-radar": "^0.3.10",
-    "@backstage/plugin-techdocs": "^0.9.1",
-    "@backstage/plugin-user-settings": "^0.2.9",
-    "@backstage/test-utils": "^0.1.10",
-    "@backstage/theme": "^0.2.7",
+    "@backstage/catalog-model": "^0.8.1",
+    "@backstage/cli": "^0.6.14",
+    "@backstage/core": "^0.7.12",
+    "@backstage/integration-react": "^0.1.2",
+    "@backstage/plugin-api-docs": "^0.4.15",
+    "@backstage/plugin-catalog": "^0.6.1",
+    "@backstage/plugin-catalog-import": "^0.5.8",
+    "@backstage/plugin-catalog-react": "^0.2.1",
+    "@backstage/plugin-github-actions": "^0.4.8",
+    "@backstage/plugin-org": "^0.3.14",
+    "@backstage/plugin-scaffolder": "^0.9.7",
+    "@backstage/plugin-search": "^0.3.7",
+    "@backstage/plugin-tech-radar": "^0.4.0",
+    "@backstage/plugin-techdocs": "^0.9.5",
+    "@backstage/plugin-user-settings": "^0.2.10",
+    "@backstage/test-utils": "^0.1.13",
+    "@backstage/theme": "^0.2.8",
     "@material-ui/core": "^4.11.0",
@@ -40,3 +40,3 @@
     "cross-env": "^7.0.0",
-    "cypress": "^4.2.0",
+    "cypress": "^7.3.0",
     "eslint-plugin-cypress": "^2.10.3",
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index 026771c..07d030b 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -15,3 +15,6 @@ import {
 import {CatalogImportPage, catalogImportPlugin} from '@backstage/plugin-catalog-import';
-import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
+import { 
+  ScaffolderPage, 
+  scaffolderPlugin 
+} from '@backstage/plugin-scaffolder';
 import { SearchPage } from '@backstage/plugin-search';
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index a302bb6..4f67dda 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -28,6 +28,11 @@ import {
   EntityAboutCard,
+  EntityDependsOnComponentsCard,
+  EntityDependsOnResourcesCard,
   EntitySystemDiagramCard,
   EntityHasComponentsCard,
+  EntityHasResourcesCard,
+  EntityHasSubcomponentsCard,
   EntityHasSystemsCard,
   EntityLayout,
+  EntityLinksCard,
   EntitySwitch,
@@ -80,2 +85,8 @@ const overviewContent = (
     </Grid>
+    <Grid item md={4} xs={12}>
+      <EntityLinksCard />
+    </Grid>
+    <Grid item md={8} xs={12}>
+      <EntityHasSubcomponentsCard variant="gridItem" />
+    </Grid>
   </Grid>
@@ -104,2 +115,13 @@ const serviceEntityPage = (
 
+    <EntityLayout.Route path="/dependencies" title="Dependencies">
+      <Grid container spacing={3} alignItems="stretch">
+        <Grid item md={6}>
+          <EntityDependsOnComponentsCard variant="gridItem" />
+        </Grid>
+        <Grid item md={6}>
+          <EntityDependsOnResourcesCard variant="gridItem" />
+        </Grid>
+      </Grid>
+    </EntityLayout.Route>
+
     <EntityLayout.Route path="/docs" title="Docs">
@@ -120,2 +142,13 @@ const websiteEntityPage = (
 
+    <EntityLayout.Route path="/dependencies" title="Dependencies">
+      <Grid container spacing={3} alignItems="stretch">
+        <Grid item md={6}>
+          <EntityDependsOnComponentsCard variant="gridItem" />
+        </Grid>
+        <Grid item md={6}>
+          <EntityDependsOnResourcesCard variant="gridItem" />
+        </Grid>
+      </Grid>
+    </EntityLayout.Route>
+
     <EntityLayout.Route path="/docs" title="Docs">
@@ -226,2 +259,5 @@ const systemPage = (
         </Grid>
+        <Grid item md={6}>
+          <EntityHasResourcesCard variant="gridItem" />
+        </Grid>
       </Grid>
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 722f9fd..9a53707 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -20,13 +20,13 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.7.0",
-    "@backstage/catalog-model": "^0.7.8",
-    "@backstage/catalog-client": "^0.3.11",
+    "@backstage/backend-common": "^0.8.1",
+    "@backstage/catalog-model": "^0.8.1",
+    "@backstage/catalog-client": "^0.3.12",
     "@backstage/config": "^0.1.5",
-    "@backstage/plugin-app-backend": "^0.3.12",
-    "@backstage/plugin-auth-backend": "^0.3.9",
-    "@backstage/plugin-catalog-backend": "^0.8.2",
-    "@backstage/plugin-proxy-backend": "^0.2.7",
-    "@backstage/plugin-scaffolder-backend": "^0.11.0",
-    "@backstage/plugin-techdocs-backend": "^0.8.0",
-    "@gitbeaker/node": "^28.0.2",
+    "@backstage/plugin-app-backend": "^0.3.13",
+    "@backstage/plugin-auth-backend": "^0.3.12",
+    "@backstage/plugin-catalog-backend": "^0.10.1",
+    "@backstage/plugin-proxy-backend": "^0.2.8",
+    "@backstage/plugin-scaffolder-backend": "^0.11.5",
+    "@backstage/plugin-techdocs-backend": "^0.8.2",
+    "@gitbeaker/node": "^29.2.0",
     "@octokit/rest": "^18.5.3",
@@ -40,3 +40,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.6.10",
+    "@backstage/cli": "^0.6.14",
     "@types/dockerode": "^3.2.1",
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index 80dc623..aebd034 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -15,2 +15,3 @@ import {
   notFoundHandler,
+  CacheManager,
   SingleConnectionDatabaseManager,
@@ -36,2 +37,3 @@ function makeCreateEnv(config: Config) {
   const databaseManager = SingleConnectionDatabaseManager.fromConfig(config);
+  const cacheManager = CacheManager.fromConfig(config);
 
@@ -40,3 +42,4 @@ function makeCreateEnv(config: Config) {
     const database = databaseManager.forPlugin(plugin);
-    return { logger, database, config, reader, discovery };
+    const cache = cacheManager.forPlugin(plugin);
+    return { logger, database, cache, config, reader, discovery };
   };
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index 757a0e5..e5bfff2 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -3,2 +3,3 @@ import { Config } from '@backstage/config';
 import {
+  PluginCacheManager,
   PluginDatabaseManager,
@@ -11,2 +12,3 @@ export type PluginEnvironment = {
   database: PluginDatabaseManager;
+  cache: PluginCacheManager;
   config: Config;
