diff --git a/.prettierignore b/.prettierignore
new file mode 100644
index 0000000..dfb0f1c
--- /dev/null
+++ b/.prettierignore
@@ -0,0 +1,4 @@
+dist
+dist-types
+coverage
+.vscode
diff --git a/README.md b/README.md
index 07cf6fd..8c7c437 100644
--- a/README.md
+++ b/README.md
@@ -9,2 +9,2 @@ yarn install
 yarn dev
-```
\ No newline at end of file
+```
diff --git a/catalog-info.yaml b/catalog-info.yaml
index 123adee..adeb66d 100644
--- a/catalog-info.yaml
+++ b/catalog-info.yaml
@@ -7,4 +7,4 @@ metadata:
   # annotations:
-    # github.com/project-slug: backstage/backstage
-    # backstage.io/techdocs-ref: dir:.
+  #   github.com/project-slug: backstage/backstage
+  #   backstage.io/techdocs-ref: dir:.
 spec:
diff --git a/package.json b/package.json
index 4bea950..7e2239a 100644
--- a/package.json
+++ b/package.json
@@ -5,3 +5,3 @@
   "engines": {
-    "node": "12 || 14"
+    "node": "14 || 16"
   },
@@ -21,2 +21,3 @@
     "lint:all": "lerna run lint --",
+    "prettier:check": "prettier --check .",
     "create-plugin": "backstage-cli create-plugin --scope internal --no-private",
@@ -35,7 +36,7 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.7",
-    "@spotify/prettier-config": "^7.0.0",
+    "@backstage/cli": "^0.7.13",
+    "@spotify/prettier-config": "^11.0.0",
     "concurrently": "^6.0.0",
     "lerna": "^4.0.0",
-    "prettier": "^1.19.1"
+    "prettier": "^2.3.2"
   },
diff --git a/packages/app/package.json b/packages/app/package.json
index b8eebdb..295f9ab 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,21 +6,21 @@
   "dependencies": {
-    "@backstage/catalog-model": "^0.9.0",
-    "@backstage/cli": "^0.7.7",
-    "@backstage/core-app-api": "^0.1.7",
-    "@backstage/core-components": "^0.3.0",
-    "@backstage/core-plugin-api": "^0.1.5",
-    "@backstage/integration-react": "^0.1.6",
-    "@backstage/plugin-api-docs": "^0.6.5",
-    "@backstage/plugin-catalog": "^0.6.10",
-    "@backstage/plugin-catalog-import": "^0.5.16",
-    "@backstage/plugin-catalog-react": "^0.4.1",
-    "@backstage/plugin-github-actions": "^0.4.15",
-    "@backstage/plugin-org": "^0.3.19",
-    "@backstage/plugin-scaffolder": "^0.10.3",
-    "@backstage/plugin-search": "^0.4.6",
-    "@backstage/plugin-tech-radar": "^0.4.4",
-    "@backstage/plugin-techdocs": "^0.10.3",
-    "@backstage/plugin-user-settings": "^0.3.3",
-    "@backstage/test-utils": "^0.1.16",
-    "@backstage/theme": "^0.2.9",
+    "@backstage/catalog-model": "^0.9.3",
+    "@backstage/cli": "^0.7.13",
+    "@backstage/core-app-api": "^0.1.14",
+    "@backstage/core-components": "^0.5.0",
+    "@backstage/core-plugin-api": "^0.1.8",
+    "@backstage/integration-react": "^0.1.10",
+    "@backstage/plugin-api-docs": "^0.6.9",
+    "@backstage/plugin-catalog": "^0.6.16",
+    "@backstage/plugin-catalog-import": "^0.6.0",
+    "@backstage/plugin-catalog-react": "^0.5.0",
+    "@backstage/plugin-github-actions": "^0.4.19",
+    "@backstage/plugin-org": "^0.3.24",
+    "@backstage/plugin-scaffolder": "^0.11.5",
+    "@backstage/plugin-search": "^0.4.12",
+    "@backstage/plugin-tech-radar": "^0.4.8",
+    "@backstage/plugin-techdocs": "^0.11.3",
+    "@backstage/plugin-user-settings": "^0.3.6",
+    "@backstage/test-utils": "^0.1.17",
+    "@backstage/theme": "^0.2.10",
     "@material-ui/core": "^4.12.2",
@@ -40,3 +40,3 @@
     "@types/node": "^14.14.32",
-    "@types/react-dom": "^16.9.8",
+    "@types/react-dom": "*",
     "cross-env": "^7.0.0",
diff --git a/packages/app/public/index.html b/packages/app/public/index.html
index 5653173..b93896c 100644
--- a/packages/app/public/index.html
+++ b/packages/app/public/index.html
@@ -67,3 +67,3 @@
   </head>
-  <body style="margin: 0;">
+  <body style="margin: 0">
     <noscript>You need to enable JavaScript to run this app.</noscript>
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index c513c7e..4cd8368 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -8,10 +8,15 @@ import {
 } from '@backstage/plugin-catalog';
-import {CatalogImportPage, catalogImportPlugin} from '@backstage/plugin-catalog-import';
-import { 
-  ScaffolderPage, 
-  scaffolderPlugin 
-} from '@backstage/plugin-scaffolder';
+import {
+  CatalogImportPage,
+  catalogImportPlugin,
+} from '@backstage/plugin-catalog-import';
+import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
 import { SearchPage } from '@backstage/plugin-search';
 import { TechRadarPage } from '@backstage/plugin-tech-radar';
-import { TechdocsPage } from '@backstage/plugin-techdocs';
+import {
+  DefaultTechDocsHome,
+  TechDocsIndexPage,
+  techdocsPlugin,
+  TechDocsReaderPage,
+} from '@backstage/plugin-techdocs';
 import { UserSettingsPage } from '@backstage/plugin-user-settings';
@@ -19,2 +24,3 @@ import { apis } from './apis';
 import { entityPage } from './components/catalog/EntityPage';
+import { searchPage } from './components/search/SearchPage';
 import { Root } from './components/Root';
@@ -29,2 +35,3 @@ const app = createApp({
       createComponent: scaffolderPlugin.routes.root,
+      viewTechDoc: techdocsPlugin.routes.docRoot,
     });
@@ -44,3 +51,3 @@ const routes = (
   <FlatRoutes>
-    <Navigate key="/" to="/catalog" />
+    <Navigate key="/" to="catalog" />
     <Route path="/catalog" element={<CatalogIndexPage />} />
@@ -52,3 +59,9 @@ const routes = (
     </Route>
-    <Route path="/docs" element={<TechdocsPage />} />
+    <Route path="/docs" element={<TechDocsIndexPage />}>
+      <DefaultTechDocsHome />
+    </Route>
+    <Route
+      path="/docs/:namespace/:kind/:name/*"
+      element={<TechDocsReaderPage />}
+    />
     <Route path="/create" element={<ScaffolderPage />} />
@@ -60,3 +73,5 @@ const routes = (
     <Route path="/catalog-import" element={<CatalogImportPage />} />
-    <Route path="/search" element={<SearchPage />} />
+    <Route path="/search" element={<SearchPage />}>
+      {searchPage}
+    </Route>
     <Route path="/settings" element={<UserSettingsPage />} />
diff --git a/packages/app/src/apis.ts b/packages/app/src/apis.ts
index 080b20f..f2cd272 100644
--- a/packages/app/src/apis.ts
+++ b/packages/app/src/apis.ts
@@ -1,5 +1,10 @@
 import {
-  ScmIntegrationsApi, scmIntegrationsApiRef
+  ScmIntegrationsApi,
+  scmIntegrationsApiRef,
 } from '@backstage/integration-react';
-import { AnyApiFactory, configApiRef, createApiFactory } from '@backstage/core-plugin-api';
+import {
+  AnyApiFactory,
+  configApiRef,
+  createApiFactory,
+} from '@backstage/core-plugin-api';
 
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index d8665a2..198e7ec 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -79,3 +79,3 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       {/* Global nav, not org-specific */}
-      <SidebarItem icon={HomeIcon} to="/catalog" text="Home" />
+      <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
       <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 6b285af..d3b4b78 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -84,4 +84,4 @@ const cicdContent = (
 
-const overviewContent = (
-  <Grid container spacing={3} alignItems="stretch">
+const entityWarningContent = (
+  <>
     <EntitySwitch>
@@ -93,3 +93,3 @@ const overviewContent = (
     </EntitySwitch>
-    
+
     <EntitySwitch>
@@ -101,3 +101,8 @@ const overviewContent = (
     </EntitySwitch>
+  </>
+);
 
+const overviewContent = (
+  <Grid container spacing={3} alignItems="stretch">
+    {entityWarningContent}
     <Grid item md={6}>
@@ -216,2 +221,3 @@ const apiPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item md={6}>
@@ -219,2 +225,5 @@ const apiPage = (
         </Grid>
+        <Grid item md={4} xs={12}>
+          <EntityLinksCard />
+        </Grid>
         <Grid container item md={12}>
@@ -244,2 +253,3 @@ const userPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item xs={12} md={6}>
@@ -259,2 +269,3 @@ const groupPage = (
       <Grid container spacing={3}>
+        {entityWarningContent}
         <Grid item xs={12} md={6}>
@@ -277,2 +288,3 @@ const systemPage = (
       <Grid container spacing={3} alignItems="stretch">
+        {entityWarningContent}
         <Grid item md={6}>
@@ -301,2 +313,3 @@ const domainPage = (
       <Grid container spacing={3} alignItems="stretch">
+        {entityWarningContent}
         <Grid item md={6}>
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 73fb14d..af1a53f 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -6,5 +6,2 @@
   "private": true,
-  "engines": {
-    "node": "12 || 14"
-  },
   "scripts": {
@@ -20,14 +17,14 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.8.8",
-    "@backstage/catalog-model": "^0.9.0",
-    "@backstage/catalog-client": "^0.3.18",
-    "@backstage/config": "^0.1.6",
-    "@backstage/plugin-app-backend": "^0.3.15",
-    "@backstage/plugin-auth-backend": "^0.3.19",
-    "@backstage/plugin-catalog-backend": "^0.13.1",
-    "@backstage/plugin-proxy-backend": "^0.2.11",
-    "@backstage/plugin-scaffolder-backend": "^0.14.2",
-    "@backstage/plugin-search-backend": "^0.2.3",
-    "@backstage/plugin-search-backend-node": "^0.4.1",
-    "@backstage/plugin-techdocs-backend": "^0.9.1",
+    "@backstage/backend-common": "^0.9.4",
+    "@backstage/catalog-model": "^0.9.3",
+    "@backstage/catalog-client": "^0.4.0",
+    "@backstage/config": "^0.1.10",
+    "@backstage/plugin-app-backend": "^0.3.16",
+    "@backstage/plugin-auth-backend": "^0.4.1",
+    "@backstage/plugin-catalog-backend": "^0.14.0",
+    "@backstage/plugin-proxy-backend": "^0.2.12",
+    "@backstage/plugin-scaffolder-backend": "^0.15.6",
+    "@backstage/plugin-search-backend": "^0.2.6",
+    "@backstage/plugin-search-backend-node": "^0.4.2",
+    "@backstage/plugin-techdocs-backend": "^0.10.3",
     "@gitbeaker/node": "^30.2.0",
@@ -42,3 +39,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.7.7",
+    "@backstage/cli": "^0.7.13",
     "@types/dockerode": "^3.2.1",
diff --git a/packages/backend/src/plugins/catalog.ts b/packages/backend/src/plugins/catalog.ts
index a132f0a..d1ded51 100644
--- a/packages/backend/src/plugins/catalog.ts
+++ b/packages/backend/src/plugins/catalog.ts
@@ -1,5 +1,2 @@
-import {
-  CatalogBuilder,
-  createRouter
-} from '@backstage/plugin-catalog-backend';
+import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
 import { Router } from 'express';
@@ -7,22 +4,9 @@ import { PluginEnvironment } from '../types';
 
-export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
+export default async function createPlugin(
+  env: PluginEnvironment,
+): Promise<Router> {
   const builder = await CatalogBuilder.create(env);
-  const {
-    entitiesCatalog,
-    locationsCatalog,
-    locationService,
-    processingEngine,
-    locationAnalyzer,
-  } = await builder.build();
-
+  const { processingEngine, router } = await builder.build();
   await processingEngine.start();
-
-  return await createRouter({
-    entitiesCatalog,
-    locationsCatalog,
-    locationService,
-    locationAnalyzer,
-    logger: env.logger,
-    config: env.config,
-  });
+  return router;
 }
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index a0201ce..6be2e97 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -1,9 +1,4 @@
-import {
-  DockerContainerRunner,
-  SingleHostDiscovery,
-} from '@backstage/backend-common';
+import { DockerContainerRunner } from '@backstage/backend-common';
 import { CatalogClient } from '@backstage/catalog-client';
-import {
-  createRouter,
-} from '@backstage/plugin-scaffolder-backend';
+import { createRouter } from '@backstage/plugin-scaffolder-backend';
 import Docker from 'dockerode';
@@ -17,2 +12,3 @@ export default async function createPlugin({
   reader,
+  discovery,
 }: PluginEnvironment): Promise<Router> {
@@ -20,4 +16,2 @@ export default async function createPlugin({
   const containerRunner = new DockerContainerRunner({ dockerClient });
-
-  const discovery = SingleHostDiscovery.fromConfig(config);
   const catalogClient = new CatalogClient({ discoveryApi: discovery });
diff --git a/packages/backend/src/plugins/search.ts b/packages/backend/src/plugins/search.ts
index 248ed37..7fc317d 100644
--- a/packages/backend/src/plugins/search.ts
+++ b/packages/backend/src/plugins/search.ts
@@ -12,2 +12,3 @@ export default async function createPlugin({
   discovery,
+  config,
 }: PluginEnvironment) {
@@ -21,3 +22,3 @@ export default async function createPlugin({
     defaultRefreshIntervalSeconds: 600,
-    collator: new DefaultCatalogCollator({ discovery }),
+    collator: DefaultCatalogCollator.fromConfig(config, { discovery }),
   });
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index e5bfff2..6c78a2a 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -14,3 +14,3 @@ export type PluginEnvironment = {
   config: Config;
-  reader: UrlReader
+  reader: UrlReader;
   discovery: PluginEndpointDiscovery;
