diff --git a/package.json b/package.json
index 1642e12..377aebb 100644
--- a/package.json
+++ b/package.json
@@ -8,2 +8,3 @@
   "scripts": {
+    "dev": "concurrently \"yarn start\" \"yarn start-backend\"",
     "start": "yarn workspace app start",
@@ -30,4 +31,5 @@
   "devDependencies": {
-    "@backstage/cli": "^0.6.2",
+    "@backstage/cli": "^0.6.4",
     "@spotify/prettier-config": "^7.0.0",
+    "concurrently": "^6.0.0",
     "lerna": "^4.0.0",
diff --git a/packages/app/package.json b/packages/app/package.json
index e3717af..fd8d5e7 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -8,18 +8,17 @@
     "@material-ui/icons": "^4.9.1",
-    "@backstage/cli": "^0.6.2",
-    "@backstage/core": "^0.6.3",
-    "@backstage/plugin-api-docs": "^0.4.7",
-    "@backstage/plugin-catalog": "^0.4.0",
-    "@backstage/plugin-catalog-react": "^0.1.0",
-    "@backstage/plugin-catalog-import": "^0.4.2",
-    "@backstage/plugin-scaffolder": "^0.6.0",
-    "@backstage/plugin-techdocs": "^0.5.8",
-    "@backstage/catalog-model": "^0.7.2",
-    "@backstage/plugin-circleci": "^0.2.10",
-    "@backstage/plugin-tech-radar": "^0.3.6",
-    "@backstage/plugin-github-actions": "^0.3.4",
-    "@backstage/plugin-user-settings": "^0.2.6",
-    "@backstage/plugin-search": "^0.3.2",
+    "@backstage/cli": "^0.6.4",
+    "@backstage/core": "^0.7.1",
+    "@backstage/plugin-api-docs": "^0.4.8",
+    "@backstage/plugin-catalog": "^0.4.2",
+    "@backstage/plugin-catalog-react": "^0.1.1",
+    "@backstage/plugin-catalog-import": "^0.4.3",
+    "@backstage/plugin-scaffolder": "^0.7.1",
+    "@backstage/plugin-techdocs": "^0.6.1",
+    "@backstage/catalog-model": "^0.7.4",
+    "@backstage/plugin-tech-radar": "^0.3.7",
+    "@backstage/plugin-github-actions": "^0.4.0",
+    "@backstage/plugin-user-settings": "^0.2.7",
+    "@backstage/plugin-search": "^0.3.3",
     "@backstage/test-utils": "^0.1.8",
-    "@backstage/theme": "^0.2.3",
+    "@backstage/theme": "^0.2.4",
     "history": "^5.0.0",
@@ -36,3 +35,3 @@
     "@types/jest": "^26.0.7",
-    "@types/node": "^12.0.0",
+    "@types/node": "^14.14.32",
     "@types/react-dom": "^16.9.8",
diff --git a/packages/app/src/App.test.tsx b/packages/app/src/App.test.tsx
index 21a2eaa..82bc479 100644
--- a/packages/app/src/App.test.tsx
+++ b/packages/app/src/App.test.tsx
@@ -6,5 +6,5 @@ describe('App', () => {
   it('should render', async () => {
-    Object.defineProperty(process.env, 'APP_CONFIG', {
-      configurable: true,
-      value: [
+    process.env = {
+      NODE_ENV: 'test',
+      APP_CONFIG: [
         {
@@ -19,4 +19,4 @@ describe('App', () => {
         },
-      ],
-    });
+      ] as any,
+    };
 
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index cf31647..b2c9d10 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -1,26 +1,26 @@
-import React from 'react';
 import {
-  createApp,
   AlertDisplay,
+  createApp,
+  FlatRoutes,
   OAuthRequestDialog,
   SidebarPage,
-  FlatRoutes,
 } from '@backstage/core';
-import { apis } from './apis';
-import * as plugins from './plugins';
-import { AppSidebar } from './sidebar';
-import { Route, Navigate } from 'react-router';
+import { apiDocsPlugin, ApiExplorerPage } from '@backstage/plugin-api-docs';
 import {
-  catalogPlugin,
-  CatalogIndexPage,
   CatalogEntityPage,
+  CatalogIndexPage,
+  catalogPlugin,
 } from '@backstage/plugin-catalog';
-import { Router as DocsRouter } from '@backstage/plugin-techdocs';
 import { CatalogImportPage } from '@backstage/plugin-catalog-import';
-import { Router as TechRadarRouter } from '@backstage/plugin-tech-radar';
-import { SearchPage as SearchRouter } from '@backstage/plugin-search';
-import { Router as SettingsRouter } from '@backstage/plugin-user-settings';
-
+import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
+import { SearchPage } from '@backstage/plugin-search';
+import { TechRadarPage } from '@backstage/plugin-tech-radar';
+import { TechdocsPage } from '@backstage/plugin-techdocs';
+import { UserSettingsPage } from '@backstage/plugin-user-settings';
+import React from 'react';
+import { Navigate, Route } from 'react-router';
+import { apis } from './apis';
 import { EntityPage } from './components/catalog/EntityPage';
-import { scaffolderPlugin, ScaffolderPage } from '@backstage/plugin-scaffolder';
+import * as plugins from './plugins';
+import { AppSidebar } from './sidebar';
 
@@ -33,3 +33,6 @@ const app = createApp({
     });
-  }
+    bind(apiDocsPlugin.externalRoutes, {
+      createComponent: scaffolderPlugin.routes.root,
+    });
+  },
 });
@@ -38,3 +41,2 @@ const AppProvider = app.getProvider();
 const AppRouter = app.getRouter();
-const deprecatedAppRoutes = app.getRoutes();
 
@@ -56,15 +58,12 @@ const App = () => (
           </Route>
-          <Route path="/docs" element={<DocsRouter />} />
+          <Route path="/docs" element={<TechdocsPage />} />
           <Route path="/create" element={<ScaffolderPage />} />
+          <Route path="/api-docs" element={<ApiExplorerPage />} />
           <Route
             path="/tech-radar"
-            element={<TechRadarRouter width={1500} height={800} />}
+            element={<TechRadarPage width={1500} height={800} />}
           />
           <Route path="/catalog-import" element={<CatalogImportPage />} />
-          <Route
-            path="/search"
-            element={<SearchRouter/>}
-          />
-          <Route path="/settings" element={<SettingsRouter />} />
-          {deprecatedAppRoutes}
+          <Route path="/search" element={<SearchPage />} />
+          <Route path="/settings" element={<UserSettingsPage />} />
         </FlatRoutes>
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index a3a3715..de6ab2b 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -19,5 +19,5 @@ import {
   ApiDefinitionCard,
-  ConsumedApisCard, 
-  ConsumingComponentsCard, 
-  ProvidedApisCard, 
+  ConsumedApisCard,
+  ConsumingComponentsCard,
+  ProvidedApisCard,
   ProvidingComponentsCard
@@ -30,5 +30,2 @@ import {
 } from '@backstage/plugin-catalog-react';
-import {
-  isPluginApplicableToEntity as isCircleCIAvailable, Router as CircleCIRouter
-} from '@backstage/plugin-circleci';
 import {
@@ -47,4 +44,2 @@ const CICDSwitcher = ({ entity }: { entity: Entity }) => {
       return <GitHubActionsRouter entity={entity} />;
-    case isCircleCIAvailable(entity):
-      return <CircleCIRouter entity={entity} />;
     default:
diff --git a/packages/app/src/plugins.ts b/packages/app/src/plugins.ts
index 28b42d5..df53885 100644
--- a/packages/app/src/plugins.ts
+++ b/packages/app/src/plugins.ts
@@ -3,3 +3,2 @@ export { plugin as CatalogPlugin } from '@backstage/plugin-catalog';
 export { plugin as CatalogImport } from '@backstage/plugin-catalog-import';
-export { plugin as Circleci } from '@backstage/plugin-circleci';
 export { plugin as GithubActions } from '@backstage/plugin-github-actions';
diff --git a/packages/backend/package.json b/packages/backend/package.json
index f347059..676582c 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -20,12 +20,12 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.5.4",
-    "@backstage/catalog-model": "^0.7.2",
-    "@backstage/catalog-client": "^0.3.6",
+    "@backstage/backend-common": "^0.5.6",
+    "@backstage/catalog-model": "^0.7.4",
+    "@backstage/catalog-client": "^0.3.7",
     "@backstage/config": "^0.1.3",
-    "@backstage/plugin-app-backend": "^0.3.8",
-    "@backstage/plugin-auth-backend": "^0.3.2",
-    "@backstage/plugin-catalog-backend": "^0.6.3",
-    "@backstage/plugin-proxy-backend": "^0.2.4",
-    "@backstage/plugin-scaffolder-backend": "^0.8.0",
-    "@backstage/plugin-techdocs-backend": "^0.6.2",
+    "@backstage/plugin-app-backend": "^0.3.9",
+    "@backstage/plugin-auth-backend": "^0.3.4",
+    "@backstage/plugin-catalog-backend": "^0.6.5",
+    "@backstage/plugin-proxy-backend": "^0.2.5",
+    "@backstage/plugin-scaffolder-backend": "^0.9.1",
+    "@backstage/plugin-techdocs-backend": "^0.6.4",
     "@gitbeaker/node": "^28.0.2",
@@ -40,3 +40,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.6.2",
+    "@backstage/cli": "^0.6.4",
     "@types/dockerode": "^3.2.1",
diff --git a/packages/backend/src/plugins/app.ts b/packages/backend/src/plugins/app.ts
index 5131694..07fb04f 100644
--- a/packages/backend/src/plugins/app.ts
+++ b/packages/backend/src/plugins/app.ts
@@ -1,2 +1,3 @@
 import { createRouter } from '@backstage/plugin-app-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
@@ -6,3 +7,3 @@ export default async function createPlugin({
   config,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   return await createRouter({
diff --git a/packages/backend/src/plugins/auth.ts b/packages/backend/src/plugins/auth.ts
index fe19855..5216510 100644
--- a/packages/backend/src/plugins/auth.ts
+++ b/packages/backend/src/plugins/auth.ts
@@ -1,2 +1,3 @@
 import { createRouter } from '@backstage/plugin-auth-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
@@ -8,3 +9,3 @@ export default async function createPlugin({
   discovery,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   return await createRouter({ logger, config, database, discovery });
diff --git a/packages/backend/src/plugins/catalog.ts b/packages/backend/src/plugins/catalog.ts
index 10ac116..883b70c 100644
--- a/packages/backend/src/plugins/catalog.ts
+++ b/packages/backend/src/plugins/catalog.ts
@@ -4,7 +4,8 @@ import {
   createRouter,
-  runPeriodically,
+  runPeriodically
 } from '@backstage/plugin-catalog-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
 
-export default async function createPlugin(env: PluginEnvironment) {
+export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
   const builder = new CatalogBuilder(env);
diff --git a/packages/backend/src/plugins/proxy.ts b/packages/backend/src/plugins/proxy.ts
index 388d3fc..506f6d9 100644
--- a/packages/backend/src/plugins/proxy.ts
+++ b/packages/backend/src/plugins/proxy.ts
@@ -1,3 +1,3 @@
-// @ts-ignore
 import { createRouter } from '@backstage/plugin-proxy-backend';
+import { Router } from 'express';
 import { PluginEnvironment } from '../types';
@@ -8,3 +8,3 @@ export default async function createPlugin({
   discovery,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   return await createRouter({ logger, config, discovery });
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 6f42aaa..a3416c5 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -1,3 +1,6 @@
+import { SingleHostDiscovery } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
 import {
   CookieCutter,
+  CreateReactAppTemplater,
   createRouter,
@@ -5,9 +8,7 @@ import {
   Publishers,
-  CreateReactAppTemplater,
-  Templaters,
+  Templaters
 } from '@backstage/plugin-scaffolder-backend';
-import { SingleHostDiscovery } from '@backstage/backend-common';
-import type { PluginEnvironment } from '../types';
 import Docker from 'dockerode';
-import { CatalogClient } from '@backstage/catalog-client';
+import { Router } from 'express';
+import type { PluginEnvironment } from '../types';
 
@@ -17,3 +18,4 @@ export default async function createPlugin({
   database,
-}: PluginEnvironment) {
+  reader,
+}: PluginEnvironment): Promise<Router> {
   const cookiecutterTemplater = new CookieCutter();
@@ -42,2 +44,3 @@ export default async function createPlugin({
     catalogClient,
+    reader
   });
diff --git a/packages/backend/src/plugins/techdocs.ts b/packages/backend/src/plugins/techdocs.ts
index 5c7ec50..0ba08ce 100644
--- a/packages/backend/src/plugins/techdocs.ts
+++ b/packages/backend/src/plugins/techdocs.ts
@@ -2,8 +2,10 @@ import {
   createRouter,
-  Preparers,
-  Generators,
-  Publisher,
+
+  Generators, Preparers,
+
+  Publisher
 } from '@backstage/plugin-techdocs-backend';
-import { PluginEnvironment } from '../types';
 import Docker from 'dockerode';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
 
@@ -14,3 +16,3 @@ export default async function createPlugin({
   reader,
-}: PluginEnvironment) {
+}: PluginEnvironment): Promise<Router> {
   // Preparers are responsible for fetching source files for documentation.
