diff --git a/backstage.json b/backstage.json
index 5a69ad5..e27bbc7 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "0.4.13"
+  "version": "0.4.15"
 }
diff --git a/package.json b/package.json
index 19a4f52..d6ac827 100644
--- a/package.json
+++ b/package.json
@@ -32,3 +32,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.12.0-next.0",
+    "@backstage/cli": "^0.13.0",
     "@spotify/prettier-config": "^12.0.0",
diff --git a/packages/app/package.json b/packages/app/package.json
index 72c01d5..1681062 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,20 +6,21 @@
   "dependencies": {
-    "@backstage/app-defaults": "^0.1.5-next.0",
-    "@backstage/catalog-model": "^0.9.10-next.0",
-    "@backstage/cli": "^0.12.0-next.0",
-    "@backstage/core-app-api": "^0.5.0-next.0",
-    "@backstage/core-components": "^0.8.5-next.0",
-    "@backstage/core-plugin-api": "^0.6.0-next.0",
-    "@backstage/integration-react": "^0.1.19-next.0",
-    "@backstage/plugin-api-docs": "^0.6.23-next.0",
-    "@backstage/plugin-catalog": "^0.7.9-next.0",
-    "@backstage/plugin-catalog-import": "^0.7.10-next.0",
-    "@backstage/plugin-catalog-react": "^0.6.12-next.0",
-    "@backstage/plugin-github-actions": "^0.4.32-next.0",
-    "@backstage/plugin-org": "^0.3.35-next.0",
-    "@backstage/plugin-scaffolder": "^0.11.19-next.0",
-    "@backstage/plugin-search": "^0.5.6-next.0",
-    "@backstage/plugin-tech-radar": "^0.5.3-next.0",
-    "@backstage/plugin-techdocs": "^0.12.15-next.0",
-    "@backstage/plugin-user-settings": "^0.3.17-next.0",
+    "@backstage/app-defaults": "^0.1.5",
+    "@backstage/catalog-model": "^0.9.10",
+    "@backstage/cli": "^0.13.0",
+    "@backstage/core-app-api": "^0.5.1",
+    "@backstage/core-components": "^0.8.6",
+    "@backstage/core-plugin-api": "^0.6.0",
+    "@backstage/integration-react": "^0.1.19",
+    "@backstage/plugin-api-docs": "^0.7.0",
+    "@backstage/plugin-catalog": "^0.7.10",
+    "@backstage/plugin-catalog-graph": "^0.2.8",
+    "@backstage/plugin-catalog-import": "^0.7.10",
+    "@backstage/plugin-catalog-react": "^0.6.12",
+    "@backstage/plugin-github-actions": "^0.4.33",
+    "@backstage/plugin-org": "^0.4.0",
+    "@backstage/plugin-scaffolder": "^0.12.0",
+    "@backstage/plugin-search": "^0.6.0",
+    "@backstage/plugin-tech-radar": "^0.5.3",
+    "@backstage/plugin-techdocs": "^0.13.1",
+    "@backstage/plugin-user-settings": "^0.3.17",
     "@backstage/theme": "^0.2.14",
@@ -28,4 +29,4 @@
     "history": "^5.0.0",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
     "react-router": "6.0.0-beta.0",
@@ -35,3 +36,3 @@
   "devDependencies": {
-    "@backstage/test-utils": "^0.2.3-next.0",
+    "@backstage/test-utils": "^0.2.3",
     "@testing-library/jest-dom": "^5.10.1",
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index 8a53583..78949b0 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -13,2 +13,3 @@ import {
 import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
+import { orgPlugin } from '@backstage/plugin-org';
 import { SearchPage } from '@backstage/plugin-search';
@@ -39,3 +40,3 @@ const app = createApp({
     bind(apiDocsPlugin.externalRoutes, {
-      createComponent: scaffolderPlugin.routes.root,
+      registerApi: catalogImportPlugin.routes.importPage,
     });
@@ -44,2 +45,5 @@ const app = createApp({
     });
+    bind(orgPlugin.externalRoutes, {
+      catalogIndex: catalogPlugin.routes.catalogIndex,
+    });
   },
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index d3b4b78..84d0944 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -29,3 +29,2 @@ import {
   EntityDependsOnResourcesCard,
-  EntitySystemDiagramCard,
   EntityHasComponentsCard,
@@ -56,2 +55,16 @@ import { EntityTechdocsContent } from '@backstage/plugin-techdocs';
 import { EmptyState } from '@backstage/core-components';
+import {
+  Direction,
+  EntityCatalogGraphCard,
+} from '@backstage/plugin-catalog-graph';
+import {
+  RELATION_API_CONSUMED_BY,
+  RELATION_API_PROVIDED_BY,
+  RELATION_CONSUMES_API,
+  RELATION_DEPENDENCY_OF,
+  RELATION_DEPENDS_ON,
+  RELATION_HAS_PART,
+  RELATION_PART_OF,
+  RELATION_PROVIDES_API,
+} from '@backstage/catalog-model';
 
@@ -110,2 +123,6 @@ const overviewContent = (
     </Grid>
+    <Grid item md={6} xs={12}>
+      <EntityCatalogGraphCard variant="gridItem" height={400} />
+    </Grid>
+
     <Grid item md={4} xs={12}>
@@ -225,2 +242,5 @@ const apiPage = (
         </Grid>
+        <Grid item md={6} xs={12}>
+          <EntityCatalogGraphCard variant="gridItem" height={400} />
+        </Grid>
         <Grid item md={4} xs={12}>
@@ -292,2 +312,5 @@ const systemPage = (
         </Grid>
+        <Grid item md={6} xs={12}>
+          <EntityCatalogGraphCard variant="gridItem" height={400} />
+        </Grid>
         <Grid item md={6}>
@@ -304,3 +327,19 @@ const systemPage = (
     <EntityLayout.Route path="/diagram" title="Diagram">
-      <EntitySystemDiagramCard />
+      <EntityCatalogGraphCard
+        variant="gridItem"
+        direction={Direction.TOP_BOTTOM}
+        title="System Diagram"
+        height={700}
+        relations={[
+          RELATION_PART_OF,
+          RELATION_HAS_PART,
+          RELATION_API_CONSUMED_BY,
+          RELATION_API_PROVIDED_BY,
+          RELATION_CONSUMES_API,
+          RELATION_PROVIDES_API,
+          RELATION_DEPENDENCY_OF,
+          RELATION_DEPENDS_ON,
+        ]}
+        unidirectional={false}
+      />
     </EntityLayout.Route>
@@ -317,2 +356,5 @@ const domainPage = (
         </Grid>
+        <Grid item md={6} xs={12}>
+          <EntityCatalogGraphCard variant="gridItem" height={400} />
+        </Grid>
         <Grid item md={6}>
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index cf380b6..a88e725 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -4,2 +4,6 @@ import { makeStyles, Theme, Grid, List, Paper } from '@material-ui/core';
 import { CatalogResultListItem } from '@backstage/plugin-catalog';
+import {
+  catalogApiRef,
+  CATALOG_FILTER_EXISTS,
+} from '@backstage/plugin-catalog-react';
 import { DocsResultListItem } from '@backstage/plugin-techdocs';
@@ -12,2 +16,3 @@ import {
   DefaultResultListItem,
+  useSearch,
 } from '@backstage/plugin-search';
@@ -20,2 +25,3 @@ import {
 } from '@backstage/core-components';
+import { useApi } from '@backstage/core-plugin-api';
 
@@ -38,2 +44,4 @@ const SearchPage = () => {
   const classes = useStyles();
+  const { types } = useSearch();
+  const catalogApi = useApi(catalogApiRef);
 
@@ -67,4 +75,26 @@ const SearchPage = () => {
             <Paper className={classes.filters}>
+              {types.includes('techdocs') && (
+                <SearchFilter.Select
+                  className={classes.filter}
+                  label="Entity"
+                  name="name"
+                  values={async () => {
+                    // Return a list of entities which are documented.
+                    const { items } = await catalogApi.getEntities({
+                      fields: ['metadata.name'],
+                      filter: {
+                        'metadata.annotations.backstage.io/techdocs-ref':
+                          CATALOG_FILTER_EXISTS,
+                      },
+                    });
+
+                    const names = items.map(entity => entity.metadata.name);
+                    names.sort();
+                    return names;
+                  }}
+                />
+              )}
               <SearchFilter.Select
                 className={classes.filter}
+                label="Kind"
                 name="kind"
@@ -74,2 +104,3 @@ const SearchPage = () => {
                 className={classes.filter}
+                label="Lifecycle"
                 name="lifecycle"
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 63c5c0d..a5ff8d4 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -17,17 +17,17 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.10.4-next.0",
-    "@backstage/backend-tasks": "^0.1.4-next.0",
-    "@backstage/catalog-model": "^0.9.10-next.0",
-    "@backstage/catalog-client": "^0.5.5-next.0",
-    "@backstage/config": "^0.1.13-next.0",
-    "@backstage/plugin-app-backend": "^0.3.22-next.0",
-    "@backstage/plugin-auth-backend": "^0.7.0-next.0",
-    "@backstage/plugin-catalog-backend": "^0.21.0-next.0",
-    "@backstage/plugin-permission-common": "^0.4.0-next.0",
-    "@backstage/plugin-permission-node": "^0.4.0-next.0",
-    "@backstage/plugin-proxy-backend": "^0.2.16-next.0",
-    "@backstage/plugin-scaffolder-backend": "^0.15.21-next.0",
-    "@backstage/plugin-search-backend": "^0.3.1-next.0",
-    "@backstage/plugin-search-backend-node": "^0.4.4",
-    "@backstage/plugin-techdocs-backend": "^0.12.4-next.0",
+    "@backstage/backend-common": "^0.10.5",
+    "@backstage/backend-tasks": "^0.1.4",
+    "@backstage/catalog-model": "^0.9.10",
+    "@backstage/catalog-client": "^0.5.5",
+    "@backstage/config": "^0.1.13",
+    "@backstage/plugin-app-backend": "^0.3.22",
+    "@backstage/plugin-auth-backend": "^0.8.0",
+    "@backstage/plugin-catalog-backend": "^0.21.1",
+    "@backstage/plugin-permission-common": "^0.4.0",
+    "@backstage/plugin-permission-node": "^0.4.1",
+    "@backstage/plugin-proxy-backend": "^0.2.16",
+    "@backstage/plugin-scaffolder-backend": "^0.15.22",
+    "@backstage/plugin-search-backend": "^0.4.0",
+    "@backstage/plugin-search-backend-node": "^0.4.5",
+    "@backstage/plugin-techdocs-backend": "^0.13.1",
     "@gitbeaker/node": "^34.6.0",
@@ -42,3 +42,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.12.0-next.0",
+    "@backstage/cli": "^0.13.0",
     "@types/dockerode": "^3.3.0",
diff --git a/packages/backend/src/plugins/search.ts b/packages/backend/src/plugins/search.ts
index f23b0c7..a0a1cc3 100644
--- a/packages/backend/src/plugins/search.ts
+++ b/packages/backend/src/plugins/search.ts
@@ -12,2 +12,3 @@ export default async function createPlugin({
   logger,
+  permissions,
   discovery,
@@ -51,2 +52,5 @@ export default async function createPlugin({
     engine: indexBuilder.getSearchEngine(),
+    types: indexBuilder.getDocumentTypes(),
+    permissions,
+    config,
     logger,
