diff --git a/backstage.json b/backstage.json
index b565e30..e91ab70 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "0.4.9"
+  "version": "0.4.14"
 }
diff --git a/package.json b/package.json
index 564d0d1..2454755 100644
--- a/package.json
+++ b/package.json
@@ -32,3 +32,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.10.3",
+    "@backstage/cli": "^0.12.0",
     "@spotify/prettier-config": "^12.0.0",
@@ -36,3 +36,4 @@
     "lerna": "^4.0.0",
-    "prettier": "^2.3.2"
+    "prettier": "^2.3.2",
+    "typescript": "~4.5.4"
   },
diff --git a/packages/app/package.json b/packages/app/package.json
index ba91715..8ce600d 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -6,20 +6,20 @@
   "dependencies": {
-    "@backstage/app-defaults": "^0.1.3",
-    "@backstage/catalog-model": "^0.9.8",
-    "@backstage/cli": "^0.10.3",
-    "@backstage/core-app-api": "^0.3.0",
-    "@backstage/core-components": "^0.8.2",
-    "@backstage/core-plugin-api": "^0.4.0",
-    "@backstage/integration-react": "^0.1.16",
-    "@backstage/plugin-api-docs": "^0.6.19",
-    "@backstage/plugin-catalog": "^0.7.5",
-    "@backstage/plugin-catalog-import": "^0.7.6",
-    "@backstage/plugin-catalog-react": "^0.6.8",
-    "@backstage/plugin-github-actions": "^0.4.28",
-    "@backstage/plugin-org": "^0.3.32",
-    "@backstage/plugin-scaffolder": "^0.11.15",
-    "@backstage/plugin-search": "^0.5.2",
-    "@backstage/plugin-tech-radar": "^0.5.0",
-    "@backstage/plugin-techdocs": "^0.12.11",
-    "@backstage/plugin-user-settings": "^0.3.14",
+    "@backstage/app-defaults": "^0.1.5",
+    "@backstage/catalog-model": "^0.9.10",
+    "@backstage/cli": "^0.12.0",
+    "@backstage/core-app-api": "^0.5.0",
+    "@backstage/core-components": "^0.8.5",
+    "@backstage/core-plugin-api": "^0.6.0",
+    "@backstage/integration-react": "^0.1.19",
+    "@backstage/plugin-api-docs": "^0.7.0",
+    "@backstage/plugin-catalog": "^0.7.9",
+    "@backstage/plugin-catalog-import": "^0.7.10",
+    "@backstage/plugin-catalog-react": "^0.6.12",
+    "@backstage/plugin-github-actions": "^0.4.32",
+    "@backstage/plugin-org": "^0.4.0",
+    "@backstage/plugin-scaffolder": "^0.12.0",
+    "@backstage/plugin-search": "^0.5.6",
+    "@backstage/plugin-tech-radar": "^0.5.3",
+    "@backstage/plugin-techdocs": "^0.13.0",
+    "@backstage/plugin-user-settings": "^0.3.17",
     "@backstage/theme": "^0.2.14",
@@ -28,4 +28,4 @@
     "history": "^5.0.0",
-    "react": "^16.13.1",
-    "react-dom": "^16.13.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2",
     "react-router": "6.0.0-beta.0",
@@ -35,3 +35,3 @@
   "devDependencies": {
-    "@backstage/test-utils": "^0.2.0",
+    "@backstage/test-utils": "^0.2.3",
     "@testing-library/jest-dom": "^5.10.1",
diff --git a/packages/app/public/index.html b/packages/app/public/index.html
index 1bd6001..a936c73 100644
--- a/packages/app/public/index.html
+++ b/packages/app/public/index.html
@@ -44,7 +44,2 @@
     />
-    <style>
-      #root {
-        min-height: 100%;
-      }
-    </style>
     <title><%= config.getString('app.title') %></title>
@@ -69,3 +64,3 @@
   </head>
-  <body style="margin: 0">
+  <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index 8a53583..78949b0 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -13,2 +13,3 @@ import {
 import { ScaffolderPage, scaffolderPlugin } from '@backstage/plugin-scaffolder';
+import { orgPlugin } from '@backstage/plugin-org';
 import { SearchPage } from '@backstage/plugin-search';
@@ -39,3 +40,3 @@ const app = createApp({
     bind(apiDocsPlugin.externalRoutes, {
-      createComponent: scaffolderPlugin.routes.root,
+      registerApi: catalogImportPlugin.routes.importPage,
     });
@@ -44,2 +45,5 @@ const app = createApp({
     });
+    bind(orgPlugin.externalRoutes, {
+      catalogIndex: catalogPlugin.routes.catalogIndex,
+    });
   },
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index b4fa04f..7e98c7d 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -26,3 +26,6 @@ import LogoIcon from './LogoIcon';
 import { NavLink } from 'react-router-dom';
-import { Settings as SidebarSettings } from '@backstage/plugin-user-settings';
+import {
+  Settings as SidebarSettings,
+  UserSettingsSignInAvatar,
+} from '@backstage/plugin-user-settings';
 import {
@@ -33,10 +36,13 @@ import {
   Sidebar,
-  SidebarPage,
   sidebarConfig,
   SidebarContext,
-  SidebarItem,
   SidebarDivider,
-  SidebarSpace,
+  SidebarGroup,
+  SidebarItem,
+  SidebarPage,
   SidebarScrollWrapper,
+  SidebarSpace,
 } from '@backstage/core-components';
+import MenuIcon from '@material-ui/icons/Menu';
+import SearchIcon from '@material-ui/icons/Search';
 
@@ -79,19 +85,29 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
       <SidebarLogo />
-      <SearchContextProvider>
-        <SidebarSearchModal />
-      </SearchContextProvider>
+      <SidebarGroup label="Search" icon={<SearchIcon />} to="/search">
+        <SearchContextProvider>
+          <SidebarSearchModal />
+        </SearchContextProvider>{' '}
+      </SidebarGroup>
       <SidebarDivider />
-      {/* Global nav, not org-specific */}
-      <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
-      <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
-      <SidebarItem icon={LibraryBooks} to="docs" text="Docs" />
-      <SidebarItem icon={CreateComponentIcon} to="create" text="Create..." />
-      {/* End global nav */}
-      <SidebarDivider />
-      <SidebarScrollWrapper>
-        <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
-      </SidebarScrollWrapper>
+      <SidebarGroup label="Menu" icon={<MenuIcon />}>
+        {/* Global nav, not org-specific */}
+        <SidebarItem icon={HomeIcon} to="catalog" text="Home" />
+        <SidebarItem icon={ExtensionIcon} to="api-docs" text="APIs" />
+        <SidebarItem icon={LibraryBooks} to="docs" text="Docs" />
+        <SidebarItem icon={CreateComponentIcon} to="create" text="Create..." />
+        {/* End global nav */}
+        <SidebarDivider />
+        <SidebarScrollWrapper>
+          <SidebarItem icon={MapIcon} to="tech-radar" text="Tech Radar" />
+        </SidebarScrollWrapper>
+      </SidebarGroup>
       <SidebarSpace />
       <SidebarDivider />
-      <SidebarSettings />
+      <SidebarGroup
+        label="Settings"
+        icon={<UserSettingsSignInAvatar />}
+        to="/settings"
+      >
+        <SidebarSettings />
+      </SidebarGroup>
     </Sidebar>
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index 95c8c64..cf380b6 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -13,3 +13,9 @@ import {
 } from '@backstage/plugin-search';
-import { Content, Header, Page } from '@backstage/core-components';
+import {
+  CatalogIcon,
+  Content,
+  DocsIcon,
+  Header,
+  Page,
+} from '@backstage/core-components';
 
@@ -21,2 +27,3 @@ const useStyles = makeStyles((theme: Theme) => ({
     padding: theme.spacing(2),
+    marginTop: theme.spacing(2),
   },
@@ -43,8 +50,19 @@ const SearchPage = () => {
           <Grid item xs={3}>
+            <SearchType.Accordion
+              name="Result Type"
+              defaultValue="software-catalog"
+              types={[
+                {
+                  value: 'software-catalog',
+                  name: 'Software Catalog',
+                  icon: <CatalogIcon />,
+                },
+                {
+                  value: 'techdocs',
+                  name: 'Documentation',
+                  icon: <DocsIcon />,
+                },
+              ]}
+            />
             <Paper className={classes.filters}>
-              <SearchType
-                values={['techdocs', 'software-catalog']}
-                name="type"
-                defaultValue="software-catalog"
-              />
               <SearchFilter.Select
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 330a913..5d64224 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -17,14 +17,17 @@
     "app": "0.0.0",
-    "@backstage/backend-common": "^0.10.0",
-    "@backstage/catalog-model": "^0.9.8",
-    "@backstage/catalog-client": "^0.5.3",
-    "@backstage/config": "^0.1.11",
-    "@backstage/plugin-app-backend": "^0.3.20",
-    "@backstage/plugin-auth-backend": "^0.5.2",
-    "@backstage/plugin-catalog-backend": "^0.19.3",
-    "@backstage/plugin-proxy-backend": "^0.2.15",
-    "@backstage/plugin-scaffolder-backend": "^0.15.18",
-    "@backstage/plugin-search-backend": "^0.3.0",
-    "@backstage/plugin-search-backend-node": "^0.4.3",
-    "@backstage/plugin-techdocs-backend": "^0.12.1",
+    "@backstage/backend-common": "^0.10.4",
+    "@backstage/backend-tasks": "^0.1.4",
+    "@backstage/catalog-model": "^0.9.10",
+    "@backstage/catalog-client": "^0.5.5",
+    "@backstage/config": "^0.1.13",
+    "@backstage/plugin-app-backend": "^0.3.22",
+    "@backstage/plugin-auth-backend": "^0.7.0",
+    "@backstage/plugin-catalog-backend": "^0.21.0",
+    "@backstage/plugin-permission-common": "^0.4.0",
+    "@backstage/plugin-permission-node": "^0.4.0",
+    "@backstage/plugin-proxy-backend": "^0.2.16",
+    "@backstage/plugin-scaffolder-backend": "^0.15.21",
+    "@backstage/plugin-search-backend": "^0.3.1",
+    "@backstage/plugin-search-backend-node": "^0.4.4",
+    "@backstage/plugin-techdocs-backend": "^0.13.0",
     "@gitbeaker/node": "^34.6.0",
@@ -39,3 +42,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.10.3",
+    "@backstage/cli": "^0.12.0",
     "@types/dockerode": "^3.3.0",
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index 3f12122..70bc66b 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -21,2 +21,3 @@ import {
 } from '@backstage/backend-common';
+import { TaskScheduler } from '@backstage/backend-tasks';
 import { Config } from '@backstage/config';
@@ -30,2 +31,3 @@ import search from './plugins/search';
 import { PluginEnvironment } from './types';
+import { ServerPermissionClient } from '@backstage/plugin-permission-node';
 
@@ -35,5 +37,2 @@ function makeCreateEnv(config: Config) {
   const discovery = SingleHostDiscovery.fromConfig(config);
-
-  root.info(`Created UrlReader ${reader}`);
-
   const cacheManager = CacheManager.fromConfig(config);
@@ -41,2 +40,9 @@ function makeCreateEnv(config: Config) {
   const tokenManager = ServerTokenManager.noop();
+  const taskScheduler = TaskScheduler.fromConfig(config);
+  const permissions = ServerPermissionClient.fromConfig(config, {
+    discovery,
+    tokenManager,
+  });
+
+  root.info(`Created UrlReader ${reader}`);
 
@@ -46,3 +52,14 @@ function makeCreateEnv(config: Config) {
     const cache = cacheManager.forPlugin(plugin);
-    return { logger, database, cache, config, reader, discovery, tokenManager };
+    const scheduler = taskScheduler.forPlugin(plugin);
+    return {
+      logger,
+      database,
+      cache,
+      config,
+      reader,
+      discovery,
+      tokenManager,
+      scheduler,
+      permissions,
+    };
   };
@@ -72,2 +89,4 @@ async function main() {
   apiRouter.use('/search', await search(searchEnv));
+
+  // Add backends ABOVE this line; this 404 handler is the catch-all fallback
   apiRouter.use(notFoundHandler());
diff --git a/packages/backend/src/plugins/app.ts b/packages/backend/src/plugins/app.ts
index 07fb04f..14e19a1 100644
--- a/packages/backend/src/plugins/app.ts
+++ b/packages/backend/src/plugins/app.ts
@@ -7,2 +7,3 @@ export default async function createPlugin({
   config,
+  database,
 }: PluginEnvironment): Promise<Router> {
@@ -11,2 +12,3 @@ export default async function createPlugin({
     config,
+    database,
     appPackageName: 'app',
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index b1e2e0a..0862b0e 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -9,2 +9,4 @@ import {
 } from '@backstage/backend-common';
+import { PluginTaskScheduler } from '@backstage/backend-tasks';
+import { PermissionAuthorizer } from '@backstage/plugin-permission-common';
 
@@ -18,2 +20,4 @@ export type PluginEnvironment = {
   tokenManager: TokenManager;
+  scheduler: PluginTaskScheduler;
+  permissions: PermissionAuthorizer;
 };
