diff --git a/.dockerignore b/.dockerignore
index 5c3b136..505a7b5 100644
--- a/.dockerignore
+++ b/.dockerignore
@@ -2,4 +2,4 @@
 node_modules
-packages
-!packages/backend/dist
+packages/*/src
+packages/*/node_modules
 plugins
diff --git a/app-config.production.yaml b/app-config.production.yaml
index b9dd00b..6535d96 100644
--- a/app-config.production.yaml
+++ b/app-config.production.yaml
@@ -34 +34,7 @@ backend:
       #     $file: <file-path>/ca/server.crt
+
+catalog:
+  # Overrides the default list locations from app-config.yaml as these contain example data.
+  # See https://backstage.io/docs/features/software-catalog/software-catalog-overview#adding-components-to-the-catalog for more details
+  # on how to get entities into the catalog.
+  locations: []
diff --git a/app-config.yaml b/app-config.yaml
index 2bf366d..cd6998d 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -78,34 +78,26 @@ catalog:
   locations:
-    # Backstage example components
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-components.yaml
+    # Local example data, file locations are relative to the backend process, typically `packages/backend`
+    - type: file
+      target: ../../examples/entities.yaml
 
-    # Backstage example systems
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-systems.yaml
-
-    # Backstage example APIs
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-apis.yaml
-
-    # Backstage example resources
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-resources.yaml
-
-    # Backstage example organization groups
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme/org.yaml
-
-    # Backstage example templates
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/react-ssr-template/template.yaml
-      rules:
-        - allow: [Template]
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/springboot-grpc-template/template.yaml
+    # Local example template
+    - type: file
+      target: ../../examples/template/template.yaml
       rules:
         - allow: [Template]
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/docs-template/template.yaml
+
+    # Local example organizational data
+    - type: file
+      target: ../../examples/org.yaml
       rules:
-        - allow: [Template]
+        - allow: [User, Group]
+
+    ## Uncomment these lines to add more example data
+    # - type: url
+    #   target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all.yaml
+
+    ## Uncomment these lines to add an example org
+    # - type: url
+    #   target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme-corp.yaml
+    #   rules:
+    #     - allow: [User, Group]
diff --git a/backstage.json b/backstage.json
index a988709..d19a958 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "1.2.0-next.1"
+  "version": "1.4.0"
 }
diff --git a/examples/entities.yaml b/examples/entities.yaml
new file mode 100644
index 0000000..447e8b1
--- /dev/null
+++ b/examples/entities.yaml
@@ -0,0 +1,41 @@
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-system
+apiVersion: backstage.io/v1alpha1
+kind: System
+metadata:
+  name: examples
+spec:
+  owner: guests
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-component
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: example-website
+spec:
+  type: website
+  lifecycle: experimental
+  owner: guests
+  system: examples
+  providesApis: [example-grpc-api]
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-api
+apiVersion: backstage.io/v1alpha1
+kind: API
+metadata:
+  name: example-grpc-api
+spec:
+  type: grpc
+  lifecycle: experimental
+  owner: guests
+  system: examples
+  definition: |
+    syntax = "proto3";
+
+    service Exampler {
+      rpc Example (ExampleMessage) returns (ExampleMessage) {};
+    }
+
+    message ExampleMessage {
+      string example = 1;
+    };
diff --git a/examples/org.yaml b/examples/org.yaml
new file mode 100644
index 0000000..a10e81f
--- /dev/null
+++ b/examples/org.yaml
@@ -0,0 +1,17 @@
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-user
+apiVersion: backstage.io/v1alpha1
+kind: User
+metadata:
+  name: guest
+spec:
+  memberOf: [guests]
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-group
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: guests
+spec:
+  type: team
+  children: []
diff --git a/examples/template/content/catalog-info.yaml b/examples/template/content/catalog-info.yaml
new file mode 100644
index 0000000..d4ccca4
--- /dev/null
+++ b/examples/template/content/catalog-info.yaml
@@ -0,0 +1,8 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: ${{ values.name | dump }}
+spec:
+  type: service
+  owner: user:guest
+  lifecycle: experimental
diff --git a/examples/template/content/index.js b/examples/template/content/index.js
new file mode 100644
index 0000000..071ce5a
--- /dev/null
+++ b/examples/template/content/index.js
@@ -0,0 +1 @@
+console.log('Hello from ${{ values.name }}!');
diff --git a/examples/template/content/package.json b/examples/template/content/package.json
new file mode 100644
index 0000000..86f968a
--- /dev/null
+++ b/examples/template/content/package.json
@@ -0,0 +1,5 @@
+{
+  "name": "${{ values.name }}",
+  "private": true,
+  "dependencies": {}
+}
diff --git a/examples/template/template.yaml b/examples/template/template.yaml
new file mode 100644
index 0000000..50052b7
--- /dev/null
+++ b/examples/template/template.yaml
@@ -0,0 +1,74 @@
+apiVersion: scaffolder.backstage.io/v1beta3
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-template
+kind: Template
+metadata:
+  name: example-nodejs-template
+  title: Example Node.js Template
+  description: An example template for the scaffolder that creates a simple Node.js service
+spec:
+  owner: user:guest
+  type: service
+
+  # These parameters are used to generate the input form in the frontend, and are
+  # used to gather input data for the execution of the template.
+  parameters:
+    - title: Fill in some steps
+      required:
+        - name
+      properties:
+        name:
+          title: Name
+          type: string
+          description: Unique name of the component
+          ui:autofocus: true
+          ui:options:
+            rows: 5
+    - title: Choose a location
+      required:
+        - repoUrl
+      properties:
+        repoUrl:
+          title: Repository Location
+          type: string
+          ui:field: RepoUrlPicker
+          ui:options:
+            allowedHosts:
+              - github.com
+
+  # These steps are executed in the scaffolder backend, using data that we gathered
+  # via the parameters above.
+  steps:
+    # Each step executes an action, in this case one templates files into the working directory.
+    - id: fetch-base
+      name: Fetch Base
+      action: fetch:template
+      input:
+        url: ./content
+        values:
+          name: ${{ parameters.name }}
+
+    # This step publishes the contents of the working directory to GitHub.
+    - id: publish
+      name: Publish
+      action: publish:github
+      input:
+        allowedHosts: ['github.com']
+        description: This is ${{ parameters.name }}
+        repoUrl: ${{ parameters.repoUrl }}
+
+    # The final step is to register our new component in the catalog.
+    - id: register
+      name: Register
+      action: catalog:register
+      input:
+        repoContentsUrl: ${{ steps.publish.output.repoContentsUrl }}
+        catalogInfoPath: '/catalog-info.yaml'
+
+  # Outputs are displayed to the user after a successful execution of the template.
+  output:
+    links:
+      - title: Repository
+        url: ${{ steps.publish.output.remoteUrl }}
+      - title: Open in catalog
+        icon: catalog
+        entityRef: ${{ steps.register.output.entityRef }}
diff --git a/package.json b/package.json
index 9e94ba6..03b8e70 100644
--- a/package.json
+++ b/package.json
@@ -32,3 +32,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.17.1-next.1",
+    "@backstage/cli": "^0.18.0",
     "@spotify/prettier-config": "^12.0.0",
@@ -37,3 +37,4 @@
     "prettier": "^2.3.2",
-    "typescript": "~4.6.4"
+    "typescript": "~4.6.4",
+    "node-gyp": "^9.0.0"
   },
diff --git a/packages/app/cypress.json b/packages/app/cypress.json
index 5de7ebf..0cb845a 100644
--- a/packages/app/cypress.json
+++ b/packages/app/cypress.json
@@ -3,3 +3,4 @@
   "fixturesFolder": false,
-  "pluginsFile": false
+  "pluginsFile": false,
+  "retries": 3
 }
diff --git a/packages/app/package.json b/packages/app/package.json
index cdf14ca..d88156a 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -9,25 +9,27 @@
   "dependencies": {
-    "@backstage/app-defaults": "^1.0.2-next.0",
-    "@backstage/catalog-model": "^1.0.1",
-    "@backstage/cli": "^0.17.1-next.1",
-    "@backstage/core-app-api": "^1.0.2-next.0",
-    "@backstage/core-components": "^0.9.4-next.0",
-    "@backstage/core-plugin-api": "^1.0.2-next.0",
-    "@backstage/integration-react": "^1.1.0-next.1",
-    "@backstage/plugin-api-docs": "^0.8.5-next.1",
-    "@backstage/plugin-catalog": "^1.2.0-next.1",
-    "@backstage/plugin-catalog-common": "^1.0.1",
-    "@backstage/plugin-catalog-graph": "^0.2.17-next.1",
-    "@backstage/plugin-catalog-import": "^0.8.8-next.1",
-    "@backstage/plugin-catalog-react": "^1.1.0-next.1",
-    "@backstage/plugin-github-actions": "^0.5.5-next.1",
-    "@backstage/plugin-org": "^0.5.5-next.1",
-    "@backstage/plugin-permission-react": "^0.4.1-next.0",
-    "@backstage/plugin-scaffolder": "^1.2.0-next.1",
-    "@backstage/plugin-search": "^0.8.1-next.1",
-    "@backstage/plugin-search-react": "^0.2.0-next.1",
-    "@backstage/plugin-tech-radar": "^0.5.12-next.0",
-    "@backstage/plugin-techdocs": "^1.1.1-next.1",
-    "@backstage/plugin-user-settings": "^0.4.4-next.0",
-    "@backstage/theme": "^0.2.15",
+    "@backstage/app-defaults": "^1.0.4",
+    "@backstage/catalog-model": "^1.1.0",
+    "@backstage/cli": "^0.18.0",
+    "@backstage/core-app-api": "^1.0.4",
+    "@backstage/core-components": "^0.10.0",
+    "@backstage/core-plugin-api": "^1.0.4",
+    "@backstage/integration-react": "^1.1.2",
+    "@backstage/plugin-api-docs": "^0.8.7",
+    "@backstage/plugin-catalog": "^1.4.0",
+    "@backstage/plugin-catalog-common": "^1.0.4",
+    "@backstage/plugin-catalog-graph": "^0.2.19",
+    "@backstage/plugin-catalog-import": "^0.8.10",
+    "@backstage/plugin-catalog-react": "^1.1.2",
+    "@backstage/plugin-github-actions": "^0.5.7",
+    "@backstage/plugin-org": "^0.5.7",
+    "@backstage/plugin-permission-react": "^0.4.3",
+    "@backstage/plugin-scaffolder": "^1.4.0",
+    "@backstage/plugin-search": "^1.0.0",
+    "@backstage/plugin-search-react": "^1.0.0",
+    "@backstage/plugin-tech-radar": "^0.5.14",
+    "@backstage/plugin-techdocs": "^1.3.0",
+    "@backstage/plugin-techdocs-react": "^1.0.2",
+    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.2",
+    "@backstage/plugin-user-settings": "^0.4.6",
+    "@backstage/theme": "^0.2.16",
     "@material-ui/core": "^4.12.2",
@@ -42,3 +44,3 @@
   "devDependencies": {
-    "@backstage/test-utils": "^1.1.0-next.1",
+    "@backstage/test-utils": "^1.1.2",
     "@testing-library/jest-dom": "^5.10.1",
@@ -50,3 +52,3 @@
     "cross-env": "^7.0.0",
-    "cypress": "^7.3.0",
+    "cypress": "^9.7.0",
     "eslint-plugin-cypress": "^2.10.3",
@@ -63,3 +65,3 @@
     "cy:dev": "cypress open",
-    "cy:run": "cypress run"
+    "cy:run": "cypress run --browser chrome"
   },
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index f4ff424..c487726 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -21,2 +21,4 @@ import {
 } from '@backstage/plugin-techdocs';
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+import { ReportIssue } from '@backstage/plugin-techdocs-module-addons-contrib';
 import { UserSettingsPage } from '@backstage/plugin-user-settings';
@@ -70,3 +72,7 @@ const routes = (
       element={<TechDocsReaderPage />}
-    />
+    >
+      <TechDocsAddons>
+        <ReportIssue />
+      </TechDocsAddons>
+    </Route>
     <Route path="/create" element={<ScaffolderPage />} />
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index 90738c4..b1164a3 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -16,3 +16,3 @@
 
-import React, { useContext, PropsWithChildren } from 'react';
+import React, { PropsWithChildren } from 'react';
 import { Link, makeStyles } from '@material-ui/core';
@@ -34,3 +34,2 @@ import {
   sidebarConfig,
-  SidebarContext,
   SidebarDivider,
@@ -41,2 +40,3 @@ import {
   SidebarSpace,
+  useSidebarOpenState,
 } from '@backstage/core-components';
@@ -62,3 +62,3 @@ const SidebarLogo = () => {
   const classes = useSidebarLogoStyles();
-  const { isOpen } = useContext(SidebarContext);
+  const { isOpen } = useSidebarOpenState();
 
@@ -71,2 +71,3 @@ const SidebarLogo = () => {
         className={classes.link}
+        aria-label="Home"
       >
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 84d0944..6ec4da0 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -70,2 +70,13 @@ import {
 
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+import { ReportIssue } from '@backstage/plugin-techdocs-module-addons-contrib';
+
+const techdocsContent = (
+  <EntityTechdocsContent>
+    <TechDocsAddons>
+      <ReportIssue />
+    </TechDocsAddons>
+  </EntityTechdocsContent>
+);
+
 const cicdContent = (
@@ -169,3 +180,3 @@ const serviceEntityPage = (
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      {techdocsContent}
     </EntityLayout.Route>
@@ -196,3 +207,3 @@ const websiteEntityPage = (
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      {techdocsContent}
     </EntityLayout.Route>
@@ -215,3 +226,3 @@ const defaultEntityPage = (
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      {techdocsContent}
     </EntityLayout.Route>
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index 0595e4e..928b820 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -10,3 +10,5 @@ import { TechDocsSearchResultListItem } from '@backstage/plugin-techdocs';
 
+import { SearchType } from '@backstage/plugin-search';
 import {
+  DefaultResultListItem,
   SearchBar,
@@ -14,6 +16,4 @@ import {
   SearchResult,
-  SearchType,
-  DefaultResultListItem,
-} from '@backstage/plugin-search';
-import { useSearch } from '@backstage/plugin-search-react';
+  useSearch,
+} from '@backstage/plugin-search-react';
 import {
@@ -114,3 +114,3 @@ const SearchPage = () => {
                 <List>
-                  {results.map(({ type, document }) => {
+                  {results.map(({ type, document, highlight, rank }) => {
                     switch (type) {
@@ -121,2 +121,4 @@ const SearchPage = () => {
                             result={document}
+                            highlight={highlight}
+                            rank={rank}
                           />
@@ -128,2 +130,4 @@ const SearchPage = () => {
                             result={document}
+                            highlight={highlight}
+                            rank={rank}
                           />
@@ -135,2 +139,4 @@ const SearchPage = () => {
                             result={document}
+                            highlight={highlight}
+                            rank={rank}
                           />
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 9c9bfc3..fa1c517 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -19,18 +19,18 @@
     "app": "link:../app",
-    "@backstage/backend-common": "^0.13.3-next.1",
-    "@backstage/backend-tasks": "^0.3.1-next.0",
-    "@backstage/catalog-model": "^1.0.1",
-    "@backstage/catalog-client": "^1.0.1",
-    "@backstage/config": "^1.0.0",
-    "@backstage/plugin-app-backend": "^0.3.32-next.0",
-    "@backstage/plugin-auth-backend": "^0.13.1-next.1",
-    "@backstage/plugin-catalog-backend": "^1.1.2-next.1",
-    "@backstage/plugin-permission-common": "^0.6.0",
-    "@backstage/plugin-permission-node": "^0.6.1-next.0",
-    "@backstage/plugin-proxy-backend": "^0.2.26-next.0",
-    "@backstage/plugin-scaffolder-backend": "^1.2.0-next.0",
-    "@backstage/plugin-search-backend": "^0.5.2-next.0",
-    "@backstage/plugin-search-backend-module-pg": "^0.3.3-next.0",
-    "@backstage/plugin-search-backend-node": "^0.6.1-next.0",
-    "@backstage/plugin-techdocs-backend": "^1.1.1-next.0",
+    "@backstage/backend-common": "^0.14.1",
+    "@backstage/backend-tasks": "^0.3.3",
+    "@backstage/catalog-model": "^1.1.0",
+    "@backstage/catalog-client": "^1.0.4",
+    "@backstage/config": "^1.0.1",
+    "@backstage/plugin-app-backend": "^0.3.34",
+    "@backstage/plugin-auth-backend": "^0.15.0",
+    "@backstage/plugin-catalog-backend": "^1.3.0",
+    "@backstage/plugin-permission-common": "^0.6.3",
+    "@backstage/plugin-permission-node": "^0.6.3",
+    "@backstage/plugin-proxy-backend": "^0.2.28",
+    "@backstage/plugin-scaffolder-backend": "^1.4.0",
+    "@backstage/plugin-search-backend": "^1.0.0",
+    "@backstage/plugin-search-backend-module-pg": "^0.3.5",
+    "@backstage/plugin-search-backend-node": "^1.0.0",
+    "@backstage/plugin-techdocs-backend": "^1.2.0",
     "dockerode": "^3.3.1",
@@ -38,3 +38,2 @@
     "express-promise-router": "^4.1.0",
-    "luxon": "^2.0.2",
     "pg": "^8.3.0",
@@ -43,3 +42,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.17.1-next.1",
+    "@backstage/cli": "^0.18.0",
     "@types/dockerode": "^3.3.0",
diff --git a/packages/backend/src/plugins/auth.ts b/packages/backend/src/plugins/auth.ts
index 13f18c5..159116d 100644
--- a/packages/backend/src/plugins/auth.ts
+++ b/packages/backend/src/plugins/auth.ts
@@ -20,8 +20,17 @@ export default async function createPlugin(
 
-      // This overrides the default GitHub auth provider with a custom one.
-      // Since the options are empty it will behave just like the default
-      // provider, but if you uncomment the `signIn` section you will enable
-      // sign-in via GitHub. This particular configuration uses a resolver
-      // that matches the username to the user entity name. See the auth
-      // documentation for more details on how to enable and customize sign-in:
+      // This replaces the default GitHub auth provider with a customized one.
+      // The `signIn` option enables sign-in for this provider, using the
+      // identity resolution logic that's provided in the `resolver` callback.
+      //
+      // This particular resolver makes all users share a single "guest" identity.
+      // It should only be used for testing and trying out Backstage.
+      //
+      // If you want to use a production ready resolver you can switch to the
+      // the one that is commented out below, it looks up a user entity in the
+      // catalog using the GitHub username of the authenticated user.
+      // That resolver requires you to have user entities populated in the catalog,
+      // for example using https://backstage.io/docs/integrations/github/org
+      //
+      // There are other resolvers to choose from, and you can also create
+      // your own, see the auth documentation for more details:
       //
@@ -29,5 +38,14 @@ export default async function createPlugin(
       github: providers.github.create({
-        // signIn: {
-        //   resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
-        // },
+        signIn: {
+          resolver(_, ctx) {
+            const userRef = 'user:default/guest'; // Must be a full entity reference
+            return ctx.issueToken({
+              claims: {
+                sub: userRef, // The user's own identity
+                ent: [userRef], // A list of identities that the user claims ownership through
+              },
+            });
+          },
+          // resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
+        },
       }),
diff --git a/packages/backend/src/plugins/search.ts b/packages/backend/src/plugins/search.ts
index 27c42fc..e9469dc 100644
--- a/packages/backend/src/plugins/search.ts
+++ b/packages/backend/src/plugins/search.ts
@@ -10,3 +10,2 @@ import { DefaultTechDocsCollatorFactory } from '@backstage/plugin-techdocs-backe
 import { Router } from 'express';
-import { Duration } from 'luxon';
 
@@ -25,7 +24,7 @@ export default async function createPlugin(
   const schedule = env.scheduler.createScheduledTaskRunner({
-    frequency: Duration.fromObject({ minutes: 10 }),
-    timeout: Duration.fromObject({ minutes: 15 }),
+    frequency: { minutes: 10 },
+    timeout: { minutes: 15 },
     // A 3 second delay gives the backend server a chance to initialize before
     // any collators are executed, which may attempt requests against the API.
-    initialDelay: Duration.fromObject({ seconds: 3 }),
+    initialDelay: { seconds: 3 },
   });
