diff --git a/.gitignore b/.gitignore
index fdc2a5d..d452ac2 100644
--- a/.gitignore
+++ b/.gitignore
@@ -48 +48,4 @@ site
 *-credentials.yaml
+
+# vscode database functionality support files
+*.session.sql
diff --git a/backstage.json b/backstage.json
index 39936e2..6076f7d 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "1.6.0-next.3"
+  "version": "1.7.0"
 }
diff --git a/package.json b/package.json
index 42a1a87..7a27e83 100644
--- a/package.json
+++ b/package.json
@@ -16,4 +16,4 @@
     "clean": "backstage-cli repo clean",
-    "test": "backstage-cli test",
-    "test:all": "lerna run test -- --coverage",
+    "test": "backstage-cli repo test",
+    "test:all": "backstage-cli repo test --coverage",
     "lint": "backstage-cli repo lint --since origin/master",
@@ -31,3 +31,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.19.0-next.3",
+    "@backstage/cli": "^0.20.0",
     "@spotify/prettier-config": "^12.0.0",
diff --git a/packages/app/package.json b/packages/app/package.json
index abda8d9..dd1b831 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -8,27 +8,38 @@
   },
+  "scripts": {
+    "start": "backstage-cli package start",
+    "build": "backstage-cli package build",
+    "clean": "backstage-cli package clean",
+    "test": "backstage-cli package test",
+    "lint": "backstage-cli package lint",
+    "test:e2e": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:dev",
+    "test:e2e:ci": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:run",
+    "cy:dev": "cypress open",
+    "cy:run": "cypress run --browser chrome"
+  },
   "dependencies": {
-    "@backstage/app-defaults": "^1.0.6-next.2",
-    "@backstage/catalog-model": "^1.1.1-next.0",
-    "@backstage/cli": "^0.19.0-next.3",
-    "@backstage/core-app-api": "^1.1.0-next.3",
-    "@backstage/core-components": "^0.11.1-next.3",
-    "@backstage/core-plugin-api": "^1.0.6-next.3",
-    "@backstage/integration-react": "^1.1.4-next.2",
-    "@backstage/plugin-api-docs": "^0.8.9-next.3",
-    "@backstage/plugin-catalog": "^1.5.1-next.3",
-    "@backstage/plugin-catalog-common": "^1.0.6-next.0",
-    "@backstage/plugin-catalog-graph": "^0.2.21-next.2",
-    "@backstage/plugin-catalog-import": "^0.8.12-next.3",
-    "@backstage/plugin-catalog-react": "^1.1.4-next.2",
-    "@backstage/plugin-github-actions": "^0.5.9-next.3",
-    "@backstage/plugin-org": "^0.5.9-next.3",
-    "@backstage/plugin-permission-react": "^0.4.5-next.2",
-    "@backstage/plugin-scaffolder": "^1.6.0-next.3",
-    "@backstage/plugin-search": "^1.0.2-next.3",
-    "@backstage/plugin-search-react": "^1.1.0-next.2",
-    "@backstage/plugin-tech-radar": "^0.5.16-next.3",
-    "@backstage/plugin-techdocs": "^1.3.2-next.3",
-    "@backstage/plugin-techdocs-react": "^1.0.4-next.2",
-    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.4-next.2",
-    "@backstage/plugin-user-settings": "^0.4.8-next.3",
+    "@backstage/app-defaults": "^1.0.7",
+    "@backstage/catalog-model": "^1.1.2",
+    "@backstage/cli": "^0.20.0",
+    "@backstage/core-app-api": "^1.1.1",
+    "@backstage/core-components": "^0.11.2",
+    "@backstage/core-plugin-api": "^1.0.7",
+    "@backstage/integration-react": "^1.1.5",
+    "@backstage/plugin-api-docs": "^0.8.10",
+    "@backstage/plugin-catalog": "^1.6.0",
+    "@backstage/plugin-catalog-common": "^1.0.7",
+    "@backstage/plugin-catalog-graph": "^0.2.22",
+    "@backstage/plugin-catalog-import": "^0.9.0",
+    "@backstage/plugin-catalog-react": "^1.2.0",
+    "@backstage/plugin-github-actions": "^0.5.10",
+    "@backstage/plugin-org": "^0.5.10",
+    "@backstage/plugin-permission-react": "^0.4.6",
+    "@backstage/plugin-scaffolder": "^1.7.0",
+    "@backstage/plugin-search": "^1.0.3",
+    "@backstage/plugin-search-react": "^1.2.0",
+    "@backstage/plugin-tech-radar": "^0.5.17",
+    "@backstage/plugin-techdocs": "^1.3.3",
+    "@backstage/plugin-techdocs-react": "^1.0.5",
+    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.5",
+    "@backstage/plugin-user-settings": "^0.5.0",
     "@backstage/theme": "^0.2.16",
@@ -39,12 +50,12 @@
     "react-dom": "^17.0.2",
-    "react-router": "6.0.0-beta.0",
-    "react-router-dom": "6.0.0-beta.0",
-    "react-use": "^15.3.3"
+    "react-router": "^6.3.0",
+    "react-router-dom": "^6.3.0",
+    "react-use": "^17.2.4"
   },
   "devDependencies": {
-    "@backstage/test-utils": "^1.2.0-next.3",
+    "@backstage/test-utils": "^1.2.1",
     "@testing-library/jest-dom": "^5.10.1",
     "@testing-library/react": "^12.1.3",
-    "@testing-library/user-event": "^12.0.7",
-    "@types/node": "^14.14.32",
+    "@testing-library/user-event": "^14.0.0",
+    "@types/node": "^16.11.26",
     "@types/react-dom": "*",
@@ -55,13 +66,2 @@
   },
-  "scripts": {
-    "start": "backstage-cli package start",
-    "build": "backstage-cli package build",
-    "clean": "backstage-cli package clean",
-    "test": "backstage-cli package test",
-    "lint": "backstage-cli package lint",
-    "test:e2e": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:dev",
-    "test:e2e:ci": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:run",
-    "cy:dev": "cypress open",
-    "cy:run": "cypress run --browser chrome"
-  },
   "browserslist": {
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index c487726..46cb786 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -33,3 +33,3 @@ import { FlatRoutes } from '@backstage/core-app-api';
 import { CatalogGraphPage } from '@backstage/plugin-catalog-graph';
-import { PermissionedRoute } from '@backstage/plugin-permission-react';
+import { RequirePermission } from '@backstage/plugin-permission-react';
 import { catalogEntityCreatePermission } from '@backstage/plugin-catalog-common/alpha';
@@ -60,3 +60,3 @@ const routes = (
   <FlatRoutes>
-    <Navigate key="/" to="catalog" />
+    <Route path="/" element={<Navigate to="catalog" />} />
     <Route path="/catalog" element={<CatalogIndexPage />} />
@@ -83,6 +83,9 @@ const routes = (
     />
-    <PermissionedRoute
+    <Route
       path="/catalog-import"
-      permission={catalogEntityCreatePermission}
-      element={<CatalogImportPage />}
+      element={
+        <RequirePermission permission={catalogEntityCreatePermission}>
+          <CatalogImportPage />
+        </RequirePermission>
+      }
     />
diff --git a/packages/app/src/components/Root/LogoFull.tsx b/packages/app/src/components/Root/LogoFull.tsx
index c7b1c84..47e3b73 100644
--- a/packages/app/src/components/Root/LogoFull.tsx
+++ b/packages/app/src/components/Root/LogoFull.tsx
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 import React from 'react';
diff --git a/packages/app/src/components/Root/LogoIcon.tsx b/packages/app/src/components/Root/LogoIcon.tsx
index 073cf6e..7eae8c7 100644
--- a/packages/app/src/components/Root/LogoIcon.tsx
+++ b/packages/app/src/components/Root/LogoIcon.tsx
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 import React from 'react';
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index b1164a3..5400421 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 import React, { PropsWithChildren } from 'react';
diff --git a/packages/app/src/components/Root/index.ts b/packages/app/src/components/Root/index.ts
index dff706f..3528881 100644
--- a/packages/app/src/components/Root/index.ts
+++ b/packages/app/src/components/Root/index.ts
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 export { Root } from './Root';
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index d98153f..54a05ee 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -1,16 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 import React from 'react';
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index 928b820..9f11d0c 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -16,2 +16,3 @@ import {
   SearchResult,
+  SearchPagination,
   useSearch,
@@ -111,2 +112,3 @@ const SearchPage = () => {
           <Grid item xs={9}>
+            <SearchPagination />
             <SearchResult>
diff --git a/packages/backend/Dockerfile b/packages/backend/Dockerfile
index 682798b..f0be3ac 100644
--- a/packages/backend/Dockerfile
+++ b/packages/backend/Dockerfile
@@ -15,5 +15,6 @@ FROM node:16-bullseye-slim
 # in which case you should also move better-sqlite3 to "devDependencies" in package.json.
-RUN apt-get update && \
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update && \
     apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
-    rm -rf /var/lib/apt/lists/* && \
     yarn config set python /usr/bin/python3
@@ -22,2 +23,6 @@ RUN apt-get update && \
 USER node
+
+# This should create the app dir as `node`.
+# If it is instead created as `root` then the `tar` command below will fail: `can't create directory 'packages/': Permission denied`.
+# If this occurs, then ensure BuildKit is enabled (`DOCKER_BUILDKIT=1`) so the app dir is correctly created as `node`.
 WORKDIR /app
@@ -33,3 +38,4 @@ RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
 
-RUN yarn install --frozen-lockfile --production --network-timeout 300000 && rm -rf "$(yarn cache dir)"
+RUN --mount=type=cache,target=/home/node/.cache/yarn,sharing=locked,uid=1000,gid=1000 \
+    yarn install --frozen-lockfile --production --network-timeout 300000
 
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 294edd3..7154350 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -19,19 +19,19 @@
     "app": "link:../app",
-    "@backstage/backend-common": "^0.15.1-next.3",
-    "@backstage/backend-tasks": "^0.3.5-next.1",
-    "@backstage/catalog-model": "^1.1.1-next.0",
-    "@backstage/catalog-client": "^1.1.0-next.2",
-    "@backstage/config": "^1.0.2-next.0",
-    "@backstage/plugin-app-backend": "^0.3.36-next.3",
-    "@backstage/plugin-auth-backend": "^0.16.0-next.3",
-    "@backstage/plugin-auth-node": "^0.2.5-next.3",
-    "@backstage/plugin-catalog-backend": "^1.4.0-next.3",
-    "@backstage/plugin-permission-common": "^0.6.4-next.2",
-    "@backstage/plugin-permission-node": "^0.6.5-next.3",
-    "@backstage/plugin-proxy-backend": "^0.2.30-next.2",
-    "@backstage/plugin-scaffolder-backend": "^1.6.0-next.3",
-    "@backstage/plugin-search-backend": "^1.0.2-next.1",
-    "@backstage/plugin-search-backend-module-pg": "^0.4.0-next.2",
-    "@backstage/plugin-search-backend-node": "^1.0.2-next.2",
-    "@backstage/plugin-techdocs-backend": "^1.3.0-next.2",
+    "@backstage/backend-common": "^0.15.2",
+    "@backstage/backend-tasks": "^0.3.6",
+    "@backstage/catalog-model": "^1.1.2",
+    "@backstage/catalog-client": "^1.1.1",
+    "@backstage/config": "^1.0.3",
+    "@backstage/plugin-app-backend": "^0.3.37",
+    "@backstage/plugin-auth-backend": "^0.17.0",
+    "@backstage/plugin-auth-node": "^0.2.6",
+    "@backstage/plugin-catalog-backend": "^1.5.0",
+    "@backstage/plugin-permission-common": "^0.7.0",
+    "@backstage/plugin-permission-node": "^0.7.0",
+    "@backstage/plugin-proxy-backend": "^0.2.31",
+    "@backstage/plugin-scaffolder-backend": "^1.7.0",
+    "@backstage/plugin-search-backend": "^1.1.0",
+    "@backstage/plugin-search-backend-module-pg": "^0.4.1",
+    "@backstage/plugin-search-backend-node": "^1.0.3",
+    "@backstage/plugin-techdocs-backend": "^1.4.0",
     "better-sqlite3": "^7.5.0",
@@ -44,3 +44,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.19.0-next.3",
+    "@backstage/cli": "^0.20.0",
     "@types/dockerode": "^3.3.0",
diff --git a/yarn.lock b/yarn.lock
new file mode 100644
index 0000000..5ad7fe2
--- /dev/null
+++ b/yarn.lock
@@ -0,0 +1 @@
+# intentionally left empty
