diff --git a/.dockerignore b/.dockerignore
index 63c9c34..05edb62 100644
--- a/.dockerignore
+++ b/.dockerignore
@@ -1,5 +1,8 @@
 .git
+.yarn/cache
+.yarn/install-state.gz
 node_modules
-packages
-!packages/backend/dist
+packages/*/src
+packages/*/node_modules
 plugins
+*.local.yaml
diff --git a/.gitignore b/.gitignore
index d16a8d3..d452ac2 100644
--- a/.gitignore
+++ b/.gitignore
@@ -17,2 +17,11 @@ node_modules/
 
+# Yarn 3 files
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+
 # Node version directives
@@ -39 +48,4 @@ site
 *-credentials.yaml
+
+# vscode database functionality support files
+*.session.sql
diff --git a/app-config.production.yaml b/app-config.production.yaml
index 5e36c23..6535d96 100644
--- a/app-config.production.yaml
+++ b/app-config.production.yaml
@@ -1,3 +1,3 @@
 app:
-  # Should be the same as backend.baseUrl when using the `app-backend` plugin
+  # Should be the same as backend.baseUrl when using the `app-backend` plugin.
   baseUrl: http://localhost:7007
@@ -5,2 +5,7 @@ app:
 backend:
+  # Note that the baseUrl should be the URL that the browser and other clients
+  # should use when communicating with the backend, i.e. it needs to be
+  # reachable not just from within the backend host, but from all of your
+  # callers. When its value is "http://localhost:7007", it's strictly private
+  # and can't be reached by others.
   baseUrl: http://localhost:7007
@@ -8 +13,28 @@ backend:
     port: 7007
+    # The following host directive binds to all IPv4 interfaces when its value
+    # is "0.0.0.0". This is the most permissive setting. The right value depends
+    # on your specific deployment. If you remove the host line entirely, the
+    # backend will bind on the interface that corresponds to the backend.baseUrl
+    # hostname.
+    host: 0.0.0.0
+
+  # config options: https://node-postgres.com/api/client
+  database:
+    client: pg
+    connection:
+      host: ${POSTGRES_HOST}
+      port: ${POSTGRES_PORT}
+      user: ${POSTGRES_USER}
+      password: ${POSTGRES_PASSWORD}
+      # https://node-postgres.com/features/ssl
+      # you can set the sslmode configuration option via the `PGSSLMODE` environment variable
+      # see https://www.postgresql.org/docs/current/libpq-ssl.html Table 33.1. SSL Mode Descriptions (e.g. require)
+      # ssl:
+      #   ca: # if you have a CA file and want to verify it you can uncomment this section
+      #     $file: <file-path>/ca/server.crt
+
+catalog:
+  # Overrides the default list locations from app-config.yaml as these contain example data.
+  # See https://backstage.io/docs/features/software-catalog/software-catalog-overview#adding-components-to-the-catalog for more details
+  # on how to get entities into the catalog.
+  locations: []
diff --git a/app-config.yaml b/app-config.yaml
index e7732cf..4a058de 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -17,2 +17,5 @@ backend:
     port: 7007
+    # Uncomment the following host directive to bind to all IPv4 interfaces and
+    # not just the baseUrl hostname.
+    # host: 0.0.0.0
   csp:
@@ -23,4 +26,6 @@ backend:
     origin: http://localhost:3000
-    methods: [GET, POST, PUT, DELETE]
+    methods: [GET, HEAD, PATCH, POST, PUT, DELETE]
     credentials: true
+  # This is for local development only, it is not recommended to use this in production
+  # The production database configuration is stored in app-config.production.yaml
   database:
@@ -35,2 +40,4 @@ integrations:
     - host: github.com
+      # This is a Personal Access Token or PAT from GitHub. You can find out how to generate this token, and more information
+      # about setting up the GitHub integration here: https://backstage.io/docs/getting-started/configuration#setting-up-a-github-integration
       token: ${GITHUB_TOKEN}
@@ -69,36 +76,28 @@ catalog:
   rules:
-    - allow: [Component, System, API, Group, User, Resource, Location]
+    - allow: [Component, System, API, Resource, Location]
   locations:
-    # Backstage example components
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-components.yaml
+    # Local example data, file locations are relative to the backend process, typically `packages/backend`
+    - type: file
+      target: ../../examples/entities.yaml
 
-    # Backstage example systems
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-systems.yaml
-
-    # Backstage example APIs
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-apis.yaml
-
-    # Backstage example resources
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all-resources.yaml
-
-    # Backstage example organization groups
-    - type: url
-      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme/org.yaml
-
-    # Backstage example templates
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/react-ssr-template/template.yaml
-      rules:
-        - allow: [Template]
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/springboot-grpc-template/template.yaml
+    # Local example template
+    - type: file
+      target: ../../examples/template/template.yaml
       rules:
         - allow: [Template]
-    - type: url
-      target: https://github.com/backstage/software-templates/blob/main/scaffolder-templates/docs-template/template.yaml
+
+    # Local example organizational data
+    - type: file
+      target: ../../examples/org.yaml
       rules:
-        - allow: [Template]
+        - allow: [User, Group]
+
+    ## Uncomment these lines to add more example data
+    # - type: url
+    #   target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/all.yaml
+
+    ## Uncomment these lines to add an example org
+    # - type: url
+    #   target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme-corp.yaml
+    #   rules:
+    #     - allow: [User, Group]
diff --git a/backstage.json b/backstage.json
index 1587a66..3e73a3c 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "1.0.0"
+  "version": "1.7.0-next.0"
 }
diff --git a/examples/entities.yaml b/examples/entities.yaml
new file mode 100644
index 0000000..447e8b1
--- /dev/null
+++ b/examples/entities.yaml
@@ -0,0 +1,41 @@
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-system
+apiVersion: backstage.io/v1alpha1
+kind: System
+metadata:
+  name: examples
+spec:
+  owner: guests
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-component
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: example-website
+spec:
+  type: website
+  lifecycle: experimental
+  owner: guests
+  system: examples
+  providesApis: [example-grpc-api]
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-api
+apiVersion: backstage.io/v1alpha1
+kind: API
+metadata:
+  name: example-grpc-api
+spec:
+  type: grpc
+  lifecycle: experimental
+  owner: guests
+  system: examples
+  definition: |
+    syntax = "proto3";
+
+    service Exampler {
+      rpc Example (ExampleMessage) returns (ExampleMessage) {};
+    }
+
+    message ExampleMessage {
+      string example = 1;
+    };
diff --git a/examples/org.yaml b/examples/org.yaml
new file mode 100644
index 0000000..a10e81f
--- /dev/null
+++ b/examples/org.yaml
@@ -0,0 +1,17 @@
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-user
+apiVersion: backstage.io/v1alpha1
+kind: User
+metadata:
+  name: guest
+spec:
+  memberOf: [guests]
+---
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-group
+apiVersion: backstage.io/v1alpha1
+kind: Group
+metadata:
+  name: guests
+spec:
+  type: team
+  children: []
diff --git a/examples/template/content/catalog-info.yaml b/examples/template/content/catalog-info.yaml
new file mode 100644
index 0000000..d4ccca4
--- /dev/null
+++ b/examples/template/content/catalog-info.yaml
@@ -0,0 +1,8 @@
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: ${{ values.name | dump }}
+spec:
+  type: service
+  owner: user:guest
+  lifecycle: experimental
diff --git a/examples/template/content/index.js b/examples/template/content/index.js
new file mode 100644
index 0000000..071ce5a
--- /dev/null
+++ b/examples/template/content/index.js
@@ -0,0 +1 @@
+console.log('Hello from ${{ values.name }}!');
diff --git a/examples/template/content/package.json b/examples/template/content/package.json
new file mode 100644
index 0000000..86f968a
--- /dev/null
+++ b/examples/template/content/package.json
@@ -0,0 +1,5 @@
+{
+  "name": "${{ values.name }}",
+  "private": true,
+  "dependencies": {}
+}
diff --git a/examples/template/template.yaml b/examples/template/template.yaml
new file mode 100644
index 0000000..50052b7
--- /dev/null
+++ b/examples/template/template.yaml
@@ -0,0 +1,74 @@
+apiVersion: scaffolder.backstage.io/v1beta3
+# https://backstage.io/docs/features/software-catalog/descriptor-format#kind-template
+kind: Template
+metadata:
+  name: example-nodejs-template
+  title: Example Node.js Template
+  description: An example template for the scaffolder that creates a simple Node.js service
+spec:
+  owner: user:guest
+  type: service
+
+  # These parameters are used to generate the input form in the frontend, and are
+  # used to gather input data for the execution of the template.
+  parameters:
+    - title: Fill in some steps
+      required:
+        - name
+      properties:
+        name:
+          title: Name
+          type: string
+          description: Unique name of the component
+          ui:autofocus: true
+          ui:options:
+            rows: 5
+    - title: Choose a location
+      required:
+        - repoUrl
+      properties:
+        repoUrl:
+          title: Repository Location
+          type: string
+          ui:field: RepoUrlPicker
+          ui:options:
+            allowedHosts:
+              - github.com
+
+  # These steps are executed in the scaffolder backend, using data that we gathered
+  # via the parameters above.
+  steps:
+    # Each step executes an action, in this case one templates files into the working directory.
+    - id: fetch-base
+      name: Fetch Base
+      action: fetch:template
+      input:
+        url: ./content
+        values:
+          name: ${{ parameters.name }}
+
+    # This step publishes the contents of the working directory to GitHub.
+    - id: publish
+      name: Publish
+      action: publish:github
+      input:
+        allowedHosts: ['github.com']
+        description: This is ${{ parameters.name }}
+        repoUrl: ${{ parameters.repoUrl }}
+
+    # The final step is to register our new component in the catalog.
+    - id: register
+      name: Register
+      action: catalog:register
+      input:
+        repoContentsUrl: ${{ steps.publish.output.repoContentsUrl }}
+        catalogInfoPath: '/catalog-info.yaml'
+
+  # Outputs are displayed to the user after a successful execution of the template.
+  output:
+    links:
+      - title: Repository
+        url: ${{ steps.publish.output.remoteUrl }}
+      - title: Open in catalog
+        icon: catalog
+        entityRef: ${{ steps.register.output.entityRef }}
diff --git a/package.json b/package.json
index b5fadf8..d8040bb 100644
--- a/package.json
+++ b/package.json
@@ -15,6 +15,5 @@
     "tsc:full": "tsc --skipLibCheck false --incremental false",
-    "clean": "backstage-cli clean && lerna run clean",
-    "diff": "lerna run diff --",
-    "test": "backstage-cli test",
-    "test:all": "lerna run test -- --coverage",
+    "clean": "backstage-cli repo clean",
+    "test": "backstage-cli repo test",
+    "test:all": "backstage-cli repo test --coverage",
     "lint": "backstage-cli repo lint --since origin/master",
@@ -23,3 +22,3 @@
     "create-plugin": "backstage-cli create-plugin --scope internal",
-    "remove-plugin": "backstage-cli remove-plugin"
+    "new": "backstage-cli new --scope internal"
   },
@@ -32,3 +31,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.16.0",
+    "@backstage/cli": "^0.20.0-next.0",
     "@spotify/prettier-config": "^12.0.0",
@@ -37,3 +36,8 @@
     "prettier": "^2.3.2",
-    "typescript": "~4.5.4"
+    "typescript": "~4.6.4",
+    "node-gyp": "^9.0.0"
+  },
+  "resolutions": {
+    "@types/react": "^17",
+    "@types/react-dom": "^17"
   },
diff --git a/packages/README.md b/packages/README.md
new file mode 100644
index 0000000..6327fa0
--- /dev/null
+++ b/packages/README.md
@@ -0,0 +1,9 @@
+# The Packages Folder
+
+This is where your own applications and centrally managed libraries live, each
+in a separate folder of its own.
+
+From the start there's an `app` folder (for the frontend) and a `backend` folder
+(for the Node backend), but you can also add more modules in here that house
+your core additions and adaptations, such as themes, common React component
+libraries, utilities, and similar.
diff --git a/packages/app/cypress.json b/packages/app/cypress.json
index 5de7ebf..0cb845a 100644
--- a/packages/app/cypress.json
+++ b/packages/app/cypress.json
@@ -3,3 +3,4 @@
   "fixturesFolder": false,
-  "pluginsFile": false
+  "pluginsFile": false,
+  "retries": 3
 }
diff --git a/packages/app/package.json b/packages/app/package.json
index 0cfef69..296306a 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -8,25 +8,39 @@
   },
+  "scripts": {
+    "start": "backstage-cli package start",
+    "build": "backstage-cli package build",
+    "clean": "backstage-cli package clean",
+    "test": "backstage-cli package test",
+    "lint": "backstage-cli package lint",
+    "test:e2e": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:dev",
+    "test:e2e:ci": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:run",
+    "cy:dev": "cypress open",
+    "cy:run": "cypress run --browser chrome"
+  },
   "dependencies": {
-    "@backstage/app-defaults": "^1.0.0",
-    "@backstage/catalog-model": "^1.0.0",
-    "@backstage/cli": "^0.16.0",
-    "@backstage/core-app-api": "^1.0.0",
-    "@backstage/core-components": "^0.9.2",
-    "@backstage/core-plugin-api": "^1.0.0",
-    "@backstage/integration-react": "^1.0.0",
-    "@backstage/plugin-api-docs": "^0.8.3",
-    "@backstage/plugin-catalog": "^1.0.0",
-    "@backstage/plugin-catalog-common": "^1.0.0",
-    "@backstage/plugin-catalog-graph": "^0.2.15",
-    "@backstage/plugin-catalog-import": "^0.8.6",
-    "@backstage/plugin-catalog-react": "^1.0.0",
-    "@backstage/plugin-github-actions": "^0.5.3",
-    "@backstage/plugin-org": "^0.5.3",
-    "@backstage/plugin-permission-react": "^0.3.4",
-    "@backstage/plugin-scaffolder": "^1.0.0",
-    "@backstage/plugin-search": "^0.7.4",
-    "@backstage/plugin-tech-radar": "^0.5.10",
-    "@backstage/plugin-techdocs": "^1.0.0",
-    "@backstage/plugin-user-settings": "^0.4.2",
-    "@backstage/theme": "^0.2.15",
+    "@backstage/app-defaults": "^1.0.7-next.0",
+    "@backstage/catalog-model": "^1.1.2-next.0",
+    "@backstage/cli": "^0.20.0-next.0",
+    "@backstage/core-app-api": "^1.1.1-next.0",
+    "@backstage/core-components": "^0.11.2-next.0",
+    "@backstage/core-plugin-api": "^1.0.7-next.0",
+    "@backstage/integration-react": "^1.1.5-next.0",
+    "@backstage/plugin-api-docs": "^0.8.10-next.0",
+    "@backstage/plugin-catalog": "^1.5.2-next.0",
+    "@backstage/plugin-catalog-common": "^1.0.7-next.0",
+    "@backstage/plugin-catalog-graph": "^0.2.22-next.0",
+    "@backstage/plugin-catalog-import": "^0.8.13-next.0",
+    "@backstage/plugin-catalog-react": "^1.1.5-next.0",
+    "@backstage/plugin-github-actions": "^0.5.10-next.0",
+    "@backstage/plugin-org": "^0.5.10-next.0",
+    "@backstage/plugin-permission-react": "^0.4.6-next.0",
+    "@backstage/plugin-scaffolder": "^1.7.0-next.0",
+    "@backstage/plugin-search": "^1.0.3-next.0",
+    "@backstage/plugin-search-react": "^1.1.1-next.0",
+    "@backstage/plugin-tech-radar": "^0.5.17-next.0",
+    "@backstage/plugin-techdocs": "^1.3.3-next.0",
+    "@backstage/plugin-techdocs-react": "^1.0.5-next.0",
+    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.5-next.0",
+    "@backstage/plugin-user-settings": "^0.5.0-next.0",
+    "@backstage/theme": "^0.2.16",
     "@material-ui/core": "^4.12.2",
@@ -36,16 +50,15 @@
     "react-dom": "^17.0.2",
-    "react-router": "6.0.0-beta.0",
-    "react-router-dom": "6.0.0-beta.0",
-    "react-use": "^15.3.3"
+    "react-router": "^6.3.0",
+    "react-router-dom": "^6.3.0",
+    "react-use": "^17.2.4"
   },
   "devDependencies": {
-    "@backstage/test-utils": "^1.0.0",
+    "@backstage/test-utils": "^1.2.1-next.0",
     "@testing-library/jest-dom": "^5.10.1",
     "@testing-library/react": "^12.1.3",
-    "@testing-library/user-event": "^12.0.7",
-    "@types/jest": "^26.0.7",
-    "@types/node": "^14.14.32",
+    "@testing-library/user-event": "^14.0.0",
+    "@types/node": "^16.11.26",
     "@types/react-dom": "*",
     "cross-env": "^7.0.0",
-    "cypress": "^7.3.0",
+    "cypress": "^9.7.0",
     "eslint-plugin-cypress": "^2.10.3",
@@ -53,13 +66,2 @@
   },
-  "scripts": {
-    "start": "backstage-cli package start",
-    "build": "backstage-cli package build",
-    "clean": "backstage-cli package clean",
-    "test": "backstage-cli package test",
-    "lint": "backstage-cli package lint",
-    "test:e2e": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:dev",
-    "test:e2e:ci": "cross-env PORT=3001 start-server-and-test start http://localhost:3001 cy:run",
-    "cy:dev": "cypress open",
-    "cy:run": "cypress run"
-  },
   "browserslist": {
diff --git a/packages/app/src/App.tsx b/packages/app/src/App.tsx
index f4ff424..4d5ace5 100644
--- a/packages/app/src/App.tsx
+++ b/packages/app/src/App.tsx
@@ -21,2 +21,4 @@ import {
 } from '@backstage/plugin-techdocs';
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+import { ReportIssue } from '@backstage/plugin-techdocs-module-addons-contrib';
 import { UserSettingsPage } from '@backstage/plugin-user-settings';
@@ -31,3 +33,3 @@ import { FlatRoutes } from '@backstage/core-app-api';
 import { CatalogGraphPage } from '@backstage/plugin-catalog-graph';
-import { PermissionedRoute } from '@backstage/plugin-permission-react';
+import { RequirePermission } from '@backstage/plugin-permission-react';
 import { catalogEntityCreatePermission } from '@backstage/plugin-catalog-common/alpha';
@@ -58,3 +60,3 @@ const routes = (
   <FlatRoutes>
-    <Navigate key="/" to="catalog" />
+    <Route path="/" element={<Navigate to="/catalog" />} />
     <Route path="/catalog" element={<CatalogIndexPage />} />
@@ -70,3 +72,7 @@ const routes = (
       element={<TechDocsReaderPage />}
-    />
+    >
+      <TechDocsAddons>
+        <ReportIssue />
+      </TechDocsAddons>
+    </Route>
     <Route path="/create" element={<ScaffolderPage />} />
@@ -77,6 +83,9 @@ const routes = (
     />
-    <PermissionedRoute
+    <Route
       path="/catalog-import"
-      permission={catalogEntityCreatePermission}
-      element={<CatalogImportPage />}
+      element={
+        <RequirePermission permission={catalogEntityCreatePermission}>
+          <CatalogImportPage />
+        </RequirePermission>
+      }
     />
diff --git a/packages/app/src/components/Root/LogoFull.tsx b/packages/app/src/components/Root/LogoFull.tsx
index c7b1c84..47e3b73 100644
--- a/packages/app/src/components/Root/LogoFull.tsx
+++ b/packages/app/src/components/Root/LogoFull.tsx
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 import React from 'react';
diff --git a/packages/app/src/components/Root/LogoIcon.tsx b/packages/app/src/components/Root/LogoIcon.tsx
index 073cf6e..7eae8c7 100644
--- a/packages/app/src/components/Root/LogoIcon.tsx
+++ b/packages/app/src/components/Root/LogoIcon.tsx
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 import React from 'react';
diff --git a/packages/app/src/components/Root/Root.tsx b/packages/app/src/components/Root/Root.tsx
index 90738c4..5400421 100644
--- a/packages/app/src/components/Root/Root.tsx
+++ b/packages/app/src/components/Root/Root.tsx
@@ -1,18 +1,2 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-import React, { useContext, PropsWithChildren } from 'react';
+import React, { PropsWithChildren } from 'react';
 import { Link, makeStyles } from '@material-ui/core';
@@ -34,3 +18,2 @@ import {
   sidebarConfig,
-  SidebarContext,
   SidebarDivider,
@@ -41,2 +24,3 @@ import {
   SidebarSpace,
+  useSidebarOpenState,
 } from '@backstage/core-components';
@@ -62,3 +46,3 @@ const SidebarLogo = () => {
   const classes = useSidebarLogoStyles();
-  const { isOpen } = useContext(SidebarContext);
+  const { isOpen } = useSidebarOpenState();
 
@@ -71,2 +55,3 @@ const SidebarLogo = () => {
         className={classes.link}
+        aria-label="Home"
       >
diff --git a/packages/app/src/components/Root/index.ts b/packages/app/src/components/Root/index.ts
index dff706f..3528881 100644
--- a/packages/app/src/components/Root/index.ts
+++ b/packages/app/src/components/Root/index.ts
@@ -1,17 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
 export { Root } from './Root';
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 84d0944..54a05ee 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -1,16 +1 @@
-/*
- * Copyright 2020 The Backstage Authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
 import React from 'react';
@@ -70,2 +55,13 @@ import {
 
+import { TechDocsAddons } from '@backstage/plugin-techdocs-react';
+import { ReportIssue } from '@backstage/plugin-techdocs-module-addons-contrib';
+
+const techdocsContent = (
+  <EntityTechdocsContent>
+    <TechDocsAddons>
+      <ReportIssue />
+    </TechDocsAddons>
+  </EntityTechdocsContent>
+);
+
 const cicdContent = (
@@ -169,3 +165,3 @@ const serviceEntityPage = (
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      {techdocsContent}
     </EntityLayout.Route>
@@ -196,3 +192,3 @@ const websiteEntityPage = (
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      {techdocsContent}
     </EntityLayout.Route>
@@ -215,3 +211,3 @@ const defaultEntityPage = (
     <EntityLayout.Route path="/docs" title="Docs">
-      <EntityTechdocsContent />
+      {techdocsContent}
     </EntityLayout.Route>
@@ -315,3 +311,6 @@ const systemPage = (
         </Grid>
-        <Grid item md={6}>
+        <Grid item md={4} xs={12}>
+          <EntityLinksCard />
+        </Grid>
+        <Grid item md={8}>
           <EntityHasComponentsCard variant="gridItem" />
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index cd4603e..928b820 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -10,3 +10,5 @@ import { TechDocsSearchResultListItem } from '@backstage/plugin-techdocs';
 
+import { SearchType } from '@backstage/plugin-search';
 import {
+  DefaultResultListItem,
   SearchBar,
@@ -14,6 +16,4 @@ import {
   SearchResult,
-  SearchType,
-  DefaultResultListItem,
   useSearch,
-} from '@backstage/plugin-search';
+} from '@backstage/plugin-search-react';
 import {
@@ -114,3 +114,3 @@ const SearchPage = () => {
                 <List>
-                  {results.map(({ type, document }) => {
+                  {results.map(({ type, document, highlight, rank }) => {
                     switch (type) {
@@ -121,2 +121,4 @@ const SearchPage = () => {
                             result={document}
+                            highlight={highlight}
+                            rank={rank}
                           />
@@ -128,2 +130,4 @@ const SearchPage = () => {
                             result={document}
+                            highlight={highlight}
+                            rank={rank}
                           />
@@ -135,2 +139,4 @@ const SearchPage = () => {
                             result={document}
+                            highlight={highlight}
+                            rank={rank}
                           />
diff --git a/packages/backend/Dockerfile b/packages/backend/Dockerfile
index dd35d4d..682798b 100644
--- a/packages/backend/Dockerfile
+++ b/packages/backend/Dockerfile
@@ -13,4 +13,16 @@ FROM node:16-bullseye-slim
 
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends libsqlite3-dev python3 build-essential && \
+    rm -rf /var/lib/apt/lists/* && \
+    yarn config set python /usr/bin/python3
+
+# From here on we use the least-privileged `node` user to run the backend.
+USER node
 WORKDIR /app
 
+# This switches many Node.js dependencies to production mode.
+ENV NODE_ENV production
+
 # Copy repo skeleton first, to avoid unnecessary docker cache invalidation.
@@ -18,11 +30,5 @@ WORKDIR /app
 # and along with yarn.lock and the root package.json, that's enough to run yarn install.
-COPY yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
+COPY --chown=node:node yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
 RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
 
-# install sqlite3 dependencies
-RUN apt-get update && \
-    apt-get install -y libsqlite3-dev python3 cmake g++ && \
-    rm -rf /var/lib/apt/lists/* && \
-    yarn config set python /usr/bin/python3
-
 RUN yarn install --frozen-lockfile --production --network-timeout 300000 && rm -rf "$(yarn cache dir)"
@@ -30,5 +36,5 @@ RUN yarn install --frozen-lockfile --production --network-timeout 300000 && rm -
 # Then copy the rest of the backend bundle, along with any other files we might want.
-COPY packages/backend/dist/bundle.tar.gz app-config.yaml ./
+COPY --chown=node:node packages/backend/dist/bundle.tar.gz app-config*.yaml ./
 RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
 
-CMD ["node", "packages/backend", "--config", "app-config.yaml"]
+CMD ["node", "packages/backend", "--config", "app-config.yaml", "--config", "app-config.production.yaml"]
diff --git a/packages/backend/package.json b/packages/backend/package.json
index f5d216b..8b3095e 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -15,4 +15,3 @@
     "clean": "backstage-cli package clean",
-    "build-image": "docker build ../.. -f Dockerfile --tag backstage",
-    "migrate:create": "knex migrate:make -x ts"
+    "build-image": "docker build ../.. -f Dockerfile --tag backstage"
   },
@@ -20,19 +19,20 @@
     "app": "link:../app",
-    "@backstage/backend-common": "^0.13.1",
-    "@backstage/backend-tasks": "^0.2.1",
-    "@backstage/catalog-model": "^1.0.0",
-    "@backstage/catalog-client": "^1.0.0",
-    "@backstage/config": "^1.0.0",
-    "@backstage/plugin-app-backend": "^0.3.30",
-    "@backstage/plugin-auth-backend": "^0.12.2",
-    "@backstage/plugin-catalog-backend": "^1.0.0",
-    "@backstage/plugin-permission-common": "^0.5.3",
-    "@backstage/plugin-permission-node": "^0.5.5",
-    "@backstage/plugin-proxy-backend": "^0.2.24",
-    "@backstage/plugin-scaffolder-backend": "^1.0.0",
-    "@backstage/plugin-search-backend": "^0.4.8",
-    "@backstage/plugin-search-backend-node": "^0.5.2",
-    "@backstage/plugin-techdocs-backend": "^1.0.0",
-    "@gitbeaker/node": "^34.6.0",
-    "@octokit/rest": "^18.5.3",
+    "@backstage/backend-common": "^0.15.2-next.0",
+    "@backstage/backend-tasks": "^0.3.6-next.0",
+    "@backstage/catalog-model": "^1.1.2-next.0",
+    "@backstage/catalog-client": "^1.1.1-next.0",
+    "@backstage/config": "^1.0.3-next.0",
+    "@backstage/plugin-app-backend": "^0.3.37-next.0",
+    "@backstage/plugin-auth-backend": "^0.17.0-next.0",
+    "@backstage/plugin-auth-node": "^0.2.6-next.0",
+    "@backstage/plugin-catalog-backend": "^1.4.1-next.0",
+    "@backstage/plugin-permission-common": "^0.6.5-next.0",
+    "@backstage/plugin-permission-node": "^0.6.6-next.0",
+    "@backstage/plugin-proxy-backend": "^0.2.31-next.0",
+    "@backstage/plugin-scaffolder-backend": "^1.7.0-next.0",
+    "@backstage/plugin-search-backend": "^1.0.3-next.0",
+    "@backstage/plugin-search-backend-module-pg": "^0.4.1-next.0",
+    "@backstage/plugin-search-backend-node": "^1.0.3-next.0",
+    "@backstage/plugin-techdocs-backend": "^1.3.1-next.0",
+    "better-sqlite3": "^7.5.0",
     "dockerode": "^3.3.1",
@@ -40,4 +40,3 @@
     "express-promise-router": "^4.1.0",
-    "knex": "^0.21.6",
-    "better-sqlite3": "^7.5.0",
+    "pg": "^8.3.0",
     "winston": "^3.2.1"
@@ -45,6 +44,7 @@
   "devDependencies": {
-    "@backstage/cli": "^0.16.0",
+    "@backstage/cli": "^0.20.0-next.0",
     "@types/dockerode": "^3.3.0",
+    "@types/express-serve-static-core": "^4.17.5",
     "@types/express": "^4.17.6",
-    "@types/express-serve-static-core": "^4.17.5"
+    "@types/luxon": "^2.0.4"
   },
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index 70bc66b..c4736a5 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -32,2 +32,3 @@ import { PluginEnvironment } from './types';
 import { ServerPermissionClient } from '@backstage/plugin-permission-node';
+import { DefaultIdentityClient } from '@backstage/plugin-auth-node';
 
@@ -38,5 +39,9 @@ function makeCreateEnv(config: Config) {
   const cacheManager = CacheManager.fromConfig(config);
-  const databaseManager = DatabaseManager.fromConfig(config);
+  const databaseManager = DatabaseManager.fromConfig(config, { logger: root });
   const tokenManager = ServerTokenManager.noop();
   const taskScheduler = TaskScheduler.fromConfig(config);
+
+  const identity = DefaultIdentityClient.create({
+    discovery,
+  });
   const permissions = ServerPermissionClient.fromConfig(config, {
@@ -63,2 +68,3 @@ function makeCreateEnv(config: Config) {
       permissions,
+      identity,
     };
@@ -106,3 +112,3 @@ module.hot?.accept();
 main().catch(error => {
-  console.error(`Backend failed to start up, ${error}`);
+  console.error('Backend failed to start up', error);
   process.exit(1);
diff --git a/packages/backend/src/plugins/auth.ts b/packages/backend/src/plugins/auth.ts
index 1476e66..77eb6aa 100644
--- a/packages/backend/src/plugins/auth.ts
+++ b/packages/backend/src/plugins/auth.ts
@@ -1,2 +1,6 @@
-import { createRouter } from '@backstage/plugin-auth-backend';
+import {
+  createRouter,
+  providers,
+  defaultAuthProviderFactories,
+} from '@backstage/plugin-auth-backend';
 import { Router } from 'express';
@@ -13,2 +17,37 @@ export default async function createPlugin(
     tokenManager: env.tokenManager,
+    providerFactories: {
+      ...defaultAuthProviderFactories,
+
+      // This replaces the default GitHub auth provider with a customized one.
+      // The `signIn` option enables sign-in for this provider, using the
+      // identity resolution logic that's provided in the `resolver` callback.
+      //
+      // This particular resolver makes all users share a single "guest" identity.
+      // It should only be used for testing and trying out Backstage.
+      //
+      // If you want to use a production ready resolver you can switch to
+      // the one that is commented out below, it looks up a user entity in the
+      // catalog using the GitHub username of the authenticated user.
+      // That resolver requires you to have user entities populated in the catalog,
+      // for example using https://backstage.io/docs/integrations/github/org
+      //
+      // There are other resolvers to choose from, and you can also create
+      // your own, see the auth documentation for more details:
+      //
+      //   https://backstage.io/docs/auth/identity-resolver
+      github: providers.github.create({
+        signIn: {
+          resolver(_, ctx) {
+            const userRef = 'user:default/guest'; // Must be a full entity reference
+            return ctx.issueToken({
+              claims: {
+                sub: userRef, // The user's own identity
+                ent: [userRef], // A list of identities that the user claims ownership through
+              },
+            });
+          },
+          // resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
+        },
+      }),
+    },
   });
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 7ce5fcf..ef46f07 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -18,2 +18,3 @@ export default async function createPlugin(
     catalogClient,
+    identity: env.identity,
   });
diff --git a/packages/backend/src/plugins/search.ts b/packages/backend/src/plugins/search.ts
index 8df6b0a..e9469dc 100644
--- a/packages/backend/src/plugins/search.ts
+++ b/packages/backend/src/plugins/search.ts
@@ -23,2 +23,10 @@ export default async function createPlugin(
 
+  const schedule = env.scheduler.createScheduledTaskRunner({
+    frequency: { minutes: 10 },
+    timeout: { minutes: 15 },
+    // A 3 second delay gives the backend server a chance to initialize before
+    // any collators are executed, which may attempt requests against the API.
+    initialDelay: { seconds: 3 },
+  });
+
   // Collators are responsible for gathering documents known to plugins. This
@@ -26,3 +34,3 @@ export default async function createPlugin(
   indexBuilder.addCollator({
-    defaultRefreshIntervalSeconds: 600,
+    schedule,
     factory: DefaultCatalogCollatorFactory.fromConfig(env.config, {
@@ -35,3 +43,3 @@ export default async function createPlugin(
   indexBuilder.addCollator({
-    defaultRefreshIntervalSeconds: 600,
+    schedule,
     factory: DefaultTechDocsCollatorFactory.fromConfig(env.config, {
@@ -46,6 +54,4 @@ export default async function createPlugin(
   const { scheduler } = await indexBuilder.build();
+  scheduler.start();
 
-  // A 3 second delay gives the backend server a chance to initialize before
-  // any collators are executed, which may attempt requests against the API.
-  setTimeout(() => scheduler.start(), 3000);
   useHotCleanup(module, () => scheduler.stop());
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index 0862b0e..9cd2c74 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -10,3 +10,4 @@ import {
 import { PluginTaskScheduler } from '@backstage/backend-tasks';
-import { PermissionAuthorizer } from '@backstage/plugin-permission-common';
+import { PermissionEvaluator } from '@backstage/plugin-permission-common';
+import { IdentityApi } from '@backstage/plugin-auth-node';
 
@@ -21,3 +22,4 @@ export type PluginEnvironment = {
   scheduler: PluginTaskScheduler;
-  permissions: PermissionAuthorizer;
+  permissions: PermissionEvaluator;
+  identity: IdentityApi;
 };
diff --git a/plugins/README.md b/plugins/README.md
new file mode 100644
index 0000000..58dc32c
--- /dev/null
+++ b/plugins/README.md
@@ -0,0 +1,9 @@
+# The Plugins Folder
+
+This is where your own plugins and their associated modules live, each in a
+separate folder of its own.
+
+If you want to create a new plugin here, go to your project root directory, run
+the command `yarn backstage-cli create`, and follow the on-screen instructions.
+
+You can also check out existing plugins on [the plugin marketplace](https://backstage.io/plugins)!
diff --git a/yarn.lock b/yarn.lock
new file mode 100644
index 0000000..5ad7fe2
--- /dev/null
+++ b/yarn.lock
@@ -0,0 +1 @@
+# intentionally left empty
