diff --git a/.dockerignore b/.dockerignore
index 505a7b5..05edb62 100644
--- a/.dockerignore
+++ b/.dockerignore
@@ -1,2 +1,4 @@
 .git
+.yarn/cache
+.yarn/install-state.gz
 node_modules
diff --git a/.gitignore b/.gitignore
index d16a8d3..fdc2a5d 100644
--- a/.gitignore
+++ b/.gitignore
@@ -17,2 +17,11 @@ node_modules/
 
+# Yarn 3 files
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
+
 # Node version directives
diff --git a/app-config.production.yaml b/app-config.production.yaml
index b9dd00b..6535d96 100644
--- a/app-config.production.yaml
+++ b/app-config.production.yaml
@@ -34 +34,7 @@ backend:
       #     $file: <file-path>/ca/server.crt
+
+catalog:
+  # Overrides the default list locations from app-config.yaml as these contain example data.
+  # See https://backstage.io/docs/features/software-catalog/software-catalog-overview#adding-components-to-the-catalog for more details
+  # on how to get entities into the catalog.
+  locations: []
diff --git a/app-config.yaml b/app-config.yaml
index cd6998d..4a058de 100644
--- a/app-config.yaml
+++ b/app-config.yaml
@@ -26,5 +26,5 @@ backend:
     origin: http://localhost:3000
-    methods: [GET, POST, PUT, DELETE]
+    methods: [GET, HEAD, PATCH, POST, PUT, DELETE]
     credentials: true
-  # This is for local developement only, it is not recommended to use this in production
+  # This is for local development only, it is not recommended to use this in production
   # The production database configuration is stored in app-config.production.yaml
diff --git a/backstage.json b/backstage.json
index 42d8c52..0dc1aeb 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "1.3.0-next.1"
+  "version": "1.6.0-next.2"
 }
diff --git a/package.json b/package.json
index 82d95a0..822806f 100644
--- a/package.json
+++ b/package.json
@@ -15,4 +15,3 @@
     "tsc:full": "tsc --skipLibCheck false --incremental false",
-    "clean": "backstage-cli clean && lerna run clean",
-    "diff": "lerna run diff --",
+    "clean": "backstage-cli repo clean",
     "test": "backstage-cli test",
@@ -23,3 +22,3 @@
     "create-plugin": "backstage-cli create-plugin --scope internal",
-    "remove-plugin": "backstage-cli remove-plugin"
+    "new": "backstage-cli new --scope internal"
   },
@@ -32,3 +31,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.17.2-next.1",
+    "@backstage/cli": "^0.19.0-next.2",
     "@spotify/prettier-config": "^12.0.0",
@@ -37,3 +36,4 @@
     "prettier": "^2.3.2",
-    "typescript": "~4.6.4"
+    "typescript": "~4.6.4",
+    "node-gyp": "^9.0.0"
   },
diff --git a/packages/app/cypress.json b/packages/app/cypress.json
index 5de7ebf..0cb845a 100644
--- a/packages/app/cypress.json
+++ b/packages/app/cypress.json
@@ -3,3 +3,4 @@
   "fixturesFolder": false,
-  "pluginsFile": false
+  "pluginsFile": false,
+  "retries": 3
 }
diff --git a/packages/app/package.json b/packages/app/package.json
index 1c6fc3c..c03faa5 100644
--- a/packages/app/package.json
+++ b/packages/app/package.json
@@ -9,27 +9,27 @@
   "dependencies": {
-    "@backstage/app-defaults": "^1.0.3-next.1",
-    "@backstage/catalog-model": "^1.0.3-next.0",
-    "@backstage/cli": "^0.17.2-next.1",
-    "@backstage/core-app-api": "^1.0.3-next.0",
-    "@backstage/core-components": "^0.9.5-next.1",
-    "@backstage/core-plugin-api": "^1.0.3-next.0",
-    "@backstage/integration-react": "^1.1.1-next.1",
-    "@backstage/plugin-api-docs": "^0.8.6-next.1",
-    "@backstage/plugin-catalog": "^1.2.1-next.1",
-    "@backstage/plugin-catalog-common": "^1.0.3-next.0",
-    "@backstage/plugin-catalog-graph": "^0.2.18-next.1",
-    "@backstage/plugin-catalog-import": "^0.8.9-next.1",
-    "@backstage/plugin-catalog-react": "^1.1.1-next.1",
-    "@backstage/plugin-github-actions": "^0.5.6-next.1",
-    "@backstage/plugin-org": "^0.5.6-next.1",
-    "@backstage/plugin-permission-react": "^0.4.2-next.0",
-    "@backstage/plugin-scaffolder": "^1.3.0-next.1",
-    "@backstage/plugin-search": "^0.8.2-next.1",
-    "@backstage/plugin-search-react": "^0.2.1-next.0",
-    "@backstage/plugin-tech-radar": "^0.5.13-next.1",
-    "@backstage/plugin-techdocs": "^1.1.2-next.1",
-    "@backstage/plugin-techdocs-react": "^1.0.1-next.1",
-    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.1-next.1",
-    "@backstage/plugin-user-settings": "^0.4.5-next.1",
-    "@backstage/theme": "^0.2.15",
+    "@backstage/app-defaults": "^1.0.6-next.1",
+    "@backstage/catalog-model": "^1.1.0",
+    "@backstage/cli": "^0.19.0-next.2",
+    "@backstage/core-app-api": "^1.1.0-next.2",
+    "@backstage/core-components": "^0.11.1-next.2",
+    "@backstage/core-plugin-api": "^1.0.6-next.2",
+    "@backstage/integration-react": "^1.1.4-next.1",
+    "@backstage/plugin-api-docs": "^0.8.9-next.2",
+    "@backstage/plugin-catalog": "^1.5.1-next.2",
+    "@backstage/plugin-catalog-common": "^1.0.6-next.0",
+    "@backstage/plugin-catalog-graph": "^0.2.21-next.1",
+    "@backstage/plugin-catalog-import": "^0.8.12-next.2",
+    "@backstage/plugin-catalog-react": "^1.1.4-next.1",
+    "@backstage/plugin-github-actions": "^0.5.9-next.2",
+    "@backstage/plugin-org": "^0.5.9-next.2",
+    "@backstage/plugin-permission-react": "^0.4.5-next.1",
+    "@backstage/plugin-scaffolder": "^1.6.0-next.2",
+    "@backstage/plugin-search": "^1.0.2-next.2",
+    "@backstage/plugin-search-react": "^1.1.0-next.2",
+    "@backstage/plugin-tech-radar": "^0.5.16-next.2",
+    "@backstage/plugin-techdocs": "^1.3.2-next.2",
+    "@backstage/plugin-techdocs-react": "^1.0.4-next.1",
+    "@backstage/plugin-techdocs-module-addons-contrib": "^1.0.4-next.1",
+    "@backstage/plugin-user-settings": "^0.4.8-next.2",
+    "@backstage/theme": "^0.2.16",
     "@material-ui/core": "^4.12.2",
@@ -44,3 +44,3 @@
   "devDependencies": {
-    "@backstage/test-utils": "^1.1.1-next.0",
+    "@backstage/test-utils": "^1.2.0-next.2",
     "@testing-library/jest-dom": "^5.10.1",
@@ -52,3 +52,3 @@
     "cross-env": "^7.0.0",
-    "cypress": "^7.3.0",
+    "cypress": "^9.7.0",
     "eslint-plugin-cypress": "^2.10.3",
@@ -65,3 +65,3 @@
     "cy:dev": "cypress open",
-    "cy:run": "cypress run"
+    "cy:run": "cypress run --browser chrome"
   },
diff --git a/packages/app/src/components/catalog/EntityPage.tsx b/packages/app/src/components/catalog/EntityPage.tsx
index 6ec4da0..d98153f 100644
--- a/packages/app/src/components/catalog/EntityPage.tsx
+++ b/packages/app/src/components/catalog/EntityPage.tsx
@@ -326,3 +326,6 @@ const systemPage = (
         </Grid>
-        <Grid item md={6}>
+        <Grid item md={4} xs={12}>
+          <EntityLinksCard />
+        </Grid>
+        <Grid item md={8}>
           <EntityHasComponentsCard variant="gridItem" />
diff --git a/packages/app/src/components/search/SearchPage.tsx b/packages/app/src/components/search/SearchPage.tsx
index d4c7c92..928b820 100644
--- a/packages/app/src/components/search/SearchPage.tsx
+++ b/packages/app/src/components/search/SearchPage.tsx
@@ -10,3 +10,5 @@ import { TechDocsSearchResultListItem } from '@backstage/plugin-techdocs';
 
+import { SearchType } from '@backstage/plugin-search';
 import {
+  DefaultResultListItem,
   SearchBar,
@@ -14,6 +16,4 @@ import {
   SearchResult,
-  SearchType,
-  DefaultResultListItem,
-} from '@backstage/plugin-search';
-import { useSearch } from '@backstage/plugin-search-react';
+  useSearch,
+} from '@backstage/plugin-search-react';
 import {
@@ -114,3 +114,3 @@ const SearchPage = () => {
                 <List>
-                  {results.map(({ type, document, highlight }) => {
+                  {results.map(({ type, document, highlight, rank }) => {
                     switch (type) {
@@ -122,2 +122,3 @@ const SearchPage = () => {
                             highlight={highlight}
+                            rank={rank}
                           />
@@ -130,2 +131,3 @@ const SearchPage = () => {
                             highlight={highlight}
+                            rank={rank}
                           />
@@ -138,2 +140,3 @@ const SearchPage = () => {
                             highlight={highlight}
+                            rank={rank}
                           />
diff --git a/packages/backend/Dockerfile b/packages/backend/Dockerfile
index a5773aa..682798b 100644
--- a/packages/backend/Dockerfile
+++ b/packages/backend/Dockerfile
@@ -13,5 +13,4 @@ FROM node:16-bullseye-slim
 
-WORKDIR /app
-
-# install sqlite3 dependencies, you can skip this if you don't use sqlite3 in the image
+# Install sqlite3 dependencies. You can skip this if you don't use sqlite3 in the image,
+# in which case you should also move better-sqlite3 to "devDependencies" in package.json.
 RUN apt-get update && \
@@ -21,2 +20,9 @@ RUN apt-get update && \
 
+# From here on we use the least-privileged `node` user to run the backend.
+USER node
+WORKDIR /app
+
+# This switches many Node.js dependencies to production mode.
+ENV NODE_ENV production
+
 # Copy repo skeleton first, to avoid unnecessary docker cache invalidation.
@@ -24,3 +30,3 @@ RUN apt-get update && \
 # and along with yarn.lock and the root package.json, that's enough to run yarn install.
-COPY yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
+COPY --chown=node:node yarn.lock package.json packages/backend/dist/skeleton.tar.gz ./
 RUN tar xzf skeleton.tar.gz && rm skeleton.tar.gz
@@ -30,3 +36,3 @@ RUN yarn install --frozen-lockfile --production --network-timeout 300000 && rm -
 # Then copy the rest of the backend bundle, along with any other files we might want.
-COPY packages/backend/dist/bundle.tar.gz app-config*.yaml ./
+COPY --chown=node:node packages/backend/dist/bundle.tar.gz app-config*.yaml ./
 RUN tar xzf bundle.tar.gz && rm bundle.tar.gz
diff --git a/packages/backend/package.json b/packages/backend/package.json
index 256a0c3..e8ae907 100644
--- a/packages/backend/package.json
+++ b/packages/backend/package.json
@@ -19,18 +19,20 @@
     "app": "link:../app",
-    "@backstage/backend-common": "^0.13.6-next.1",
-    "@backstage/backend-tasks": "^0.3.2-next.1",
-    "@backstage/catalog-model": "^1.0.3-next.0",
-    "@backstage/catalog-client": "^1.0.3-next.0",
+    "@backstage/backend-common": "^0.15.1-next.2",
+    "@backstage/backend-tasks": "^0.3.5-next.0",
+    "@backstage/catalog-model": "^1.1.0",
+    "@backstage/catalog-client": "^1.0.5-next.1",
     "@backstage/config": "^1.0.1",
-    "@backstage/plugin-app-backend": "^0.3.33-next.1",
-    "@backstage/plugin-auth-backend": "^0.14.1-next.1",
-    "@backstage/plugin-catalog-backend": "^1.2.0-next.1",
-    "@backstage/plugin-permission-common": "^0.6.2-next.0",
-    "@backstage/plugin-permission-node": "^0.6.2-next.1",
-    "@backstage/plugin-proxy-backend": "^0.2.27-next.0",
-    "@backstage/plugin-scaffolder-backend": "^1.3.0-next.1",
-    "@backstage/plugin-search-backend": "^0.5.3-next.1",
-    "@backstage/plugin-search-backend-module-pg": "^0.3.4-next.1",
-    "@backstage/plugin-search-backend-node": "^0.6.2-next.1",
-    "@backstage/plugin-techdocs-backend": "^1.1.2-next.1",
+    "@backstage/plugin-app-backend": "^0.3.36-next.2",
+    "@backstage/plugin-auth-backend": "^0.16.0-next.2",
+    "@backstage/plugin-auth-node": "^0.2.5-next.2",
+    "@backstage/plugin-catalog-backend": "^1.4.0-next.2",
+    "@backstage/plugin-permission-common": "^0.6.4-next.1",
+    "@backstage/plugin-permission-node": "^0.6.5-next.2",
+    "@backstage/plugin-proxy-backend": "^0.2.30-next.1",
+    "@backstage/plugin-scaffolder-backend": "^1.6.0-next.2",
+    "@backstage/plugin-search-backend": "^1.0.2-next.0",
+    "@backstage/plugin-search-backend-module-pg": "^0.4.0-next.1",
+    "@backstage/plugin-search-backend-node": "^1.0.2-next.1",
+    "@backstage/plugin-techdocs-backend": "^1.3.0-next.1",
+    "better-sqlite3": "^7.5.0",
     "dockerode": "^3.3.1",
@@ -42,3 +44,3 @@
   "devDependencies": {
-    "@backstage/cli": "^0.17.2-next.1",
+    "@backstage/cli": "^0.19.0-next.2",
     "@types/dockerode": "^3.3.0",
@@ -46,4 +48,3 @@
     "@types/express": "^4.17.6",
-    "@types/luxon": "^2.0.4",
-    "better-sqlite3": "^7.5.0"
+    "@types/luxon": "^2.0.4"
   },
diff --git a/packages/backend/src/index.ts b/packages/backend/src/index.ts
index 70bc66b..c4736a5 100644
--- a/packages/backend/src/index.ts
+++ b/packages/backend/src/index.ts
@@ -32,2 +32,3 @@ import { PluginEnvironment } from './types';
 import { ServerPermissionClient } from '@backstage/plugin-permission-node';
+import { DefaultIdentityClient } from '@backstage/plugin-auth-node';
 
@@ -38,5 +39,9 @@ function makeCreateEnv(config: Config) {
   const cacheManager = CacheManager.fromConfig(config);
-  const databaseManager = DatabaseManager.fromConfig(config);
+  const databaseManager = DatabaseManager.fromConfig(config, { logger: root });
   const tokenManager = ServerTokenManager.noop();
   const taskScheduler = TaskScheduler.fromConfig(config);
+
+  const identity = DefaultIdentityClient.create({
+    discovery,
+  });
   const permissions = ServerPermissionClient.fromConfig(config, {
@@ -63,2 +68,3 @@ function makeCreateEnv(config: Config) {
       permissions,
+      identity,
     };
@@ -106,3 +112,3 @@ module.hot?.accept();
 main().catch(error => {
-  console.error(`Backend failed to start up, ${error}`);
+  console.error('Backend failed to start up', error);
   process.exit(1);
diff --git a/packages/backend/src/plugins/auth.ts b/packages/backend/src/plugins/auth.ts
index 13f18c5..77eb6aa 100644
--- a/packages/backend/src/plugins/auth.ts
+++ b/packages/backend/src/plugins/auth.ts
@@ -20,8 +20,17 @@ export default async function createPlugin(
 
-      // This overrides the default GitHub auth provider with a custom one.
-      // Since the options are empty it will behave just like the default
-      // provider, but if you uncomment the `signIn` section you will enable
-      // sign-in via GitHub. This particular configuration uses a resolver
-      // that matches the username to the user entity name. See the auth
-      // documentation for more details on how to enable and customize sign-in:
+      // This replaces the default GitHub auth provider with a customized one.
+      // The `signIn` option enables sign-in for this provider, using the
+      // identity resolution logic that's provided in the `resolver` callback.
+      //
+      // This particular resolver makes all users share a single "guest" identity.
+      // It should only be used for testing and trying out Backstage.
+      //
+      // If you want to use a production ready resolver you can switch to
+      // the one that is commented out below, it looks up a user entity in the
+      // catalog using the GitHub username of the authenticated user.
+      // That resolver requires you to have user entities populated in the catalog,
+      // for example using https://backstage.io/docs/integrations/github/org
+      //
+      // There are other resolvers to choose from, and you can also create
+      // your own, see the auth documentation for more details:
       //
@@ -29,5 +38,14 @@ export default async function createPlugin(
       github: providers.github.create({
-        // signIn: {
-        //   resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
-        // },
+        signIn: {
+          resolver(_, ctx) {
+            const userRef = 'user:default/guest'; // Must be a full entity reference
+            return ctx.issueToken({
+              claims: {
+                sub: userRef, // The user's own identity
+                ent: [userRef], // A list of identities that the user claims ownership through
+              },
+            });
+          },
+          // resolver: providers.github.resolvers.usernameMatchingUserEntityName(),
+        },
       }),
diff --git a/packages/backend/src/plugins/scaffolder.ts b/packages/backend/src/plugins/scaffolder.ts
index 7ce5fcf..ef46f07 100644
--- a/packages/backend/src/plugins/scaffolder.ts
+++ b/packages/backend/src/plugins/scaffolder.ts
@@ -18,2 +18,3 @@ export default async function createPlugin(
     catalogClient,
+    identity: env.identity,
   });
diff --git a/packages/backend/src/types.ts b/packages/backend/src/types.ts
index 8e0a864..9cd2c74 100644
--- a/packages/backend/src/types.ts
+++ b/packages/backend/src/types.ts
@@ -11,2 +11,3 @@ import { PluginTaskScheduler } from '@backstage/backend-tasks';
 import { PermissionEvaluator } from '@backstage/plugin-permission-common';
+import { IdentityApi } from '@backstage/plugin-auth-node';
 
@@ -22,2 +23,3 @@ export type PluginEnvironment = {
   permissions: PermissionEvaluator;
+  identity: IdentityApi;
 };
