diff --git a/.yarnrc.yml b/.yarnrc.yml
index 76457f8..10392c9 100644
--- a/.yarnrc.yml
+++ b/.yarnrc.yml
@@ -5,3 +5,3 @@ plugins:
     path: .yarn/plugins/@yarnpkg/plugin-backstage.cjs
-    spec: "https://versions.backstage.io/v1/releases/1.47.0-next.1/yarn-plugin"
+    spec: "https://versions.backstage.io/v1/releases/1.48.0-next.0/yarn-plugin"
 
diff --git a/backstage.json b/backstage.json
index dbffe71..8166a8b 100644
--- a/backstage.json
+++ b/backstage.json
@@ -1,3 +1,3 @@
 {
-  "version": "1.47.0-next.1"
+  "version": "1.48.0-next.0"
 }
diff --git a/packages/backend/Dockerfile b/packages/backend/Dockerfile
index ee80245..a70705f 100644
--- a/packages/backend/Dockerfile
+++ b/packages/backend/Dockerfile
@@ -14,3 +14,3 @@
 
-FROM node:22-bookworm-slim
+FROM node:24-trixie-slim
 
